---
title: "Zygote de novo loci"
author: "Chenxin Li"
date: "10/8/2020"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(RColorBrewer)
library(viridis)
library(stringr)
library(svglite)
library(cowplot)
library(forcats)
library(ggrepel)
```
 
# 50kb windows loci coverage

## load data
```{r, message=FALSE}
bin50_list <- list.files("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/FullSetLoci3_cov/", 
                             pattern = "50KB.*.cov", full.names = T)

bin50_data <- sapply(bin50_list, read_delim, simplify = FALSE, col_names = F, delim = "\t") %>% 
  bind_rows(.id = "id")

head(bin50_data)
```
```{r message=F}
bin50_list2 <- list.files("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_zygote_GR_revision/short_long_TE_coverage/", 
                             pattern = "50KB.*.cov", full.names = T)

bin50_data2 <- sapply(bin50_list2, read_delim, simplify = FALSE, col_names = F, delim = "\t") %>% 
  bind_rows(.id = "id")

head(bin50_data2)
```
 
 
```{r message=FALSE}
gene.density <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/50KB_windows/MSU7_filtered_genes_50Kb.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

MSU7_filtered_genes <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/MSU7_filtered_genes.gff", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)  

total_genes <- MSU7_filtered_genes %>% nrow()
```

```{r message=F}
gypsy <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/50KbWindowsCov/home/gent/references/Oryza_sativa.IRGSP-1.Gypsy.out.gff.50KbWindows.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 

gypsy_bed <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/Gypsy.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 

total_gypsy <- gypsy_bed %>% nrow()
```

```{r message=F}
TIR <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/50KbWindowsCov/home/gent/references/Oryza_sativa.IRGSP-1.non-CACTA_DNA.out.gff.50KbWindows.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 

TIR_bed <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/tir.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 

total_TIR <- TIR_bed %>% nrow()
```
 
```{r}
drm2_chh <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/Bed_cov_2020_10_14/50KbLociCov/50KB.CHH_wt-52_dmr100_hypo_cutoffs_5-0.001-0.25-0.3.bed.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

total_drm2_chh <- drm2_chh %>% 
  summarise(total = sum(X9))

total_drm2_chh
```


## keying data
```{r}
beds_total <- bin50_data %>% 
  rbind(bin50_data2) %>% 
  group_by(id) %>% 
  summarise(NN = sum(X9)) %>%  
  mutate(locus = case_when(
    str_detect(id, "egg-zygote_comp") ~ "E loci - Z loci", #1
    str_detect(id, "50KB.egg_loci.bed") ~ "egg siRNA loci", #2
    str_detect(id, "embryo_loci") ~ "embryo siRNA loci", #3
    str_detect(id, "seedling_loci.") ~ "seedling siRNA loci", #4
    str_detect(id, "zygote-egg_intersect") ~ "Z/E loci intersect",  #5
    str_detect(id, "sperm_loci") &
      str_detect(id, "endo") == F ~ "sperm siRNA loci", #6
    str_detect(id, "zygote_loci.bed") ~ "zygote siRNA loci", #7
     str_detect(id, "zygote-egg_comp") ~ "Z loci - E loci", #8
    str_detect(id, "CACTA_DNA") ~ "CACTA", #9
    str_detect(id, "50-250") ~ "short TEs", #10
    str_detect(id, "251") ~ "long TEs" #11 
  )) %>% 
  filter(is.na(locus) == F) %>% 
  ungroup() %>% 
  select(-id) %>% 
  rbind(
    data.frame(NN = c(total_genes, total_gypsy, total_drm2_chh$total, total_TIR),  
               locus = c("genes", "Gypsy", "embryo DRM2 target loci", "TIR"))) %>% 
  arrange(locus)
  

beds_total
```
 
```{r}
bin50_data_select <- bin50_data %>% 
  rbind(bin50_data2) %>% 
  rbind(gene.density %>% 
          mutate(id = "genes")) %>%  #1
  rbind(gypsy %>% 
          mutate(id = "Gypsy")) %>% #2
  rbind(TIR %>% 
          mutate(id = "TIR")) %>% #3
  rbind(drm2_chh %>%   #4 
          mutate(id = "embryo DMR2 target loci")) %>% 
  rename(Chr = X1,
         start = X4,
         end = X5,
         count = X9,
         bases = X10, 
         coverage = X12) %>%  
  mutate(locus = case_when(
    str_detect(id, "egg-zygote_comp") ~ "E loci - Z loci", #1
    str_detect(id, "50KB.egg_loci.bed") ~ "egg siRNA loci", #2
    str_detect(id, "embryo_loci") ~ "embryo siRNA loci", #3
    str_detect(id, "seedling_loci.") ~ "seedling siRNA loci", #4
    str_detect(id, "zygote-egg_intersect") ~ "Z/E loci intersect",  #5
    str_detect(id, "sperm_loci") &
      str_detect(id, "endo") == F ~ "sperm siRNA loci", #6
    str_detect(id, "zygote_loci.bed") ~ "zygote siRNA loci", #7
     str_detect(id, "zygote-egg_comp") ~ "Z loci - E loci", #8
    str_detect(id, "genes") ~ "genes", #9
    str_detect(id, "TIR") ~ "TIR", #10
    str_detect(id, "Gypsy") ~ "Gypsy", #11
    str_detect(id, "embryo DMR2 target loci") ~ "embryo DRM2 target loci", #12
    str_detect(id, "CACTA_DNA") ~ "CACTA", #13
    str_detect(id, "50-250") ~ "short TEs", #14
    str_detect(id, "251") ~ "long TEs" #15 
  )) %>% 
  filter(is.na(locus) == F) 

head(bin50_data_select)
```
```{r}
bin50_data_select %>% 
  group_by(locus) %>% 
  count() %>% 
  arrange(locus)
```
 
 
 
```{r}
bin50_data_select_wide <- 
  bin50_data_select %>% 
  inner_join(beds_total, by = "locus") %>% 
  mutate(RPM = count/NN * 10^6) %>% 
  select(Chr, start, end, locus, RPM) %>% 
  spread(locus, RPM) %>% 
  select(-CACTA, -`Z/E loci intersect`)

head(bin50_data_select_wide)
```
 
 

## PCA
```{r}
pc <- prcomp(t(bin50_data_select_wide[, 4:ncol(bin50_data_select_wide)]))
pc_importance <- as.data.frame(t(summary(pc)$importance))
pc_importance 
```

```{r}
pc$x %>% 
  as.data.frame() %>% 
  mutate(locus_type1 = row.names(.)) %>% 
  mutate(locus_type = case_when(
    locus_type1 == "embryo siRNA loci" ~ "*embryo siRNA loci",
    T ~ locus_type1
  )) %>% 
  mutate(type2 = case_when(
    locus_type == "genes" |
      locus_type == "Gypsy" ~ "Gs",
    T ~ "else"
  )) %>% 
  filter(locus_type != "zygote/egg loci intersect") %>% 
  ggplot(aes(x = PC1, y = PC2)) +
  geom_point(aes(fill = locus_type, color = type2), size = 3, alpha = 0.8, shape = 21) +
  scale_fill_manual(values = c("white","gold", "orange2","blue2", "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2", "slateblue4", "grey80"),
                    limits = c("genes", "short TEs", "TIR", "embryo DRM2 target loci", 
                               "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci", "long TEs", "Gypsy")) +
  scale_color_manual(values = c("black", "white"),
                     limits = c("Gs", "else")) +
  labs(x = paste("PC1 (", pc_importance[1, 2] %>% signif(3)*100, "% of Variance)", sep = ""), 
       y = paste("PC2 (", pc_importance[2, 2] %>% signif(3)*100, "% of Variance)", "  ", sep = ""),
       fill = NULL,
       color = NULL) +  
  guides(color = F) +
  theme_bw() + 
  theme(legend.position = "right") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave("50k_bin_loci_PCA.svg", width = 8, height = 4.5)
ggsave("50k_bin_loci_PCA.png", width = 8, height = 4.5)
```

# Correlation between PC1 and gene distance 
```{r message=F}
gene_distance_zy_s <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/Shortstack_code_and_output_downsampled/gene_distance_zy_s.csv")

em_meth_data_ss <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/Shortstack_code_and_output_downsampled/em_meth_data_ss.csv")

overlap_data_zygote_TE_s <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/Shortstack_code_and_output_downsampled/overlap_data_zygote_TE_s.csv")

gene_distance_zy_s <- gene_distance_zy_s %>% 
  mutate(ID = case_when(
    str_detect(ID, "egg siRNA loci\n") ~ "E loci - Z loci",
    str_detect(ID, "zygote siRNA loci\n") ~ "Z loci - E loci",
    T ~ ID
  ))


head(gene_distance_zy_s)
```

## gene dist and loci length 
```{r}
gene_dist_PC <- gene_distance_zy_s %>% 
  rbind(
    data.frame(
      ID = "genes",
      NN = 39000,
      mean = 0,
      median = 0,
      SD = 0, 
      Q3 = 0, 
      Q1 = 0, 
      UL = 0,
      LL = 0,
      meanL = 2800,
      medianL = 1514,
      SDL = NA
    )
  ) %>% 
  mutate(ID = case_when(
    ID == "embryo siRNA loci" ~ "*embryo siRNA loci",
    T ~ ID
  )) %>% 
  inner_join(pc$x %>% 
  as.data.frame() %>% 
  mutate(ID = row.names(.)) %>% 
  mutate(ID = case_when(
    ID == "embryo siRNA loci" ~ "*embryo siRNA loci",
    ID == "embryo DRM2 target loci" ~ "embryo DRM2 targets",
    T ~ ID
  )), 
  by = "ID"
  ) 

gene_dist_PC
```
```{r}
cor.test(gene_dist_PC$median, gene_dist_PC$PC1, method = "s")
cor.test(gene_dist_PC$median, gene_dist_PC$PC1, method = "s")$p.value
cor.test(gene_dist_PC$medianL, gene_dist_PC$PC2, method = "s")
cor.test(gene_dist_PC$medianL, gene_dist_PC$PC2, method = "s")$p.value
```
```{r}
legend.source <- get_legend(
  gene_dist_PC %>% 
  ggplot(aes(x = PC1, y = median/1000)) +
  geom_point(aes(fill = ID), shape = 21, alpha = 0.8, size = 3, color = "black") +
  scale_fill_manual(values = c("white", "gold","orange2","blue2", "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2", "slateblue4", "grey80"),
                    limits = c("genes","short TEs", "TIR", "embryo DRM2 targets",
                               "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci", "long TEs","Gypsy")) +
  labs(x = "PC1", 
       y = "median distance to\nnearest genes (kb)",
       fill = NULL,
       color = NULL) +  
  guides(color = F,
         fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1)) +
  theme(legend.position = "top") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(legend.box.margin = margin(-12, 0, 0, 0)) +
  theme(legend.margin = margin(0, 0, 0, 0)) 
) 

dist_PC1 <- gene_dist_PC %>% 
  mutate(type2 = case_when(
    ID == "genes" |
      ID == "Gypsy" ~ "Gs",
    T ~ "else"
  )) %>% 
  ggplot(aes(x = PC1, y = median/1000)) +
  geom_point(aes(fill = ID, color = type2), shape = 21, alpha = 0.8, size = 3) +
  annotate(geom = "text", label = "rho = -0.97\nP = 0",
           size = 3.5, x = Inf, y = Inf, hjust = 1.1, vjust = 1.2,
           fontface = "bold") + 
  scale_fill_manual(values = c("white", "gold","orange2","blue2", "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2", "slateblue4", "grey80"),
                    limits = c("genes","short TEs", "TIR", "embryo DRM2 targets",
                               "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci", "long TEs","Gypsy")) +
  scale_color_manual(values = c("black", "white"),
                     limits = c("Gs", "else")) +
  labs(x = "PC1", 
       y = "median distance to\nnearest genes (kb)",
       fill = NULL,
       color = NULL) +  
  guides(color = F) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1)) +
  theme(legend.position = "none") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

length_PC2 <- gene_dist_PC %>% 
  mutate(type2 = case_when(
    ID == "genes" |
      ID == "Gypsy" ~ "Gs",
    T ~ "else"
  )) %>% 
  ggplot(aes(x = PC2, y = medianL)) +
  geom_point(aes(fill = ID, color = type2), shape = 21, alpha = 0.8, size = 3) +
  annotate(geom = "text", label = "rho = 0.91\nP = 0",
           size = 3.5, x = -Inf, y = Inf, hjust = -0.1, vjust = 1.2,
           fontface = "bold") + 
  scale_fill_manual(values = c("white", "gold","orange2","blue2", "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2", "slateblue4", "grey80"),
                    limits = c("genes","short TEs", "TIR", "embryo DRM2 targets",
                               "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci", "long TEs","Gypsy")) +
  scale_color_manual(values = c("black", "white"),
                     limits = c("Gs", "else")) +
  labs(x = "PC2", 
       y = "median loci length (bp)",
       fill = NULL,
       color = NULL) +  
  guides(color = F) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1)) +
  theme(legend.position = "none") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

lower_part <- plot_grid(dist_PC1, NULL, length_PC2, 
                        nrow = 1, rel_widths = c(1, 0.1, 1))

plot_grid(lower_part, legend.source, nrow = 2, rel_heights = c(1, 0.4))

ggsave("PC1_PC2_dist_length.svg", height = 6, width = 9)
ggsave("PC1_PC2_dist_length.png", height = 6, width = 9)
```

## TE and meth 

```{r}
cor_data <- overlap_data_zygote_TE_s %>% 
  filter(TE == "Gypsy" |
           TE == "TIR" |
           TE == "short TEs" |
           TE == "long TEs") %>% 
  select(ID, mean, TE) %>% 
  spread(TE, mean) %>% 
  inner_join(
    em_meth_data_ss %>% 
  rename(ID = locus) %>% 
  filter(genotype == "WT") %>% 
  select(ID, context, median) %>% 
  spread(context, median),
   by = "ID"
  ) %>% 
  inner_join(
    gene_distance_zy_s %>% 
  select(ID, median), by = "ID"
  ) %>% 
  rename(gene.dist = median) %>% 
  inner_join(pc$x %>% 
  as.data.frame() %>% 
  mutate(ID = row.names(.)) %>% 
    select(ID, PC1),
  by = "ID") %>% 
   mutate(ID = case_when(
    ID == "embryo siRNA loci" ~ "*embryo siRNA loci",
    T ~ ID
  ))


cor_data
```

### cor.test
```{r}
cor.test(cor_data$TIR, cor_data$PC1, method = "s")
cor.test(cor_data$Gypsy, cor_data$PC1, method = "s")

cor.test(cor_data$`long TEs`, cor_data$PC1, method = "s")
cor.test(cor_data$`short TEs`, cor_data$PC1, method = "s")

cor.test(cor_data$mCHH, cor_data$PC1, method = "s")
cor.test(cor_data$mCHG, cor_data$PC1, method = "s")
cor.test(cor_data$mCG, cor_data$PC1, method = "s")
```

```{r}
cor_matrix_data <- cor(cor_data %>% 
                         select(-ID, -PC1), method = "s") %>% 
  as.data.frame() %>% 
  mutate(var1 = row.names(.)) %>% 
  gather("var2", "rho", 1:6) 

cor_heatmap <- cor_matrix_data %>% 
  group_by(var1) %>% 
  mutate(var1 = reorder(var1, rho)) %>% 
  ungroup() %>% 
  group_by(var2) %>% 
  mutate(var2 = reorder(var2, rho)) %>% 
  ungroup() %>%
  mutate(sign = case_when(
    rho > 0 ~ "pos",
    rho < 0 ~ "neg"
  )) %>% 
  ggplot(aes(x = var1, y = var2)) + 
  geom_tile(aes(fill = rho), alpha = 0.9) +
  geom_text(aes(label = signif(rho, 2), color = sign), 
            size = 3.5, fontface = "bold") +
  scale_fill_viridis_c(breaks = c(-1, -0.5, 0, 0.5, 1), 
                       limits = c(-1, 1)) +
  scale_color_manual(values = c("white", "black")) +
  labs(x = NULL,
       y = NULL,
       fill = "rho") +
  guides(color = F) +
  theme_bw() +
  theme(legend.position = "right",
        text = element_text(size= 14, face="bold"),
        axis.text = element_text(colour = "black"),
        axis.text.y = element_text(hjust = 0.5),
        axis.text.x = element_text(angle = 45, hjust = 1),
        panel.grid = element_blank())  

cor_heatmap 

ggsave("cor_heatmap.svg", height = 3.8, width = 4.5)
ggsave("cor_heatmap.png", height = 3.8, width = 4.5)
```

```{r}
cor_matrix_data %>% 
  group_by(var1) %>% 
  mutate(var1 = reorder(var1, rho)) %>% 
  ungroup() %>% 
  group_by(var2) %>% 
  mutate(var2 = reorder(var2, rho)) %>% 
  ungroup() %>%
  mutate(sign = case_when(
    rho > 0 ~ "pos",
    rho < 0 ~ "neg"
  )) %>% 
  ggplot(aes(x = var1, y = var2)) + 
  geom_tile(aes(fill = rho), alpha = 0.9) +
  geom_text(aes(label = signif(rho, 2)), 
            size = 3.5, fontface = "bold", color = "white") +
  scale_fill_gradientn(colors = brewer.pal("RdBu", n = 11) %>% rev(), 
                       breaks = c(-1, -0.5, 0, 0.5, 1), 
                       limits = c(-1, 1)) +
  labs(x = NULL,
       y = NULL,
       fill = "rho") +
  guides(color = F) +
  theme_bw() +
  theme(legend.position = "right",
        text = element_text(size= 14, face="bold"),
        axis.text = element_text(colour = "black"),
        axis.text.y = element_text(hjust = 0.5),
        axis.text.x = element_text(angle = 45, hjust = 1),
        panel.grid = element_blank()) +
  coord_fixed()
```


### scatter plots 
```{r}
TIR_PC1 <- cor_data %>% 
  ggplot(aes(x = PC1, y = TIR)) +
  geom_point(aes(fill = ID), shape = 21, alpha = 0.8, size = 3, color = "white") +
  annotate(geom = "text", label = "rho = 1\nP = 0.0004",
           size = 3.5, x = -Inf, y = Inf, hjust = -0.1, vjust = 1.2,
           fontface = "bold") + 
  scale_fill_manual(values = c( "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c( "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci")) +
  labs(x = "PC1", 
       y = "mean fraction locus length\ncovered by TIR (%)",
       fill = NULL,
       color = NULL) +  
  guides(color = F) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1)) +
  theme(legend.position = "none") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

Gypsy_PC1 <- cor_data %>% 
  ggplot(aes(x = PC1, y = Gypsy)) +
  geom_point(aes(fill = ID), shape = 21, alpha = 0.8, size = 3, color = "white") +
  annotate(geom = "text", label = "rho = -0.96\nP = 0.003",
           size = 3.5, x = Inf, y = Inf, hjust = 1.1, vjust = 1.2,
           fontface = "bold") + 
  scale_fill_manual(values = c( "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c( "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci")) +
  labs(x = "PC1", 
       y = "mean fraction locus length\ncovered by Gypsy (%)",
       fill = NULL,
       color = NULL) +  
  guides(color = F) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1)) +
  theme(legend.position = "none") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

mCHH_PC1 <- cor_data %>% 
  ggplot(aes(x = PC1, y = mCHH)) +
  geom_point(aes(fill = ID), shape = 21, alpha = 0.8, size = 3, color = "white") +
  annotate(geom = "text", label = "rho = 0.96\nP = 0;003",
           size = 3.5, x = -Inf, y = Inf, hjust = -0.1, vjust = 1.2,
           fontface = "bold") + 
  scale_fill_manual(values = c( "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c( "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci")) +
  labs(x = "PC1", 
       y = "median mCHH in WT embryo",
       fill = NULL,
       color = NULL) +  
  guides(color = F) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1)) +
  theme(legend.position = "none") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 


mCHG_PC1 <- cor_data %>% 
  ggplot(aes(x = PC1, y = mCHG)) +
  geom_point(aes(fill = ID), shape = 21, alpha = 0.8, size = 3, color = "white") +
  annotate(geom = "text", label = "rho = -0.89\nP = 0.012",
           size = 3.5, x = Inf, y = Inf, hjust = 1.1, vjust = 1.2,
           fontface = "bold") + 
  scale_fill_manual(values = c( "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c( "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci")) +
  labs(x = "PC1", 
       y = "median mCHG in WT embryo",
       fill = NULL,
       color = NULL) +  
  guides(color = F) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1)) +
  theme(legend.position = "none") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

mCG_PC1 <- cor_data %>% 
  ggplot(aes(x = PC1, y = mCG)) +
  geom_point(aes(fill = ID), shape = 21, alpha = 0.8, size = 3, color = "white") +
  annotate(geom = "text", label = "rho = -0.99\nP = 1e-5",
           size = 3.5, x = Inf, y = Inf, hjust = 1.1, vjust = 1.2,
           fontface = "bold") + 
  scale_fill_manual(values = c( "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c( "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci")) +
  labs(x = "PC1", 
       y = "median mCG in WT embryo",
       fill = NULL,
       color = NULL) +  
  guides(color = F) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1)) +
  theme(legend.position = "none") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

plot_grid(TIR_PC1, NULL, Gypsy_PC1, 
          mCHH_PC1, NULL, mCG_PC1,
          mCHG_PC1, NULL,
          ncol = 3,  
          align = "hv", axis = "lr",
          rel_widths = c(1, 0.1, 1, 
                         1, 0.1, 1,
                         1, 0.1)) 

ggsave("feature_scatter.svg", height = 12, width = 8.5)
ggsave("feature_scatter.png", height = 12, width = 8.5)
```
```{r}
legend.source2 <- get_legend(
  cor_data %>% 
  ggplot(aes(x = PC1, y = TIR)) +
  geom_point(aes(fill = ID), shape = 21, alpha = 0.8, size = 3, color = "white") +
  annotate(geom = "text", label = "rho = -0.96\nP = 0.003",
           size = 3.5, x = -Inf, y = Inf, hjust = -0.1, vjust = 1.2,
           fontface = "bold") + 
  scale_fill_manual(values = c( "seagreen", "grey20",
                                 "violetred4", "pink3", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c( "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "zygote siRNA loci",
                               "Z loci - E loci", 
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci")) +
  labs(x = "PC1", 
       y = "mean fraction locus length\ncovered by TIR (%)",
       fill = NULL,
       color = NULL) +  
  guides(color = F,
         fill = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1)) +
  theme(legend.position = "top") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black"))
)
plot_grid(legend.source2)

ggsave("feature_legend.svg", height = 1, width = 8)
ggsave("feature_legend.png", height = 1, width = 8)
```


# Centromeric regions 
## data 

```{r}
centro_bed <- read_table2("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/rice_centromere_genetic_mapped.bed", col_names = FALSE)

centro_bed <- centro_bed %>% 
  rename(Chr.c = X1, 
         start.c = X2,
         end.c = X3)
head(centro_bed, 12)
```

```{r}
bin50_data_select_s <- bin50_data_select %>% 
  inner_join(beds_total, by = "locus") %>%
  mutate(RPM = count/NN * 10^6) %>% 
  group_by(locus) %>% 
  summarise(mean.RPM = mean(RPM),
            sd.RPM = sd(RPM))

bin50_data_select_s
```
```{r}
bin50_data_select_cent2 <- bin50_data_select %>% 
  merge(centro_bed) %>% 
  filter(Chr == Chr.c) %>% 
  filter( 
    start > start.c & end < end.c 
  ) %>% 
  mutate(tag = paste(Chr, start, end, sep = "-")) 

head(bin50_data_select_cent2)
bin50_data_select_cent2 %>% 
  group_by(Chr) %>% 
  count()
```

```{r}
bin50_data_select_cent <- bin50_data_select %>% 
  mutate(tag = paste(Chr, start, end, sep = "-")) %>% 
  mutate(cent = case_when(
    tag %in% bin50_data_select_cent2$tag ~ "C",
    tag %in% bin50_data_select_cent2$tag == F ~ "NC"
  )) %>% 
  inner_join(beds_total, by = "locus") %>%
  mutate(RPM = count/NN * 10^6) 

head(bin50_data_select_cent)
```
 
```{r}
bin50_data_select_cent %>% 
  group_by(locus) %>% 
  filter(RPM <= quantile(RPM, 0.99)) %>% 
  ungroup() %>% 
  #filter(cent == "C") %>% 
  filter(start > 5749556 & end < 8141816) %>% 
  filter(str_detect(locus, "intersect") == F) %>% 
  filter(locus != "CACTA") %>% 
  mutate(locus = case_when(
    locus == "embryo siRNA loci" ~ "*embryo siRNA loci",
    locus == "embryo DRM2 target loci" ~ "embryo DRM2 targets",
    T ~ locus
  )) %>% 
  mutate(locus = factor(locus, levels = c(
    "genes", "short TEs", "TIR", "long TEs", "Gypsy",
    "sperm siRNA loci", "seedling siRNA loci", "*embryo siRNA loci",
    "embryo DNA2 targets", "Z loci - E loci", "E loci - Z loci"
  ))) %>% 
  filter(is.na(locus) == F) %>% 
  filter(Chr == 8) %>% 
  ggplot(aes(x = start, y = RPM)) +
  facet_grid(locus ~ Chr, scales = "free", space = "free_x", switch = "y") +
  geom_rect(aes(xmin = start, xmax = end, ymax = RPM, fill = locus), ymin = 0) +
  scale_fill_manual(values = c("black", "gold","orange2","blue2", "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2", "slateblue4","grey80"),
                    limits = c("genes", "short TEs", "TIR", "embryo DRM2 targets",
                               "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci", "long TEs", "Gypsy")) +
  labs(x = NULL,
       y = NULL) +
  theme_minimal() +
  theme(legend.position = "none",
        text = element_text(size = 14, color = "black", face = "bold"), 
        axis.text.y = element_blank(),
        strip.placement = "outside",
        strip.text.y.left = element_text(hjust = 0.5, angle = 0),
        strip.text.x = element_blank(),
        axis.text.x = element_blank(),
        panel.grid = element_blank())
```
## stat
```{r}
bin50_data_select_cent_s <- bin50_data_select_cent %>% 
                   filter(cent == "C") %>% 
  filter(str_detect(locus, "intersect") == F) %>%
  group_by(locus) %>% 
  summarise(median = median(RPM))

bin50_data_select_cent_s
```
* Zygote - egg: 116 vs. 158 = -26%
* Z-E vs. E-Z: 126 vs. 176 = -28% 

```{r}
logit <- function(p){log(
  p / (1-p)
)}

logistic <- function(x){
  1/(1 + exp(-x))
}
```

```{r}
model_cent <- lm(log(RPM+1) ~ locus, 
                 data = bin50_data_select_cent %>% 
                   filter(cent == "C") %>% 
  filter(str_detect(locus, "intersect|TIR|Gypsy|CACTA") == F) %>% 
  mutate(locus = case_when(
    locus == "embryo siRNA loci" ~ "*embryo siRNA loci",
    locus == "embryo DRM2 target loci" ~ "embryo DRM2 targets",
    T ~ locus
  )))

#plot(model_cent)
anova(model_cent)
```
```{r}
est_cent <- emmeans(model_cent, pairwise ~ locus)
est_cent$contrasts %>% 
  as.data.frame() %>% 
  select(contrast, p.value)

```

* E-Z vs. Z-E = 3e-13
* egg - zygote = 5e-13 

```{r}
cent_results <- multcomp::cld(est_cent$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(grouping = str_remove_all(.group, " "))

cent_results
```


## plot
```{r}
bin50_data_select_cent %>% 
  filter(cent == "C") %>% 
  filter(str_detect(locus, "intersect|CACTA|TIR|Gypsy") == F) %>% 
  mutate(locus = case_when(
    locus == "embryo siRNA loci" ~ "*embryo siRNA loci",
    locus == "embryo DRM2 target loci" ~ "embryo DRM2 targets",
    T ~ locus
  )) %>% 
  mutate(type2 = case_when(
    locus == "genes" |
      locus == "Gypsy" ~ "Gs",
    T ~ "else"
  )) %>% 
  mutate(locus = fct_reorder(.f = locus, .x = RPM, .fun = median)) %>% 
  ggplot(aes(y = RPM, x = locus)) +
  ggbeeswarm::geom_quasirandom(alpha = 0.8, aes(fill = locus,
                                                color = type2), shape = 21, size = 1) + 
  geom_text(data = cent_results, aes(label = grouping), y = 1000,
            size = 3.5, fontface = "bold") +
  annotate(geom = "errorbarh", height = 50, xmin = 5, xmax = 8, y = 450) +
  annotate(geom = "errorbarh", height = 50, xmin = 7, xmax = 9, y = 550) + 
  annotate(geom = "text", label = "5e-13", y = 450, hjust = -0.1, x = 6, 
           size = 3.5, fontface = "bold") +
  annotate(geom = "text", label = "3e-13", y = 550, hjust = -0.1, x = 8, size = 3.5,
           fontface = "bold") +
  scale_fill_manual(values = c("white", "gold","blue2", "seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2", "slateblue4"),
                    limits = c("genes", "short TEs", "embryo DRM2 targets", "seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                               "sperm siRNA loci", "long TEs")) +
  scale_color_manual(values = c("white", "black"),
                     limits = c("else", "Gs")) + 
  labs(x = NULL,
       y = "number of loci per million loci\nat centromeric regions") + 
  guides(color = F) + 
  theme_minimal() +
  theme(legend.position = "none",
        text = element_text(size = 14, color = "black", face = "bold"),
        axis.text = element_text(color = "black"),
        axis.text.y = element_text(color = "black", hjust = 0.5),
        strip.placement = "outside",
        strip.text.y.left = element_text(hjust = 0.5, angle = 0),
        axis.line = element_line(size = 1)) +
  coord_flip()

ggsave("loci_at_centromere.svg", height = 5, width = 6)
ggsave("loci_at_centromere.png", height = 5, width = 6)
```

# siRNA counts at DRM2 targets
## data 
```{r message=F}
drm2_cov_list <- list.files("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/DRM2_target_siRNA_cov/out/", 
                             pattern = "*.cov", full.names = T)  

drm2_cov_data <- sapply(drm2_cov_list, read_delim, simplify = FALSE, col_names = F, delim = "\t") %>% 
  bind_rows(.id = "id") 

head(drm2_cov_data)
```
```{r message=F}
sample_des <-  read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/sample_des_clvs12.xlsx")  

smRNA_comp <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/codes_and_output/smRNA_comp.csv")

lengths_summary_sam <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/siRNA_length_counts_2021_04_16/lengths_summary.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE) 

lengths_summary_sirenF_sam <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/siRNA_length_counts_2021_04_16/lengths_summary.sirenF.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
## clean up data 
```{r}
drm2_cov_data_clean <- drm2_cov_data %>% 
  mutate(sample_ID = case_when(
    str_detect(id, "eggcell2") ~ "E2",
    str_detect(id, "eggcell3") ~ "E3",
    str_detect(id, "eggcell4") ~ "E4",
    str_detect(id, "egg-9a") ~ "E9a",
    str_detect(id, "egg-9b") ~ "E9b",
    str_detect(id, "egg-9c") ~ "E9c",
    str_detect(id, "zg1d") ~ "zg1d",
    str_detect(id, "zg1e") ~ "zg1e",
    str_detect(id, "zg2d") ~ "zg2d",
    str_detect(id, "zg2e") ~ "zg2e",
    str_detect(id, "zg3a") ~ "zg3a",
    str_detect(id, "zg3b") ~ "zg3b"
  ))

drm2_cov_data_clean %>% 
  group_by(sample_ID) %>% 
  count()
```

```{r}
drm2_cov_data_clean2 <- drm2_cov_data_clean %>% 
  filter(str_detect(id, "sirenF") == F) %>% 
  inner_join(sample_des, by = "sample_ID") %>% 
  inner_join(lengths_summary_sam %>% 
               mutate(sample_ID = case_when(
    str_detect(X1, "eggcell2") ~ "E2",
    str_detect(X1, "eggcell3") ~ "E3",
    str_detect(X1, "eggcell4") ~ "E4",
    str_detect(X1, "egg-9a") ~ "E9a",
    str_detect(X1, "egg-9b") ~ "E9b",
    str_detect(X1, "egg-9c") ~ "E9c",
    str_detect(X1, "zg1d") ~ "zg1d",
    str_detect(X1, "zg1e") ~ "zg1e",
    str_detect(X1, "zg2d") ~ "zg2d",
    str_detect(X1, "zg2e") ~ "zg2e",
    str_detect(X1, "zg3a") ~ "zg3a",
    str_detect(X1, "zg3b") ~ "zg3b"
  )), by = "sample_ID") %>% 
  mutate(RPM = X9/`24` * 10^6  + 1)

head(drm2_cov_data_clean2)
```
```{r}
total_24 <- lengths_summary_sam %>%  
  mutate(sample_ID = case_when(
    str_detect(X1, "eggcell2") ~ "E2",
    str_detect(X1, "eggcell3") ~ "E3",
    str_detect(X1, "eggcell4") ~ "E4",
    str_detect(X1, "egg-9a") ~ "E9a",
    str_detect(X1, "egg-9b") ~ "E9b",
    str_detect(X1, "egg-9c") ~ "E9c",
    str_detect(X1, "zg1d") ~ "zg1d",
    str_detect(X1, "zg1e") ~ "zg1e",
    str_detect(X1, "zg2d") ~ "zg2d",
    str_detect(X1, "zg2e") ~ "zg2e",
    str_detect(X1, "zg3a") ~ "zg3a",
    str_detect(X1, "zg3b") ~ "zg3b"
    )) %>% 
  inner_join(sample_des, by = "sample_ID") %>% 
  group_by(sample_type) %>% 
  summarise(total_24_mer = sum(`24`)) %>% 
  ungroup()
  
drm2_cov_data_clean_s <- drm2_cov_data_clean2 %>% 
  mutate(tag = paste(X1.x, X2, X3, sep = "-")) %>% 
  group_by(sample_type, tag) %>% 
  summarise(total.count = sum(X9)) %>% 
  ungroup() %>% 
  inner_join(total_24, by = "sample_type") %>% 
  mutate(RPM = total.count/total_24_mer * 10^6) %>% 
  mutate(logRPM = log10(RPM))

head(drm2_cov_data_clean_s)
```
```{r}
drm2_cov_data_clean_s_wide <- drm2_cov_data_clean_s %>% 
  select(sample_type, tag, RPM) %>% 
  spread(sample_type, RPM) %>% 
  mutate(FC = zygote/egg) %>% 
  mutate(FC.adjusted = case_when(
    FC == Inf ~ 1024,
    FC == 0 ~ 1/1024,
    T ~FC
  )) %>% 
  mutate(logFC = log2(FC.adjusted))

head(drm2_cov_data_clean_s_wide)
```
## stat
```{r}
wilcox.test(RPM ~ sample_type, data = drm2_cov_data_clean_s, paired = T)
wilcox.test(RPM ~ sample_type, data = drm2_cov_data_clean_s, paired = T)$p.value
```

```{r}
drm2_cov_data_clean_ss <- drm2_cov_data_clean_s %>% 
  group_by(sample_type) %>% 
  summarise(mean = mean(logRPM), 
            median = median(logRPM),
            Q1 = quantile(logRPM, 0.25),
            Q3 = quantile(logRPM, 0.75),
            UL = quantile(logRPM, 0.975),
            LL = quantile(logRPM, 0.025)) %>% 
  ungroup()

drm2_cov_data_clean_ss
```
 
```{r}
drm2_cov_data_test <- drm2_cov_data_clean_s_wide %>% 
              filter(is.na(logFC) ==F)

wilcox.test(x = drm2_cov_data_test$logFC)$p.value
```
 
## plot 
```{r}
drm2_top_stack_bar <- drm2_cov_data_clean_s_wide %>% 
  mutate(sign = case_when(
    logFC > 5 ~ "zygote expressed only",
    logFC < -5 ~ "egg expressed only",
    is.na(logFC) ~ "neither",
    T ~ "shared"
  )) %>% 
  group_by(sign) %>% 
  count() %>% 
  ggplot(aes(x = "x", y = n)) +
  geom_bar(stat = "identity", aes(fill = sign), alpha = 0.8) +
  scale_fill_manual(values = c("tomato1", "grey20", "red2", "violetred4")) +
  scale_y_continuous(trans = "reverse")+
  labs(x = NULL,
       y = "fraction of embryo DRM2 targets",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(
    legend.position = "top",
    text = element_text(size = 14, face = "bold", color = "black"),
    axis.text.y = element_blank(),
    axis.text.x = element_blank(),
    panel.grid.major.y = element_blank(),
    panel.grid.minor.y = element_blank()
  ) +
  coord_flip()

drm2_low_histo <- drm2_cov_data_clean_s_wide %>% 
  filter(is.na(logFC) == F) %>% 
  mutate(sign = case_when(
    logFC > 5 ~ "zygote expressed only",
    logFC < -5 ~ "egg expressed only",
    is.na(logFC) ~ "neither",
    T ~ "shared"
  )) %>% 
  ggplot(aes(x = logFC)) +
  geom_histogram(color = "grey90", bins = 30, aes(fill = sign), alpha = 0.8) + 
  scale_fill_manual(values = c("tomato1", "red2", "violetred4")) +
  labs(x = "log2FC zygote/egg\npositive value higher in zygote") +
  theme_minimal() +
   theme(
    legend.position = "none",
    axis.line = element_line(size = 1),
    text = element_text(size = 14, face = "bold", color = "black"),
    axis.text = element_text(color = "black") 
  )

plot_grid(drm2_top_stack_bar, drm2_low_histo, ncol = 1, rel_heights = c(0.4, 1))

ggsave("zygote_vs_egg_at_DRM2_target.svg", height = 5, width = 4.5)
ggsave("zygote_vs_egg_at_DRM2_target.png", height = 5, width = 4.5)
```

 

 
 


# PCA 50kb for siren loci
```{r}
bin50_data_siren_total <- bin50_data %>% 
  rbind(gene.density %>% 
          mutate(id = "genes")) %>% 
  mutate(locus = case_when(
    str_detect(id, "genes") ~ "genes",
    str_detect(id, "Gypsy") ~ "Gypsy", 
    str_detect(id, "egg_siren") ~ "egg siren loci",
    str_detect(id, "zygote_siren") ~ "zygote siren loci",
    str_detect(id, "fovary_siren") ~ "ovary (9hap) siren loci", 
    str_detect(id, "ovary_siren") &
      str_detect(id, "fovary") == F ~ "ovary (0hr) siren loci",
    str_detect(id, "endo_siren") ~ "endosperm siren loci",
  )) %>% 
  filter(is.na(locus) == F) %>% 
  group_by(locus) %>% 
  summarise(NN = sum(X9))

bin50_data_siren_total
```

```{r}
bin50_data_siren <- bin50_data %>% 
  rbind(gene.density %>% 
          mutate(id = "genes")) %>% 
  rename(Chr = X1,
         start = X4,
         end = X5,
         count = X9,
         bases = X10, 
         coverage = X12) %>%  
  mutate(locus = case_when(
    str_detect(id, "genes") ~ "genes",
    str_detect(id, "Gypsy") ~ "Gypsy", 
    str_detect(id, "egg_siren_loci.bed") ~ "egg siren loci",
    str_detect(id, "zygote_siren") ~ "zygote siren loci",
    str_detect(id, "fovary_siren") ~ "ovary (9hap) siren loci", 
    str_detect(id, "ovary_siren") &
      str_detect(id, "fovary") == F ~ "ovary (0hr) siren loci",
    str_detect(id, "endo_siren") ~ "endosperm siren loci",
  )) %>% 
  filter(is.na(locus) == F) %>% 
  inner_join(bin50_data_siren_total, by = "locus") %>% 
  mutate(RPM = count/NN * 10^6)

head(bin50_data_siren)

bin50_data_siren %>% 
  group_by(locus) %>% 
  count()
```
```{r}
bin50_data_siren_wide <- bin50_data_siren %>% 
  select(Chr, start, end, locus, RPM) %>% 
  spread(locus, RPM)

head(bin50_data_siren_wide)
```
```{r}
pc2 <- prcomp(t(bin50_data_siren_wide[, 4:9]))
pc2_importance <- as.data.frame(t(summary(pc2)$importance))
pc2_importance 
```

```{r}
pc2$x %>% 
  as.data.frame() %>% 
  mutate(locus_type = row.names(.)) %>% 
  mutate(type2 = case_when(
    locus_type == "genes" ~ "genes",
    T ~ "else"
  )) %>% 
  ggplot(aes(x = PC1, y = PC2)) +
  geom_point(aes(fill = locus_type, color = type2), size = 3, alpha = 0.8, shape = 21) +
  geom_text_repel(aes(label = locus_type), size = 4) +
  scale_fill_manual(values = c("white",  "tomato1", "lightgoldenrod4", "orangered3", "coral4", "violetred4", "grey80"),
                    limits = c("genes",  "egg siren loci", "endosperm siren loci",
                               "ovary (0hr) siren loci", "ovary (9hap) siren loci", "zygote siren loci", "Gypsy")) +
  scale_color_manual(values = c("black", "white"),
                     limits = c("genes", "else")) +
  labs(x = paste("PC1 (", pc2_importance[1, 2] %>% signif(3)*100, "% of Variance)", sep = ""), 
       y = paste("PC2 (", pc2_importance[2, 2] %>% signif(3)*100, "% of Variance)", "  ", sep = ""),
       fill = NULL,
       color = NULL) +  
  guides(color = F) +
  theme_bw() + 
  theme(legend.position = "none") +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", hjust = 0.8)) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave("50k_bin_loci_PCA_siren.svg", width = 5, height = 5)
ggsave("50k_bin_loci_PCA_siren.png", width = 5, height = 5)
```

 
