---
title: "Gene_distance_TE_cov"
author: "Chenxin Li"
date: "10/7/2020"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(RColorBrewer)
library(viridis)
library(stringr)
library(svglite)
library(cowplot)
library(forcats)
library(patchwork)
```

# gene distance
## load data
```{r, warning = FALSE, message = F}
gene.dist.list <- list.files("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/FullSetLoci3_closest/", 
                             pattern = "*gene.txt", full.names = T)

gene_distance  <- sapply(gene.dist.list, read_delim, simplify = FALSE, col_names = F, delim = "\t") %>% 
  bind_rows(.id = "id")

head(gene_distance)
```
```{r message=F}
CHH_dist <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/closest_genes_2021_04_16/CHH_wt-52_dmr100_hypo_cutoffs_5-0.001-0.25-0.3_sorted.bed.closest_gene.txt", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 

head(CHH_dist) #X1, X2, X3, X18 
```
```{r message=F}
TIR_dist <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/closest_genes_2021_04_16/Oryza_sativa.IRGSP-1.non-CACTA_DNA.out.gff.closest_gene.txt", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

head(TIR_dist) #X1, X4, X5, X19 
```

```{r message=F}
Gypsy_dist <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/closest_genes_2021_04_16/Oryza_sativa.IRGSP-1.Gypsy.out.gff.closest_gene.txt", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

head(Gypsy_dist) #X1, X4, X5, X19 
```
```{r message=F}
CACTA_DNA_TE_closest_gene <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_zygote_GR_revision/short_long_TE_gene_distance/CACTA_DNA_TE_closest_gene.xlsx", 
    col_names = FALSE)

shortTE_dist <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_zygote_GR_revision/short_long_TE_gene_distance/Oryza_sativa.IRGSP-1.transposons.50-250.out.gff.closest_gene.txt", col_names = F, delim = "\t")

longTE_dist <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_zygote_GR_revision/short_long_TE_gene_distance/Oryza_sativa.IRGSP-1.transposons.251.out.gff.closest_gene.txt", col_names = F, delim = "\t")
```

```{r}
colnames(CACTA_DNA_TE_closest_gene) <- colnames(shortTE_dist)

shortTE_dist %>% 
  select(1,4,5) %>% 
  write_delim("shortTE.bed", col_names = F, delim = "\t")

longTE_dist %>% 
  select(1,4,5) %>% 
  write_delim("longTE.bed", col_names = F, delim = "\t")
```

## composition and lengths of TE classifications 
### composition
 

### lengths 
```{r}
TE_dist <- rbind(
  TIR_dist %>% 
    mutate(class = "TIR"),
  Gypsy_dist %>% 
    mutate(class = "Gypsy"),
  shortTE_dist %>% 
    mutate(class = "short TEs"),
  longTE_dist %>% 
    mutate(class = "long TEs")
) %>% 
  mutate(length = X5 - X4)  

head(TE_dist)
```

```{r}
TE_dist %>%
  rbind(CACTA_DNA_TE_closest_gene %>% 
          mutate(class = "CACTA") %>% 
          mutate(length = X5 - X4)) %>% 
  group_by(class) %>% 
  summarise(
    NN = n()
  )

summary(shortTE_dist$X5 - shortTE_dist$X4)
summary(longTE_dist$X5 - longTE_dist$X4)
summary(CACTA_DNA_TE_closest_gene$X5 - CACTA_DNA_TE_closest_gene$X4)
```


```{r}
TE_dist %>% 
  mutate(class = factor(class, levels = c("TIR", "Gypsy", "short TEs", "long TEs"))) %>% 
  ggplot(aes(x = class, y = log10(length))) +
  geom_boxplot(aes(fill = class), alpha = 0.8, width= 0.5) +
  scale_fill_manual(values = c("orange2", "gold", "slateblue4", "grey80"),
                    limits = c("TIR", "short TEs", "long TEs", "Gypsy")) +
  labs(y = "log10(bp)",
       x = "TE superfamility") + 
  theme_classic() +
  theme(legend.position = "none",
        text = element_text(size = 14, color = "black", face = "bold"),
        axis.text = element_text(color = "black"),
        axis.text.y.left = element_text(hjust = 0.5)
        ) +
  coord_flip()

ggsave("TE_length.svg", height = 2.5, width = 5)
ggsave("TE_length.png", height = 2.5, width = 5)
```

```{r}
TE_dist %>% 
  rbind(CACTA_DNA_TE_closest_gene %>% 
          mutate(class = "CACTA") %>% 
          mutate(length = X5 - X4)) %>% 
  mutate(class = factor(class, levels = 
                          c("CACTA", "TIR", "Gypsy", "short TEs", "long TEs"))) %>% 
  ggplot(aes(x = class, y = log10(length))) +
  geom_boxplot(aes(fill = class), alpha = 0.8, width= 0.5) +
  scale_fill_manual(values = c("indianred3", "orange2", "gold", "slateblue4", "grey80"),
                    limits = c("CACTA", "TIR", "short TEs", "long TEs", "Gypsy")) +
  labs(y = "log10(bp)",
       x = "TE superfamility") + 
  theme_classic() +
  theme(legend.position = "none",
        text = element_text(size = 14, color = "black", face = "bold"),
        axis.text = element_text(color = "black"),
        axis.text.y.left = element_text(hjust = 0.5)
        ) +
  coord_flip()

ggsave("TE_length_CACTA.svg", height = 2.8, width = 5)
ggsave("TE_length_CACTA.png", height = 2.8, width = 5)
```

## egg, sperm, zygote, zygote de novo, seedling, embryo 
```{r}
gene_distance_zygote <- gene_distance %>% 
  mutate(ID = case_when(
    str_detect(id, "zygote-egg_com") ~ "zygote siRNA loci\n(minus egg siRNA loci)", #12
    str_detect(id, "egg-zygote_com") ~ "egg siRNA loci\n(minus zygote siRNA loci)", #3
    str_detect(id, "egg_loci.bed") ~ "egg siRNA loci", #4
    str_detect(id, "embryo_loci") ~ "embryo siRNA loci", #5
    str_detect(id, "seedling_loci.") ~ "seedling siRNA loci", #6
    str_detect(id, "zygote-egg_intersect") ~ "zygote/egg\nloci intersect",  #7
    str_detect(id, "sperm_loci") &
      str_detect(id, "endo") == F ~ "sperm siRNA loci", #9
    str_detect(id, "zygote_loci.bed") ~ "zygote siRNA loci", #10
    str_detect(id, "fovary_loci") ~ "ovary (9hap) siRNA loci", 
    str_detect(id, "ovary_loci") ~ "ovary (0hr) siRNA loci",
  )) %>% 
  filter(is.na(ID) == F) %>% 
  select(X1, X2, X3, X13, ID)

gene_distance_zygote2 <- gene_distance_zygote %>% 
  select(X1, X2, X3, X13, ID) %>% 
  rbind(
    CHH_dist %>% 
      select(X1, X2, X3, X18) %>% 
      rename(X13 = X18) %>% 
      mutate(ID = "embryo DRM2 targets")
  ) %>% 
  rbind(
    TE_dist %>% 
      mutate(ID = class) %>% 
      select(X1, X4, X5, X19, ID) %>% 
      rename(X2 = X4, 
             X3 = X5,
             X13 = X19)  
  ) %>% 
  rbind(
    CACTA_DNA_TE_closest_gene %>% 
      mutate(ID = "CACTA") %>% 
      select(X1, X4, X5, X19, ID) %>% 
      rename(X2 = X4, 
             X3 = X5,
             X13 = X19) 
  )

head(gene_distance_zygote2)
```

### summary stats

```{r}
#median length of genes 
Gypsy_dist %>% 
  mutate(Length = X14- X13) %>% 
  summarise(
    medianL = median(Length),
    meanL = mean(Length)
  )
```

```{r}
gene_distance_zy_s <- gene_distance_zygote2 %>% 
  rename(distance = X13) %>% 
  filter(distance >= 0) %>% 
  mutate(Length = X3 - X2) %>% 
  group_by(ID) %>% 
   summarise( 
            NN = n(),
            mean = mean(distance),
            median = median(distance),
            SD = sd(distance), 
            Q3 = quantile(distance, 0.75),
            Q1 = quantile(distance, 0.25),
            UL = quantile(distance, 0.975),
            LL = quantile(distance, 0.025),
            meanL = mean(Length),
            medianL = median(Length),
            SDL = sd(Length)) %>% 
  ungroup() %>% 
  arrange(median)

gene_distance_zy_s 

write_excel_csv(gene_distance_zy_s, "gene_distance_zy_s.csv")
```

* zygote NOT egg: 1635
* egg NOT zygote: 2845

* zygote: 1625
* egg: 2394

### linear model 
```{r}
gene_distance_zygote2 %>% 
                         rename(distance = X13) %>% 
  mutate(distance = ceiling(distance))
```

```{r}
model_zygote_dist <- glm(distance ~ ID, gene_distance_zygote2 %>% 
                         rename(distance = X13) %>% 
                         filter(distance >= 0) %>% 
                          mutate(distance = ceiling(distance)) %>% 
                          filter(str_detect(ID, "ovary") == F) %>% 
                          filter(str_detect(ID, "inter")==F) %>% 
                          filter(str_detect(ID, "long|short|CACTA") == F) %>% 
                          group_by(ID) %>% 
                          sample_n(2000) %>% 
                          ungroup(),
                         family = poisson()
                        )

anova(model_zygote_dist, test = "LRT")
```
 

```{r}
est_gene_dist_zy <- emmeans(model_zygote_dist, pairwise ~ ID)
gene_dist_zy_results <- multcomp::cld(est_gene_dist_zy$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(grouping = str_remove_all(.group, " ")) %>% 
  select(ID, grouping, emmean) %>% 
  mutate(mean = exp(emmean)) 

gene_dist_zy_results
```

```{r}
est_gene_dist_zy$contrasts %>% as.data.frame() %>% 
  select(contrast, p.value, z.ratio)
```

* egg siRNA loci - zygote siRNA loci	0	
* egg NOT zygote - zygote NOT egg 0
 


### plot 
```{r}
gene_distance_zy_s %>% 
  #inner_join(gene_dist_zy_results, by = "ID") %>% 
  filter(ID == "TIR" |
           ID == "Gypsy" |
           ID == "CACTA") %>% 
  mutate(ID = reorder(ID, median)) %>% 
  mutate(text.pos = case_when(
    UL < 20000 ~ UL,
    T ~ Q3
  )) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("n = ", NN, ", median = ", signif(median/1000, 2), "-kb", sep = ""), 
                x = ID, y = text.pos/1000),
                vjust = -0.2,  hjust = -0.1, size = 4, fontface = "bold") + 
  # geom_text(aes(label = grouping, y = Q3/1000), size = 4, hjust = -0.2, vjust = -0.2,
  #           fontface = "bold") +
  # annotate(geom = "text", x = 1.5, y= 35, 
  #          label = "NS", size = 4, fontface = "bold", hjust = -0.2) +
  # annotate(geom = "errorbarh", xmin = 1.3, xmax = 1.7, y = 35, 
  #          height = 0.3, size = 1) +
  scale_fill_manual(values = c("orange2", "grey80", "indianred3"),
                    limits = c("TIR", "Gypsy", "CACTA"),
  ) +
  #scale_y_continuous(breaks = c(0, 20, 40), limits = c(0, 40)) +
  labs(x = NULL,
       y = "kilobases to nearest genes",
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 14, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("CACTA_TIR_Gypsy_gene_dist.svg", height = 2, width = 5)
ggsave("CACTA_TIR_Gypsy_gene_dist.png", height = 2, width = 5)
```

#### 0
```{r}
gene_distance_zy_s %>% 
  #inner_join(gene_dist_zy_results, by = "ID") %>% 
  filter(ID == "TIR" |
           ID == "Gypsy") %>% 
  mutate(ID = reorder(ID, median)) %>% 
  mutate(text.pos = case_when(
    UL < 20000 ~ UL,
    T ~ Q3
  )) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("n = ", NN, ", median = ", signif(median/1000, 2), "-kb", sep = ""), 
                x = ID, y = text.pos/1000),
                vjust = -0.2,  hjust = -0.1, size = 4, fontface = "bold") + 
  # geom_text(aes(label = grouping, y = Q3/1000), size = 4, hjust = -0.2, vjust = -0.2,
  #           fontface = "bold") +
  # annotate(geom = "text", x = 1.5, y= 35, 
  #          label = "NS", size = 4, fontface = "bold", hjust = -0.2) +
  # annotate(geom = "errorbarh", xmin = 1.3, xmax = 1.7, y = 35, 
  #          height = 0.3, size = 1) +
  scale_fill_manual(values = c("orange2", "grey80"),
                    limits = c("TIR", "Gypsy"),
  ) +
  #scale_y_continuous(breaks = c(0, 20, 40), limits = c(0, 40)) +
  labs(x = NULL,
       y = "kilobases to nearest genes",
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 14, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("TIR_Gypsy_gene_dist.svg", height = 1.5, width = 5)
ggsave("TIR_Gypsy_gene_dist.png", height = 1.5, width = 5)
```
```{r}
gene_distance_zy_s %>% 
  #inner_join(gene_dist_zy_results, by = "ID") %>% 
  filter(ID == "short TEs" |
           ID == "long TEs") %>% 
  mutate(ID = reorder(ID, median)) %>% 
  mutate(text.pos = case_when(
    UL < 20000 ~ Q3,
    T ~ Q3
  )) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("n = ", NN, ", median = ", signif(median/1000, 2), "-kb", sep = ""), 
                x = ID, y = text.pos/1000),
                vjust = -0.2,  hjust = -0.01, size = 4, fontface = "bold") + 
  # geom_text(aes(label = grouping, y = Q3/1000), size = 4, hjust = -0.2, vjust = -0.2,
  #           fontface = "bold") +
  # annotate(geom = "text", x = 1.5, y= 35, 
  #          label = "NS", size = 4, fontface = "bold", hjust = -0.2) +
  # annotate(geom = "errorbarh", xmin = 1.3, xmax = 1.7, y = 35, 
  #          height = 0.3, size = 1) +
  scale_fill_manual(values = c("gold", "slateblue4"),
                    limits = c("short TEs", "long TEs"),
  ) +
  #scale_y_continuous(breaks = c(0, 20, 40), limits = c(0, 40)) +
  labs(x = NULL,
       y = "kilobases to nearest genes",
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 14, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("shortlong_gene_dist.svg", height = 1.5, width = 5)
ggsave("shortlong_gene_dist.png", height = 1.5, width = 5)
```


#### 1
```{r}
group1 <- gene_distance_zy_s %>% 
  #inner_join(gene_dist_zy_results, by = "ID") %>% 
  filter(ID == "ovary (0hr) siRNA loci" |
           ID == "ovary (9hap) siRNA loci") %>% 
  mutate(ID = reorder(ID, median)) %>% 
  mutate(text.pos = case_when(
    UL < 20000 ~ Q3,
    T ~ Q3
  )) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("n = ", NN, ", median = ", signif(median/1000, 2), "-kb", sep = ""), 
                x = ID, y = text.pos/1000),
                vjust = -0.2,  hjust = -0.1, size = 4, fontface = "bold") + 
  # geom_text(aes(label = grouping, y = Q3/1000), size = 4, hjust = -0.2, vjust = -0.2,
  #           fontface = "bold") +
  # annotate(geom = "text", x = 1.5, y= 35, 
  #          label = "NS", size = 4, fontface = "bold", hjust = -0.2) +
  # annotate(geom = "errorbarh", xmin = 1.3, xmax = 1.7, y = 35, 
  #          height = 0.3, size = 1) +
  scale_fill_manual(values = c("orangered3", "coral4"),
                    limits = c("ovary (0hr) siRNA loci", "ovary (9hap) siRNA loci"),
  ) +
 # scale_y_continuous(breaks = c(0, 20, 40), limits = c(0, 40)) +
  labs(x = NULL,
       y = "kilobases to nearest genes",
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 14, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

group1
```


#### 2 
```{r}
group2 <- gene_distance_zy_s %>% 
  inner_join(gene_dist_zy_results, by = "ID") %>%  
  filter(ID == "egg siRNA loci" |
           ID == "zygote siRNA loci") %>% 
  mutate(ID = reorder(ID, median)) %>% 
  mutate(text.pos = case_when(
    UL < 20000 ~ Q3 + 1500,
    T ~ Q3 + 1500
  )) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("n = ", NN, ", median = ", signif(median/1000, 2), "-kb", sep = ""), 
                x = ID, y = text.pos/1000),
                vjust = -0.2,  hjust = 0, size = 4, fontface = "bold") +  
  geom_text(aes(label = grouping, y = Q3/1000), size = 4, hjust = -0.2, vjust = -0.2,
            fontface = "bold") +
  annotate(geom = "text", x = 1.5, y= 20, 
           label = "P < 2.2e-16", size = 4, fontface = "bold", hjust = -0.05) +
  annotate(geom = "errorbarh", xmin = 1.3, xmax = 1.7, y = 20, 
           height = 0.3, size = 1) +
  scale_fill_manual(values = c("tomato1", "violetred4"),
                    limits = c("egg siRNA loci", "zygote siRNA loci"),
  ) + 
  #scale_y_continuous(breaks = c(0, 20, 40), limits = c(0, 40)) +
  labs(x = NULL,
       y = NULL,
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 14, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

group2
```

#### 3
```{r}
group3 <- gene_distance_zy_s %>% 
  inner_join(gene_dist_zy_results, by = "ID") %>%  
  filter(str_detect(ID, "minus")) %>% 
  mutate(ID = reorder(ID, median)) %>% 
  mutate(text.pos = case_when(
    UL > 30000 ~ Q3 + 1500,
    T ~ Q3 + 1500
  )) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("n = ", NN, ", median = ", signif(median/1000, 2), "-kb", sep = ""), 
                x = ID, y = text.pos/1000),
                vjust = -0.2,  hjust = 0, size = 4, fontface = "bold") + 
  geom_text(aes(label = grouping, y = Q3/1000), size = 4, hjust = -0.2, vjust = -0.2,
            fontface = "bold") +
  annotate(geom = "text", x = 1.5, y= 25, 
           label = "P < 2.2e-16", size = 4, fontface = "bold", hjust = -0.05) +
  annotate(geom = "errorbarh", xmin = 1.3, xmax = 1.7, y = 25, 
           height = 0.3, size = 1) +
  scale_fill_manual(values = c("violetred4", "tomato1")
  ) +
  scale_x_discrete(labels = c("Z loci - E loci", "E loci - Z loci")) +
  labs(x = NULL,
       y = NULL,
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 14, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

group3
```




#### 5 
```{r}
group5 <- gene_distance_zy_s %>% 
  inner_join(gene_dist_zy_results, by = "ID") %>%  
  filter(ID == "seedling siRNA loci" |
           ID == "sperm siRNA loci" |
           ID == "embryo siRNA loci") %>% 
  mutate(ID2 = case_when(
    str_detect(ID, "emb") ~ "*embryo siRNA loci",
    T ~ ID
  )) %>% 
  mutate(ID = reorder(ID2, median)) %>% 
  mutate(text.pos = case_when(
    UL < 25000 ~ UL + 1000,
    T ~ Q3 + 3000
  )) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("n = ", NN, ", median = ", signif(median/1000, 2), "-kb", sep = ""), 
                x = ID, y = text.pos/1000),
                vjust = -0.2,  hjust = 0, size = 4, fontface = "bold") +  
  geom_text(aes(label = grouping, y = Q3/1000), size = 4, hjust = -0.2, vjust = -0.2,
            fontface = "bold") + 
  # annotate(geom = "text", x = 1.5, y= 32, 
  #          label = "NS", size = 4, fontface = "bold", hjust = -0.2) +
  # annotate(geom = "errorbarh", xmin = 1.2, xmax = 1.8, y = 32, 
  #          height = 0.3, size = 1) + 
  scale_fill_manual(values = c("grey20", "seagreen", "dodgerblue2"),
                    limits = c("*embryo siRNA loci", "seedling siRNA loci", "sperm siRNA loci"),
  ) +
  #scale_y_continuous(breaks = c(0, 20, 40), limits = c(0, 40)) +
  labs(x = NULL,
       y = NULL,
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 14, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

group5
```
####  6
```{r}
group6 <- gene_distance_zy_s %>% 
  inner_join(gene_dist_zy_results, by = "ID") %>%  
  filter(ID == "TIR" |
           ID == "Gypsy" |
           ID == "embryo DRM2 targets") %>% 
  mutate(ID = reorder(ID, median)) %>% 
  mutate(text.pos = case_when(
    UL < 25000 ~ UL + 1000,
    T ~ Q3 + 3000
  )) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("n = ", NN, ", median = ", signif(median/1000, 2), "-kb", sep = ""), 
                x = ID, y = text.pos/1000),
                vjust = -0.2,  hjust = 0, size = 4, fontface = "bold") +  
  geom_text(aes(label = grouping, y = Q3/1000), size = 4, hjust = -0.2, vjust = -0.2,
            fontface = "bold") + 
  # annotate(geom = "text", x = 1.5, y= 32, 
  #          label = "NS", size = 4, fontface = "bold", hjust = -0.2) +
  # annotate(geom = "errorbarh", xmin = 1.2, xmax = 1.8, y = 32, 
  #          height = 0.3, size = 1) + 
  scale_fill_manual(values = c("orange2", "blue2", "grey80"),
                    limits = c("TIR", "embryo DRM2 targets", "Gypsy"),
  ) +
  #scale_x_discrete(label = c("short TE", "embryo DRM2 targets", "long TE")) +
  labs(x = NULL,
       y = "kilobases to nearest genes",
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 14, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

group6
```

#### put them together 
```{r}
plot_grid(group2, group3, group5, group6, 
          rel_heights = c(1, 1, 1.3 ,1.5),
          ncol = 1,
          axis = "lr",
          align = "v")

ggsave("distance_near_gene_zygote.svg", width = 7, height = 5)
ggsave("distance_near_gene_zygote.png", width = 7, height = 5) 
```
 

# TE overlaps 
## load data
```{r, warning=F, message=F}
overlap_files_seedling <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/FullSetLoci3_cov/", 
                            pattern = "*.cov", full.names = T)

overlap_files_seedling <- overlap_files_seedling[str_detect(overlap_files_seedling, "50KB") == F]
overlap_files_seedling <- overlap_files_seedling[str_detect(overlap_files_seedling,
                                                            "TIR|Gypsy|genes|seedling_loci.cov")]

overlap_data_seedling <- sapply(overlap_files_seedling, read_delim, simplify = FALSE, 
                       col_names = F, trim_ws = T, escape_double = FALSE, delim = "\t") %>% 
  bind_rows(.id = "id") 

head(overlap_data_seedling)
```

```{r, warning= F, message = F} 
overlap_files <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_zygote_GR_revision/loci_TE_coverages_2021_09_22/", 
                            pattern = "*out.cov", full.names = T)

overlap_data <- sapply(overlap_files, read_delim, simplify = FALSE, 
                       col_names = F, trim_ws = T, escape_double = FALSE, delim = "\t") %>% 
  bind_rows(.id = "id") 

head(overlap_data)
```
```{r message=F}
overlap_files_drm2 <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/mCHH_DRM2_DMR_coverages/", 
                            pattern = "*.cov", full.names = T)

overlap_data_drm2 <- sapply(overlap_files_drm2, read_delim, simplify = FALSE, 
                       col_names = F, trim_ws = T, escape_double = FALSE, delim = "\t") %>% 
  bind_rows(.id = "id") 

head(overlap_data_drm2)
```

## Egg, seedling, sperm, zygote
```{r}
overlap_data_zyogte <- overlap_data %>% 
  rbind(overlap_data_drm2, overlap_data_seedling) %>% 
  mutate(ID = case_when(
    str_detect(id, "zygote-egg_com") ~ "Z loci - E loci", #12
    str_detect(id, "egg-zygote_com") ~ "E loci - Z loci", #3
    str_detect(id, "egg_loci.bed") ~ "egg siRNA loci", #4
    str_detect(id, "embryo_loci") ~ "embryo siRNA loci", #5
    str_detect(id, "seedling_loci.bed") ~ "seedling siRNA loci", #6
    str_detect(id, "zygote-egg_intersect") ~ "Z/E loci intersect",  #7
    str_detect(id, "sperm_loci") &
      str_detect(id, "endo") == F ~ "sperm siRNA loci", #9
    str_detect(id, "zygote_loci.bed") ~ "zygote siRNA loci", #10
  )) %>% 
  filter(is.na(ID) == F) %>% 
   mutate(TE = case_when(
   str_detect(id, "longTEs.out") ~ "long TEs",     #1
   str_detect(id, "shortTEs.out") ~ "short TEs",  #2
   str_detect(id, "CACTA") ~ "CACTA",  #3
   str_detect(id, "seedling_loci.cov") ~ "seedling siRNA loci", #4
   str_detect(id, "DRM2") ~ "embryo DRM2 targets", #5
   str_detect(id, "TIR") ~ "TIR",     #6
   str_detect(id, "Gypsy") ~ "Gypsy",  #7
   str_detect(id, "genes") ~ "genes"   #8
  )) %>% 
  filter(is.na(TE) == F)  

head(overlap_data_zyogte)
```
 
 

### summary stats

```{r}
overlap_data_zygote_TE_s <- overlap_data_zyogte %>%
  group_by(ID, TE) %>%
  summarise(
    NN = n(),
    mean = mean(X7 * 100),
    median = median(X7)
  ) %>% 
  ungroup()  

write_excel_csv(overlap_data_zygote_TE_s, "overlap_data_zygote_TE_s.csv")

head(overlap_data_zygote_TE_s)
```
```{r}
overlap_data_zygote_TE_s %>% 
  group_by(TE) %>% 
  count()
```
 

### linear model 
```{r}
logit <- function(p){
  log(
    p / (1 - p)
  )
}

logistic <- function(x){
  1/(
    1 + exp(-x)
  )
}
```

#### Long and short TEs 
```{r}
model_zygote_cov_logit_longshort <- glm(X7 ~ ID*TE, data = overlap_data_zyogte %>% 
                                filter(TE == "long TEs" |
                                         TE== "short TEs"|
                                         TE == "embryo DRM2 targets") %>% 
                         filter(ID == "seedling siRNA loci" |
                                 ID == "egg siRNA loci" |
                                 ID == "zygote siRNA loci" |
                                 ID == "sperm siRNA loci" |
                                 ID == "embryo siRNA loci" |
                                  str_detect(ID, "-")) %>% 
                        mutate(ID = case_when(
                                   str_detect(ID, "embryo siRNA loci") ~ "*embryo siRNA loci",
                                   T ~ ID
                               )) %>% 
                          group_by(ID) %>% 
                          sample_n(10000),
                       family = quasibinomial(link = "logit"))

anova(model_zygote_cov_logit_longshort, test = "LRT")
```
```{r}
est_z_cov_longshort <- emmeans(model_zygote_cov_logit_longshort, pairwise ~ ID | TE)
est_z_cov_longshort$contrasts %>% 
  as.data.frame() %>% 
  select(contrast, TE, p.value)
```
```{r}
z_cov_lm_results_longshort <- multcomp::cld(est_z_cov_longshort$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(mean_cov = logistic(emmean)) %>% 
  mutate(group = str_remove_all(.group, " ")) %>% 
  mutate(group = case_when(
    TE == "short TEs" ~ group, 
    T ~ toupper(group)
  ))
  
z_cov_lm_results_longshort %>% 
  select(ID, TE, group)
```

#### TIR and Gypsy TEs 
```{r}
model_zygote_cov_logit <- glm(X7 ~ ID*TE, data = overlap_data_zyogte %>% 
                                filter(TE == "TIR" |
                                         TE== "Gypsy"|
                                         TE == "embryo DRM2 targets") %>% 
                         filter(ID == "seedling siRNA loci" |
                                 ID == "egg siRNA loci" |
                                 ID == "zygote siRNA loci" |
                                 ID == "sperm siRNA loci" |
                                 ID == "embryo siRNA loci" |
                                  str_detect(ID, "-")) %>% 
                        mutate(ID = case_when(
                                   str_detect(ID, "embryo siRNA loci") ~ "*embryo siRNA loci",
                                   T ~ ID
                               )) %>% 
                                 group_by(ID) %>% 
                                 sample_n(10000),
                       family = quasibinomial(link = "logit"))

anova(model_zygote_cov_logit, test = "LRT")
```


```{r}
est_z_cov <- emmeans(model_zygote_cov_logit, pairwise ~ ID | TE)
est_z_cov$contrasts %>% 
  as.data.frame() %>% 
  select(contrast, TE, p.value)
```

```{r}
z_cov_lm_results <- multcomp::cld(est_z_cov$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(mean_cov = logistic(emmean)) %>% 
  mutate(group = str_remove_all(.group, " ")) %>% 
  mutate(group = case_when(
    TE == "TIR" ~ group, 
    T ~ toupper(group)
  ))
  
z_cov_lm_results %>% 
  select(ID, TE, group)
```

#### seedling siRNA loci coverage 
This have to be done separately, because seedling with seedling is 1 obviously
```{r}
model_zygote_seedling_cov_logit <- glm(X7 ~ ID, data = overlap_data_zyogte %>% 
                         filter(TE == "seedling siRNA loci" ) %>% 
                         filter(ID == "egg siRNA loci" |
                                 ID == "zygote siRNA loci" |
                                 ID == "sperm siRNA loci" |
                                 ID == "embryo siRNA loci"|
                                  str_detect(ID, "-")) %>% 
                        mutate(ID = case_when(
                                   str_detect(ID, "embryo") ~ "*embryo siRNA loci",
                                   T ~ ID
                               )) %>% 
                          group_by(ID) %>% 
                          sample_n(10000),
                       family = quasibinomial(link = "logit"))

anova(model_zygote_seedling_cov_logit, test = "LRT")
```
```{r}
est_z_sd_cov <- emmeans(model_zygote_seedling_cov_logit, pairwise ~ ID)
est_z_sd_cov$contrasts %>% 
  as.data.frame() %>% 
  select(contrast, z.ratio, p.value)
```
```{r}
z_sd_cov_lm_results <- multcomp::cld(est_z_sd_cov$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(mean_cov = logistic(emmean)) %>% 
  mutate(group = str_remove_all(.group, " ")) 

z_sd_cov_lm_results
```


### plot 
```{r}
overlap_data_zygote_TE_s %>% 
  filter(TE == "TIR" | 
        TE == "Gypsy") %>% 
  filter(ID == "seedling siRNA loci" |
           ID == "egg siRNA loci" |
           ID == "zygote siRNA loci" |
           ID == "sperm siRNA loci" |
           ID == "embryo siRNA loci" |
           str_detect(ID, "-")) %>% 
  mutate(ID = case_when(
    str_detect(ID, "embryo") ~ "*embryo siRNA loci",
    T ~ ID
  )) %>% 
  mutate(ID = factor(ID, levels = c(
    "sperm siRNA loci",
    "E loci - Z loci",
    "egg siRNA loci",
    "zygote siRNA loci",
    "Z loci - E loci",
    "*embryo siRNA loci",
    "seedling siRNA loci"
  ))) %>%
  ggplot(aes(x = TE, y = mean)) +
  facet_grid(ID ~ ., switch = "y") +
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  geom_text(aes(label = signif(mean, 3)), size = 3.5, fontface = "bold", hjust = 0, y = 0) +
  geom_text(data = z_cov_lm_results %>% 
              filter(TE == "TIR" |
                       TE == "Gypsy"), 
            aes(label = group, y = mean_cov * 100, hjust = -0.3),
            size = 3.5, fontface = "bold") +
  scale_fill_manual(values = c("orange2", "grey70"),
                 limits = c("TIR", "Gypsy")) +
  labs(x = NULL,
       y = "mean fraction of locus length\noverlapped by TEs (%)",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 1)) + 
  guides(color = F) +
  theme_minimal() +
  theme(legend.position = "top") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_blank()) +
  theme(strip.placement = "outside") + 
  theme(strip.text.y.left = element_text(angle = 0)) +
  coord_flip()

ggsave("TE_overlap_lite.svg", width = 6, height = 4)
ggsave("TE_overlap_lite.png", width = 6, height = 4)
```

```{r}
overlap_data_zygote_TE_s %>% 
  filter(TE == "short TEs" | 
        TE == "long TEs") %>% 
  filter(ID == "seedling siRNA loci" |
           ID == "egg siRNA loci" |
           ID == "zygote siRNA loci" |
           ID == "sperm siRNA loci" |
           ID == "embryo siRNA loci" |
           str_detect(ID, "-")) %>% 
  mutate(ID = case_when(
    str_detect(ID, "embryo") ~ "*embryo siRNA loci",
    T ~ ID
  )) %>% 
  mutate(ID = factor(ID, levels = c(
    "sperm siRNA loci",
    "E loci - Z loci",
    "egg siRNA loci",
    "zygote siRNA loci",
    "Z loci - E loci",
    "*embryo siRNA loci",
    "seedling siRNA loci"
  ))) %>%
  ggplot(aes(x = TE, y = mean)) +
  facet_grid(ID ~ ., switch = "y") +
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  geom_text(aes(label = signif(mean, 3)), size = 3.5, fontface = "bold", hjust = 0, y = 0) +
  geom_text(data = z_cov_lm_results_longshort %>%
              filter(TE == "short TEs" |
                       TE == "long TEs"),
            aes(label = group, y = mean_cov * 100, hjust = -0.3),
            size = 3.5, fontface = "bold") +
  scale_fill_manual(values = c("gold", "slateblue4"),
                 limits = c("short TEs", "long TEs")) +
  labs(x = NULL,
       y = "mean fraction of locus length\noverlapped by TEs (%)",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 1)) + 
  guides(color = F) +
  theme_minimal() +
  theme(legend.position = "top") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_blank()) +
  theme(strip.placement = "outside") + 
  theme(strip.text.y.left = element_text(angle = 0)) +
  coord_flip()

ggsave("short_longTE_overlap_lite.svg", width = 6, height = 4)
ggsave("short_longTE_overlap_lite.png", width = 6, height = 4)
```

```{r}
overlap_data_zygote_TE_s %>% 
  filter(TE == "short TEs" | 
        TE == "long TEs"|
          TE == "CACTA") %>% 
  filter(ID == "seedling siRNA loci" |
           ID == "egg siRNA loci" |
           ID == "zygote siRNA loci" |
           ID == "sperm siRNA loci" |
           ID == "embryo siRNA loci" |
           str_detect(ID, "-")) %>% 
  mutate(ID = case_when(
    str_detect(ID, "embryo") ~ "*embryo siRNA loci",
    T ~ ID
  )) %>% 
  mutate(ID = factor(ID, levels = c(
    "sperm siRNA loci",
    "E loci - Z loci",
    "egg siRNA loci",
    "zygote siRNA loci",
    "Z loci - E loci",
    "*embryo siRNA loci",
    "seedling siRNA loci"
  ))) %>%
  ggplot(aes(x = TE, y = mean)) +
  facet_grid(ID ~ ., switch = "y") +
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  geom_text(aes(label = signif(mean, 3)), size = 3.5, fontface = "bold", hjust = 0, y = 0) +
  scale_fill_manual(values = c("gold", "slateblue4", "indianred3"),
                 limits = c("short TEs", "long TEs", "CACTA")) +
  labs(x = NULL,
       y = "mean fraction of locus length\noverlapped by TEs (%)",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 1)) + 
  guides(color = F) +
  theme_minimal() +
  theme(legend.position = "top") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_blank()) +
  theme(strip.placement = "outside") + 
  theme(strip.text.y.left = element_text(angle = 0)) +
  coord_flip()

ggsave("CACTA_overlap_lite.svg", width = 6, height = 5)
ggsave("CACTA_short_longTE_overlap_lite.png", width = 6, height = 5)
```

 
```{r}
overlap_data_zygote_TE_s %>% 
  filter(TE == "seedling siRNA loci" |
           TE == "embryo DRM2 targets") %>% 
  filter(ID == "seedling siRNA loci" |
           ID == "egg siRNA loci" |
           ID == "zygote siRNA loci" |
           ID == "sperm siRNA loci" |
           ID == "embryo siRNA loci" |
           str_detect(ID, "-")) %>% 
  mutate(ID = case_when(
    str_detect(ID, "embryo") ~ "*embryo siRNA loci",
    T ~ ID
  )) %>% 
  mutate(ID = factor(ID, levels = c(
    "sperm siRNA loci",
    "E loci - Z loci",
    "egg siRNA loci",
    "*embryo siRNA loci",
    "Z loci - E loci",
    "zygote siRNA loci",
    "seedling siRNA loci" 
  ))) %>% 
  ggplot(aes(x = TE, y = mean)) +
  facet_grid(ID ~ ., switch = "y") +
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.5) +
  geom_text(aes(label = signif(mean, 3)), size = 3.5, fontface = "bold", hjust = 0, y = 0) +
  geom_text(data = z_sd_cov_lm_results %>%
              mutate(TE = "seedling siRNA loci"),
            aes(label = group, y = mean_cov * 100, hjust = -0.3, vjust = 0.5),
            size = 3.5, fontface = "bold") +
  geom_text(data = z_cov_lm_results %>%
              filter(TE == "embryo DRM2 targets"),
            aes(label = group, y = mean_cov * 100, hjust = -0.3, vjust = 0.5),
            size = 3.5, fontface = "bold") +
  scale_fill_manual(values = c("seagreen", "blue2"),
                   limits = c("seedling siRNA loci", "embryo DRM2 targets")) +
  labs(x = NULL,
       y = "mean fraction of locus length\noverlapped canonical RdDM loci",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 2)) + 
  guides(color = F) +
  theme_minimal() +
  theme(legend.position = "top") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_blank()) +
  theme(strip.placement = "outside") + 
  theme(strip.text.y.left = element_text(angle = 0)) +
  coord_flip()

ggsave("RdDM_overlap_lite.svg", width = 5, height = 4.2)
ggsave("RdDM_overlap_lite.png", width = 5, height = 4.2)
```
```{r}
DRM2_stuff <- overlap_data_zyogte %>% 
  filter(ID == "seedling siRNA loci" |
                                 ID == "egg siRNA loci" |
                                 ID == "zygote siRNA loci" |
                                 ID == "sperm siRNA loci" |
                                 ID == "embryo siRNA loci" |
                                  str_detect(ID, "-")) %>%
                        mutate(ID = case_when(
                                   str_detect(ID, "embryo siRNA loci") ~ "*embryo siRNA loci",
                                   T ~ ID
                               )) %>% 
  mutate(ID = factor(ID, levels = c(
    "sperm siRNA loci",
    "E loci - Z loci",
    "egg siRNA loci",
    "*embryo siRNA loci",
    "Z loci - E loci",
    "zygote siRNA loci",
    "seedling siRNA loci"
  ))) %>%
  filter(TE == "embryo DRM2 targets") %>% 
  mutate(Length = X3 - X2) %>% 
  group_by(ID) %>% 
  summarise(count = sum(X4),
            total_MB = sum(Length)/10^6,
            NN = n()) %>% 
  cbind(total = 479604) %>% 
  mutate(perc = count/total * 100) %>% 
  mutate(count.per.MB = count/total_MB)  

DRM2_stuff 
```

```{r}
percent_loci <- DRM2_stuff %>% 
  ggplot(aes(x = ID, y = perc)) +
  geom_bar(stat = "identity", aes(fill = ID), alpha = 0.8, width = 0.7) +
  geom_text(aes(label = paste(signif(perc, 2), "%", sep = "")), 
            size = 3.5, fontface = "bold", hjust = 0, y = 0) + 
  scale_fill_manual(values = c("seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c("seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                                "sperm siRNA loci")) +
  labs(x = "siRNA loci categories",
       y = "Partition of DRM2 target (%)") +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(color = "black", hjust = 0.5)) +
  theme(strip.placement = "outside") + 
  coord_flip()                          

count_loci_perMB <- DRM2_stuff %>% 
  ggplot(aes(x = ID, y = count.per.MB)) +
  geom_bar(stat = "identity", aes(fill = ID), alpha = 0.8, width = 0.7) +
  geom_text(aes(label = signif(count.per.MB, 4)), 
            size = 3.5, fontface = "bold", hjust = 0, y = 0) + 
  scale_fill_manual(values = c("seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c("seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                                "sperm siRNA loci")) +
  labs(x = NULL,
       y = "Number of DRM2 targets\n(per Mb genome space)") +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_blank()) +
  theme(strip.placement = "outside") + 
  coord_flip()  

total_MB <- DRM2_stuff %>% 
  ggplot(aes(x = ID, y = total_MB)) +
  geom_bar(stat = "identity", aes(fill = ID), alpha = 0.8, width = 0.7) +
  geom_text(aes(label = paste(
    signif(total_MB, 2),
    " Mb     n = ",
    NN, 
    " loci", sep = ""
  )), 
            size = 3.5, fontface = "bold", hjust = 0, y = 0) + 
  scale_fill_manual(values = c("seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c("seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                                "sperm siRNA loci")) +
  labs(x = NULL,
       y = "Total Mb occupied") +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_blank()) +
  theme(strip.placement = "outside") + 
  coord_flip()

plot_grid(percent_loci, count_loci_perMB, total_MB, nrow = 1,
          align = "h", rel_widths = c(1, 0.6, 0.6))

ggsave("partition_of_DRM2_targets.svg", height = 3, width = 10)
ggsave("partition_of_DRM2_targets.png", height = 3, width = 10)
```
```{r}
DRM2_stuff %>% 
  ggplot(aes(x = ID, y = count.per.MB)) +
  geom_bar(stat = "identity", aes(fill = ID), alpha = 0.8, width = 0.7) +
  geom_text(aes(label = signif(count.per.MB, 4)), 
            size = 3.5, fontface = "bold", hjust = 0, y = 0) + 
  scale_fill_manual(values = c("seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c("seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                                "sperm siRNA loci")) +
  labs(x = NULL,
       y = "Number of DRM2 targets\n(per Mb genome space)") +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(axis.text.x=element_text(colour = "black", hjust = 0.8)) +
  theme(axis.text.y=element_text(color = "black", hjust = 0.5)) +
  theme(strip.placement = "outside") + 
  coord_flip()  

ggsave("partition_of_DRM2_targets_mid.svg", height = 3, width = 4.2)
ggsave("partition_of_DRM2_targets_mid.png", height = 3, width = 4.2)
```

```{r}
DRM2_stuff %>% 
  ggplot(aes(x = total_MB/430 * 100, y = perc)) +
  geom_smooth(method = "lm", alpha = 0.5) +
  geom_point(aes(fill = ID), shape = 21, color = "white", size = 3, alpha = 0.8) +
  scale_fill_manual(values = c("seagreen", "grey20",
                                "pink3", "violetred4", "tomato1",
                               "coral4", "dodgerblue2"),
                    limits = c("seedling siRNA loci",
                               "*embryo siRNA loci",
                               "Z loci - E loci", 
                               "zygote siRNA loci",
                               "egg siRNA loci",
                               "E loci - Z loci",
                                "sperm siRNA loci")) +
  scale_y_continuous(breaks = c(0, 25, 50, 75)) +
  labs(x = "% of the rice genome (Mb/430 Mb)",
       y = "% of DRM2 targets partitioned",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "top") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(axis.text.x=element_text(colour = "black", hjust = 0.8)) +
  theme(axis.text.y=element_text(colour = "black"))  

ggsave("partition_of_DRM2_targets_scatter.svg", height = 6, width = 5)
ggsave("partition_of_DRM2_targets_scatter.png", height = 6, width = 5)
```


## sirens overlap
```{r}
overlap_data_siren <- overlap_data %>% 
  rbind(overlap_data_seedling) %>% 
   filter(str_detect(id, "NOT") == F) %>% 
   filter(str_detect(id, "siren")) 

overlap_data_siren <- overlap_data_siren %>% 
  mutate(ID = case_when(
           str_detect(id, "egg_siren") ~ "egg siren loci",   #1
           str_detect(id, "zygote_siren") ~ "zygote siren loci",  #2
           str_detect(id, "ovary_siren") &
             str_detect(id, "fovary") == F ~ "ovary (0hr) siren loci", #3 
           str_detect(id, "endo_siren") ~ "endosperm siren loci", #4 
           str_detect(id, "fovary_siren") ~ "ovary (9hap) siren loci" #5 
  )) %>% 
   mutate(TE = case_when(
   str_detect(id, "TIR") ~ "TIR",     #1
   str_detect(id, "CentO") ~ "CentO",  #2
   str_detect(id, "Gypsy") ~ "Gypsy",  #3
  # str_detect(id, "Copia") ~ "Copia",   #4
  # str_detect(id, "LINE") ~ "LINE",    #5
  # str_detect(id, "SINE") ~ "SINE",     #6
   str_detect(id, "CACTA") ~ "CACTA",    #7
  # str_detect(id, "Helitron") ~ "Helitron",  #8
   str_detect(id, "genes") ~ "genes",  #9
  str_detect(id, "longTE") ~ "long TEs", #10
  str_detect(id, "shortTE") ~ "short TEs" #11 
  )) %>% 
  filter(is.na(TE) == F)

head(overlap_data_siren)
```

### summary stats
```{r}
overlap_data_siren_TE_s <- overlap_data_siren %>%
  group_by(ID, TE) %>%
  summarise(
    NN = n(),
    mean = mean(X7 * 100)
  ) %>% 
  ungroup()  

head(overlap_data_siren_TE_s)
```
```{r}
overlap_data_siren_TE_s %>% 
  filter(TE == "Gypsy") %>% 
  arrange(mean)

overlap_data_siren_TE_s %>% 
  filter(TE == "long TEs") %>% 
  arrange(mean)
```
### plot
```{r}
##total overlap
overlap_data_siren_TE_s %>% 
  mutate(ID = factor(ID, levels = c(
    "endosperm siren loci",
    "ovary (9hap) siren loci",
    "ovary (0hr) siren loci",
    "zygote siren loci",
    "egg siren loci"
  ))) %>%
  mutate(type2 = case_when(
    TE == "genes" ~ "genes",
    T ~ "TE"
  )) %>% 
  filter(TE == "genes" | 
           TE == "short TEs" |
           TE == "long TEs") %>% 
  mutate(TE = factor(TE, levels = c(
     "genes", "short TEs", "long TEs" 
  ))) %>% 
  ggplot(aes(x = ID, y = mean)) +
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  annotate(geom = "text", label = c("b", "a", "a", "a", "a"),
           x = c(1, 2, 3, 4, 5),
           y = c(30, 35, 35, 50, 50),
           hjust = -0.1, size = 5, fontface = "bold", color = "grey90") +
  scale_fill_manual(values = c("grey10", 
                               "gold", "slateblue4"),
                   limits = c("genes", "short TEs", "long TEs")) +
  labs(x = NULL,
       y = "mean fraction of locus length\noverlapped by TEs or genes (%)",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 1)) + 
  guides(color = F) +
  theme_minimal() +
  theme(legend.position = "bottom") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("TE_overlap_stackbar.svg", height = 3.5, width = 7)
ggsave("TE_overlap_stackbar.png", height = 3.5, width = 7)
```


```{r}
##total overlap
overlap_data_siren_TE_s %>% 
  mutate(ID = factor(ID, levels = c(
    "endosperm siren loci",
    "ovary (9hap) siren loci",
    "ovary (0hr) siren loci",
    "zygote siren loci",
    "egg siren loci"
  ))) %>%
  mutate(type2 = case_when(
    TE == "genes" ~ "genes",
    T ~ "TE"
  )) %>% 
  filter(TE == "genes" | 
           TE == "TIR" |
           TE == "Gypsy") %>% 
  mutate(TE = factor(TE, levels = c(
     "genes", "TIR", "Gypsy" 
  ))) %>% 
  ggplot(aes(x = ID, y = mean)) +
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  annotate(geom = "text", label = c("b", "a", "a", "a", "a"),
           x = c(1, 2, 3, 4, 5),
           y = c(30, 25, 25, 40, 40),
           hjust = -0.1, size = 5, fontface = "bold", color = "grey90") +
  scale_fill_manual(values = c("grey10", 
                               "orange2", "grey80"),
                   limits = c("genes", "TIR", "Gypsy")) +
  labs(x = NULL,
       y = "mean fraction of locus length\noverlapped by TEs or genes (%)",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 1)) + 
  guides(color = F) +
  theme_minimal() +
  theme(legend.position = "bottom") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black", hjust = 0.5)) +
  coord_flip()


ggsave("TE_overlap_stackbar2.svg", height = 3.5, width = 7)
ggsave("TE_overlap_stackbar2.png", height = 3.5, width = 7)
```

```{r}
overlap_data_siren_TE_s %>% 
  filter(TE == "genes") %>% 
  filter(str_detect(ID, "siren")) %>% 
  arrange(mean)
```

* endosperm ~30%
* others ~12 - 14% 

### stats 
```{r}
overlap_data_siren %>% head()

model_siren_overlap <- glm(X7 ~ ID * TE, data = overlap_data_siren %>% 
                             filter(str_detect(ID, "siren")) %>%
                             filter(str_detect(TE, "genes|short TEs|long TEs")),
                                    family = quasibinomial()) 

anova(model_siren_overlap, test = "LRT")
```
```{r}
est_siren_TE_overlap <- emmeans(model_siren_overlap, pairwise ~ ID | TE)
multcomp::cld(est_siren_TE_overlap$emmeans, Letters = letters)
```
```{r}
est_siren_TE_overlap$contrasts %>% 
 # summary(infer = c(T, T)) %>% 
  as.data.frame() %>% 
  filter(TE == "genes") %>% 
  select(contrast, p.value)
```


 

 

