---
title: "siRNA lengths and TE abundances"
author: "Chenxin Li"
date: "Sep 03, 2019"
output:
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(dplyr)
library(tidyr)
library(readr)
library(readxl)
library(emmeans) 
library(stringr)
library(RColorBrewer)
library(svglite)
```

# smRNA composition 
```{r}
sample_des <-  read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/sample_des_clvs12.xlsx")  

smRNA_comp <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/codes_and_output/smRNA_comp.csv")
```


## overview of smRNA composition
```{r}
smRNA_comp %>% 
  full_join(sample_des, by = "sample_ID") %>% 
  select(-`minus mRNAs`, -`minus tRNAs`, -`minus NOR RNAs`, -`minus 5S RNAs`, -`minus 21nt phasi loci RNAs`) 
  
```

```{r}
smRNA_comp_L <- smRNA_comp %>%
  full_join(sample_des, by = "sample_ID") %>% 
  select(-`minus mRNAs`, -`minus tRNAs`, -`minus NOR RNAs`, -`minus 5S RNAs`, -`minus 21nt phasi loci RNAs`) %>% 
  gather("RNA_type", "count", 2:8) %>% 
  mutate(pro = count / `total read`) %>% 
  select(-count, -library) %>% 
  select(sample_ID, sample_type, genotype, pro, RNA_type) %>% 
  mutate(RNA_type1 = case_when(
    RNA_type == "general siRNA" ~ "general siRNA",
    RNA_type == "miRNA" ~ "miRNAs",
    str_detect(RNA_type, "tRNA") ~ "tRNAs", 
    str_detect(RNA_type, "NOR") ~ "NOR RNAs", 
    str_detect(RNA_type, "5S") ~ "5S rRNAs",
    str_detect(RNA_type, "21nt") ~ "phasiRNAs (21nt)",
    str_detect(RNA_type, "24nt") ~ "phasiRNAs (24nt)"
  )) %>% 
  as.data.frame() 


smRNA_comp_L
```

 
```{r}
smRNA_comp_L %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap") %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" , 
    T ~ sample_type
  )) %>% 
  filter(genotype == "WT") %>% 
  filter(RNA_type != "general siRNA") %>% 
  mutate(RNA_type2 = factor(RNA_type1, levels = c(
    "miRNAs", "phasiRNAs (24nt)", "phasiRNAs (21nt)", "tRNAs", "NOR RNAs", "5S rRNAs"
  ))) %>% 
  # mutate(sample_type.f = factor(sample_type, levels = c(
  #      "egg", "seedling shoot", "sperm", "zygote"
  # ))) %>%
  ggplot(aes(x = sample_type, y = pro * 100)) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_bar(stat = "summary", aes(fill = sample_type), fun = mean, alpha = 0.8) +
  geom_point(color = "white", shape = 21, aes(fill = sample_type), alpha = 0.8, position = position_jitter(0.1, seed = 666)) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(), position = position_dodge(0.01), width = 0.3) +
  facet_wrap( ~ RNA_type2,  scales = "free_y", ncol = 2) +
  scale_fill_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2", "violetred4", "coral4"),
                    limits = c("ovary (0hr)",  "egg", "seedling", "sperm", "zygote", "ovary (9hap)")) +
  guides(fill = guide_legend(ncol = 2)) + 
  labs(x = NULL, 
       y = "% small RNA",
       fill = NULL) +
  theme_minimal() +
  theme(legend.position = "bottom", legend.direction = "horizontal") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_blank()) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave("smRNA_comp_overall_essz.svg", width = 4.9, height = 6)
ggsave("smRNA_comp_overall_essz.png", width = 4.9, height = 6)
```



# abundance of siRNA at TE & loci categories across sample types 
## load data 
```{r message=F, warning= F} 
length_list <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/length_summary_shortstack_loci/", 
                        pattern = "*.txt", full.names = T) 

length_data <- sapply(length_list, read_delim, simplify = FALSE, delim = "\t") %>% 
  bind_rows(.id = "id")

head(length_data) 
```

 
```{r}
sample_des_size <- smRNA_comp %>% 
  full_join(sample_des, by = c("sample_ID")) 

sample_des_size
```
 
## keying data 
```{r}
 identify_samples2 <- function(df){
  df %>% 
    mutate(sample_ID = case_when(
    str_detect(X1, "sperm.1") ~ "BS1", #1
    str_detect(X1, "Sperm2") ~ "BS2",  #2
    str_detect(X1, "Sperm4a") ~ "BS4a",  #3
    str_detect(X1, "sperm4b") ~ "BS4b",  #4
    str_detect(X1, "sperm6a") ~ "BS6a",  #5
    str_detect(X1, "sperm6b") ~ "BS6b",  #6
    str_detect(X1, "sperm.5b") ~ "SP5b", #7
    str_detect(X1, "sperm.5c") ~ "SP5c", #8
    str_detect(X1, "Bulk.15") ~ "B15",   #9
    str_detect(X1, "Bulk.20") ~ "B20",   #10
    str_detect(X1, "Bulk.25") ~ "B25",   #11
    str_detect(X1, "seedling4") ~ "SD4", #12
    str_detect(X1, "egg.9a") ~ "E9a",    #13
    str_detect(X1, "egg.9b") ~ "E9b",    #14
    str_detect(X1, "egg.9c") ~ "E9c",    #15
    str_detect(X1, "eggcell4") ~ "E4",   #16
    str_detect(X1, "eggcell3") ~ "E3",   #17
    str_detect(X1, "eggcell2") ~ "E2",   #18
    str_detect(X1, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(X1, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(X1, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(X1, "SRR57") ~ "root",              #22
    str_detect(X1, "ab_control") ~ "ddm1_con",     #23
    str_detect(X1, "ddm1ab") ~ "ddm1",             #24
    str_detect(X1, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(X1, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(X1, "SRR2544786") ~ "em_dry", #27  
    str_detect(X1, "SRR5049778") ~ "an_pre",  #28
    str_detect(X1, "SRR5049779") ~ "an_mei",  #29
    str_detect(X1, "SRR5049780") ~ "an_mic",  #30
    str_detect(X1, "SRR5049781") ~ "an_bi",   #31
    str_detect(X1, "SRR5049786") ~ "ov_1st",  #32
    str_detect(X1, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(X1, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(X1, "SRR5049789") ~ "ov_4th",  #35
    str_detect(X1, "S3_1") ~ "S3_1",          #36
    str_detect(X1, "S3_2") ~ "S3_2",          #37
    str_detect(X1, "S3_3") ~ "S3_3",          #38
    str_detect(X1, "S5_1") ~ "S5_1",          #39
    str_detect(X1, "S5_2") ~ "S5_2",          #40
    str_detect(X1, "S5_3") ~ "S5_3",          #41
    str_detect(X1, "S7_1") ~ "S7_1",          #42
    str_detect(X1, "S7_2") ~ "S7_2",          #43
    str_detect(X1, "S7_3") ~ "S7_3",          #44
    str_detect(X1, "SRR2544787") ~ "em_12h",  #45
    str_detect(X1, "SRR2544788") ~ "em_24h",  #46
    str_detect(X1, "SRR771501") ~ "em_7DAF",  #47
    str_detect(X1, "SRR771500") ~ "en_7DAF",  #48   
    str_detect(X1, "SD7") ~ "SD7", #49
    str_detect(X1, "zg1d") ~ "zg1d", #50
    str_detect(X1, "zg1e") ~ "zg1e",  #51
    str_detect(X1, "zg2d") ~ "zg2d",  #52
    str_detect(X1, "zg2e") ~ "zg2e",  #53
    str_detect(X1, "zg3a") ~ "zg3a", #54
    str_detect(X1, "zg3b") ~ "zg3b" , #55
    str_detect(X1, "OV9H_1") ~ "OV9_1" , #56
    str_detect(X1, "OV9H_2") ~ "OV9_2" , #57
    str_detect(X1, "OV9H_3") ~ "OV9_3" , #58
    str_detect(X1, "SD8") ~ "SD8"  #59
    ))
} 
```

```{r}
length_data <- length_data %>% 
  identify_samples2() %>% 
  mutate(category = case_when(
    str_detect(id, "summary.bam.sam") ~ "general siRNA",  #1
    str_detect(id, "egg_loci") ~ "egg siRNA loci",  #2
    str_detect(id, "egg-zygote_com") ~ "egg NOT zygote\nloci",  #3 
    str_detect(id, "embryo_loci") ~ "embryo siRNA loci",   #4
    str_detect(id, "endo_siren") ~ "endosperm siren loci",   #5
    str_detect(id, "endosperm_loci") ~ "endosperm siRNA loci",  #6
    str_detect(id, "fovary_loci") ~ "ovary (9hap) siRNA loci",  #7
    str_detect(id, "fovary_siren") ~ "ovary (9hap) siren loci",  #8
    str_detect(id, "Gypsy.") ~ "Gypsy",  #9
    str_detect(id, "y.ovary_loci") ~ "ovary (0hr) siRNA loci",  #10
    str_detect(id, "y.ovary_siren") ~ "ovaery (0hr) siren loci",  #11
    str_detect(id, "q20") ~ "uniquely mapped",   #12
    str_detect(id, "seedling_loci") ~ "seedling siRNA loci", #13 
    str_detect(id, "sperm_loci") ~ "sperm siRNA loci",  #14
    str_detect(id, "TIR") ~ "TIR",   #15
    str_detect(id, "zygote_loci") ~ "zygote siRNA loci",  #16
    str_detect(id, "seedling-specific") ~ "seedling-specific\nloci", #17
    str_detect(id, "sperm-specific") ~ "sperm-specific\nloci", #18
    str_detect(id, "egg-specific") ~ "egg-specific\nloci", #19
    str_detect(id, "intersection_loci") ~ "egg/sperm/seedling\nintersect loci", #20
    str_detect(id, "zygote-egg_com") ~ "zygote NOT egg\nloci", #21
    str_detect(id, "zygote-egg_inter") ~ "zygote/egg\nintersect" #22 
  )) %>% 
  gather("length", "count", c(`20`, `21`, `22`, `23`, `24`, `25`)) %>% 
  inner_join(sample_des_size, by = "sample_ID") %>% 
  mutate(proportion = count/`general siRNA`)  

head(length_data)
```

 

 
## length profile - line plot 
```{r}
length_data %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap") %>% 
  filter(str_detect(category, "siren") == F) %>% 
  filter(str_detect(category, "TIR|Gypsy|general")) %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" ,
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) %>% 
  mutate(category = factor(category, levels = c(
    "general siRNA", "TIR", "Gypsy",
    "egg NOT zygote\nloci", "zygote NOT egg\nloci"
  ))) %>% 
  ggplot(aes(x = length, y = proportion*100 )) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  facet_wrap( ~ category, scales = "free_y", nrow = 1) +
  stat_summary(geom = "line", fun = mean, aes(group = sample_type, color = sample_type), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", 
               aes(color = sample_type), position = position_dodge(0.01)) + 
  scale_color_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2", "coral4", "violetred4" ),
                    limits = c("ovary (0hr)",  "egg", "seedling", "sperm", "ovary (9hap)","zygote")) +  
  labs(color = NULL,
       x = "length (nt)",
       y = "% siRNA") +
  theme_minimal() +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(legend.position = "top") 

ggsave(filename = "siRNA profile gamemtes.svg", width = 7, height = 4)
ggsave(filename = "siRNA profile gamemtes.png", width = 7, height = 4)
```


## stat 
### TIR, Gypsy and zygote NOT egg, egg NOT zygote 
```{r}
anova_data <- length_data %>% 
  filter(genotype == "WT") %>% 
 filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap" |
           sample_type == "embryo 7-8 DAF") %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" , 
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) %>% 
  filter(length == 24) %>% 
  filter(str_detect(category, "TIR|Gypsy|NOT|zygote/egg"))  
```

### summary stats, zygote NOT egg 
```{r}
anova_data %>% 
  filter(category == "zygote NOT egg\nloci") %>% 
  group_by(sample_type) %>% 
  summarise(
    mean = mean(proportion)
  )
```
 
* zygote / egg = 8%/1% = 8x 
* ovary9hap / ovary0h = 2.8% / 3.2% = 0.9x 

### summary stats, egg NOT zygote 
```{r}
anova_data %>% 
  filter(category == "egg NOT zygote\nloci") %>% 
  group_by(sample_type) %>% 
  summarise(
    mean = mean(proportion)
  )
```
* egg vs zygote: 7x 
* ovaries: 0.018 vs 0.012 

### summary stats, zygote egg intersect
```{r}
anova_data %>% 
  filter(category == "zygote/egg\nintersect") %>% 
  group_by(sample_type) %>% 
  summarise(
    mean = mean(proportion)
  )
```


### summary stats, TE 
```{r}
anova_data %>% 
  filter(str_detect(category, "Gypsy|TIR")) %>% 
  group_by(sample_type, category) %>% 
  summarise(
    mean = mean(proportion)
  ) %>% 
  arrange(category)
```
* Gypsy: sperm vs. zygote 3x
* TIR: zygote vs. seedling 3x 

### linear models 
```{r}
logit <- function(p){log(
  p / (1-p)
)}

logistic <- function(x){
  1/(1 + exp(-x))
}
```

```{r}
model_24 <- lm(proportion ~ category * sample_type, data = anova_data)
anova(model_24)
```
```{r}
model_24_logit <- lm(logit(proportion) ~ category * sample_type, data = anova_data)
anova(model_24_logit)
```

```{r}
est_24_logit <- emmeans(model_24_logit, pairwise  ~ sample_type | category)

NOT_results <- multcomp::cld(est_24_logit$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(grouping = str_remove_all(.group, " ")) %>% 
  filter(str_detect(category, "NOT|inter"))  %>% 
  filter(sample_type != "embryo") 
 

NOT_results
```

```{r}
TE_results <- multcomp::cld(est_24_logit$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(grouping = str_remove_all(.group, " ")) %>% 
  filter(str_detect(category, "TIR|Gypsy"))  %>% 
  filter(sample_type != "embryo") 

TE_results
```


```{r}
est_24_logit$contrasts %>% 
  as.data.frame() 
```
* TIR: zygote vs. seedling: 3.8e-14
* Gypsy: zygote vs sperm: 3.6e-14 

* egg NOT zygote: ovary 0 - ovary 9: 0.1 NS
* egg NOT zygote: egg - zygote: 4e-14 ***

* zygote NOT egg: ovary 0 - ovary 9: 1 
* zygote NOT egg: egg - zygote: 4e-14 *** 

* zygote-egg intersect: egg - zygote NS
* zygote-egg intersect: ovary 0 - ovary 9 NS

## plot 
### TE 
```{r}
annotate_TE1 <- data.frame(
    xmin = 5,
    xmax = 6
  ) %>% 
  mutate(category = "TIR") %>% 
  mutate(sample_type = "seedling" ) %>% 
  mutate(proportion = 0.15) %>% 
  mutate(text = "3.8e-14") %>% 
  mutate(vjust = 2)%>% 
  mutate(hjust = -0.08)

annotate_TE2 <-  data.frame(
    xmin = 1,
    xmax = 5
  ) %>% 
  mutate(category = "Gypsy") %>% 
  mutate(sample_type = "zygote" ) %>% 
  mutate(proportion = 0.18) %>% 
  mutate(text = "3.6e-14") %>% 
  mutate(vjust = 6)%>% 
  mutate(hjust = -0.1)

annotate_TE <- rbind(
   annotate_TE1,
   annotate_TE2
) %>% 
  mutate(category = factor(category, levels = c("TIR", "Gypsy")))
```

```{r}
length_data %>% 
  filter(genotype == "WT") %>% 
  filter(length == 24) %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap" ) %>% 
  filter(str_detect(category, "siren") == F) %>% 
  filter(str_detect(category, "TIR|Gypsy")) %>%
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" ,
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) %>% 
  mutate(sample_type = factor(sample_type, levels = rev(c(
      "seedling", "embryo", "zygote", "egg","ovary (0hr)", "ovary (9hap)", "sperm"
  )))) %>%  
  mutate(category = factor(category, levels = c("TIR", "Gypsy"))) %>% 
  ggplot(aes(x = sample_type, y = proportion*100 )) +
  facet_grid(. ~ category, scales = "free_y") +
  geom_bar(stat = "summary", fun = mean, aes(fill = sample_type), alpha = 0.8, width =0.7) +
  geom_point(aes(fill = sample_type), color = "white", shape = 21, alpha = 0.8, 
             position = position_jitter(0.1, seed = 666), size = 2) +
  geom_text(data = TE_results,
            aes(label = grouping), size = 4, fontface = "bold", y = 0,
            hjust = 0) +
  geom_errorbarh(data = annotate_TE, aes(xmin = xmin, xmax = xmax),
                  size = 1, height = 1) +
  geom_text(data = annotate_TE, aes(label = text, hjust = hjust, vjust = vjust),
            size = 4, fontface = "bold") +
  scale_fill_manual(values = c("seagreen", "violetred4", "tomato1", "orangered3", "coral4", "dodgerblue2"),
                    limits = c(
      "seedling", "zygote", "egg","ovary (0hr)", "ovary (9hap)", "sperm"
  )) +  
  labs(fill = NULL,
       x = "siRNA trancriptome",
       y = "24-nt siRNA (% of total siRNA)") +
  guides(fill = guide_legend(ncol = 2)) +
  theme_minimal() +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_blank())  +
  theme(axis.line = element_line(size = 1.1)) +
  theme(panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank()) +
  theme(legend.position = "top") +
  theme(strip.placement = "outside") +
  theme(panel.spacing = unit(0.5, "line")) +
  coord_flip()

ggsave(filename = "siRNA_24_TE_bar.svg", width = 3.5, height = 4) 
ggsave(filename = "siRNA_24_TE_bar.png", width = 3.5, height = 4) 
```

### zygote and egg 
```{r}
annotate_egg_NOT1 <- data.frame(
    xmin = 4,
    xmax = 5
  ) %>% 
  mutate(category = "egg NOT zygote\nloci") %>% 
  mutate(sample_type = "egg" ) %>% 
  mutate(proportion = 0.075) %>% 
  mutate(text = "4e-14") %>% 
  mutate(vjust = -0.8) %>% 
  mutate(hjust = -0.1) %>% 
  mutate(height = 0.5)

annotate_egg_NOT2 <-  data.frame(
    xmin = 2,
    xmax = 3
  ) %>% 
  mutate(category = "egg NOT zygote\nloci") %>% 
  mutate(sample_type = "ovary (9hap)" ) %>% 
  mutate(proportion = 0.025) %>% 
  mutate(text = "NS") %>% 
  mutate(vjust = -0.8) %>% 
  mutate(hjust = -0.2) %>% 
  mutate(height = 0.5)

annotate_zygote_NOT1 <- data.frame(
    xmin = 4,
    xmax = 5
  ) %>% 
  mutate(category = "zygote NOT egg\nloci") %>% 
  mutate(sample_type = "egg" ) %>% 
  mutate(proportion = 0.08) %>% 
  mutate(text = "4e-14") %>% 
  mutate(vjust = -0.8) %>% 
  mutate(hjust = -0.1) %>% 
  mutate(height = 0.5)

annotate_zygote_NOT2 <-  data.frame(
    xmin = 2,
    xmax = 3
  ) %>% 
  mutate(category = "zygote NOT egg\nloci") %>% 
  mutate(sample_type = "ovary (9hap)" ) %>% 
  mutate(proportion = 0.04) %>% 
  mutate(text = "NS") %>% 
  mutate(vjust = -0.8) %>% 
  mutate(hjust = -0.2) %>% 
  mutate(height = 0.5)

annotate_inter_NOT1 <-  data.frame(
    xmin = 2,
    xmax = 3
  ) %>% 
  mutate(category = "zygote/egg\nintersect") %>% 
  mutate(sample_type = "ovary (9hap)" ) %>% 
  mutate(proportion = 0.35) %>% 
  mutate(text = "NS") %>% 
  mutate(vjust = -0.8) %>% 
  mutate(hjust = -0.2) %>% 
  mutate(height = 2) 


annotate_inter_NOT2 <-  data.frame(
    xmin = 4,
    xmax = 5
  ) %>% 
  mutate(category = "zygote/egg\nintersect") %>% 
  mutate(sample_type = "egg" ) %>% 
  mutate(proportion = 0.3) %>% 
  mutate(text = "NS") %>% 
  mutate(vjust = -0.8) %>% 
  mutate(hjust = -0.2) %>% 
  mutate(height = 2) 

annotate_NOT <- rbind(
  annotate_egg_NOT1,
  annotate_egg_NOT2,
  annotate_zygote_NOT1,
  annotate_zygote_NOT2,
  annotate_inter_NOT1,
  annotate_inter_NOT2
) %>% 
  mutate(category = factor(category, levels = c(
    "egg NOT zygote\nloci", "zygote/egg\nintersect" ,"zygote NOT egg\nloci"
    ))) 
```

```{r}
length_data %>% 
  filter(genotype == "WT") %>% 
  filter(length == 24) %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap" ) %>% 
  filter(str_detect(category, "siren") == F) %>% 
  filter(str_detect(category, "NOT") | 
                      str_detect(category, "zygote/egg\ninter")) %>%
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" ,
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) %>% 
  mutate(sample_type = factor(sample_type, levels = rev(c(
      "seedling", "embryo", "zygote", "egg","ovary (0hr)", "ovary (9hap)", "sperm"
  )))) %>%  
  mutate(category = factor(category, levels = c(
    "egg NOT zygote\nloci", "zygote/egg\nintersect" ,"zygote NOT egg\nloci"
    ))) %>% 
  ggplot(aes(x = sample_type, y = proportion*100 )) +
  facet_grid(. ~ category, scales = "free_x") +
  geom_bar(stat = "summary", fun = mean, aes(fill = sample_type), alpha = 0.8, width =0.7) +
  geom_point(aes(fill = sample_type), color = "white", shape = 21, alpha = 0.8, 
             position = position_jitter(0.1, seed = 666), size = 2) +
  geom_text(data = NOT_results,
            aes(label = grouping), size = 4, fontface = "bold", y = 0,
            hjust = 0) +
  geom_errorbarh(data = annotate_NOT, aes(xmin = xmin, xmax = xmax, height = height),
                  size = 1) +
  geom_text(data = annotate_NOT, aes(label = text, vjust = vjust, hjust = hjust), 
            size = 4, fontface = "bold") +
  scale_fill_manual(values = c("seagreen", "violetred4", "tomato1", "orangered3", "coral4", "dodgerblue2"),
                    limits = c(
      "seedling", "zygote", "egg","ovary (0hr)", "ovary (9hap)", "sperm"
  )) +  
  labs(fill = NULL,
       x = "siRNA trancriptome",
       y = "24-nt siRNA (% of total siRNA)") +
  guides(fill = guide_legend(ncol = 2)) +
  theme_minimal() +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank()) +
  theme(legend.position = "none") +
  theme(strip.placement = "outside") +
  theme(panel.spacing = unit(1, "line")) +
  coord_flip()

ggsave(filename = "siRNA_24_NOT_bar.svg", width = 8, height = 3)
ggsave(filename = "siRNA_24_NOT_bar.png", width = 8, height = 3)
```
```{r}
length_data %>% 
  filter(genotype == "WT") %>% 
  filter(length == 24) %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap" ) %>% 
  filter(str_detect(category, "siren") == F) %>% 
  filter(str_detect(category, "NOT")) %>%
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" ,
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) %>% 
  mutate(sample_type = factor(sample_type, levels = rev(c(
      "seedling", "embryo", "zygote", "egg","ovary (0hr)", "ovary (9hap)", "sperm"
  )))) %>%  
  ggplot(aes(x = sample_type, y = proportion*100 )) +
  facet_grid(category ~. , scales = "free_y", switch = "y") +
  geom_bar(stat = "summary", fun = mean, aes(fill = sample_type), alpha = 0.8, width =0.7) +
  geom_point(aes(fill = sample_type), color = "white", shape = 21, alpha = 0.8, 
             position = position_jitter(0.1, seed = 666), size = 2) +
  geom_text(data = NOT_results %>% 
                   filter(str_detect(category, "NOT")),
            aes(label = grouping), size = 4, fontface = "bold", y = 0,
            hjust = 0) +
  geom_errorbarh(data = annotate_NOT %>% 
                   filter(str_detect(category, "NOT")), aes(xmin = xmin, xmax = xmax),
                  size = 1, height = 0.3) +
  geom_text(data = annotate_NOT %>%
                   filter(str_detect(category, "NOT")), aes(label = text, vjust = vjust, hjust = hjust), 
            size = 4, fontface = "bold") +
  scale_fill_manual(values = c("seagreen", "violetred4", "tomato1", "orangered3", "coral4", "dodgerblue2"),
                    limits = c(
      "seedling", "zygote", "egg","ovary (0hr)", "ovary (9hap)", "sperm"
  )) +  
  labs(fill = NULL,
       x = NULL,
       y = "24-nt siRNA (% of total siRNA)") +
  guides(fill = guide_legend(ncol = 2)) +
  theme_minimal() +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_blank()) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank()) +
  theme(legend.position = "top") +
  theme(strip.placement = "outside") +
  theme(panel.spacing = unit(1, "line")) +
  coord_flip()

ggsave(filename = "siRNA_24_NOT_bar_thin.svg", width = 4, height = 5)
ggsave(filename = "siRNA_24_NOT_bar_thin.png", width = 4, height = 5)
```

## stat for seedling and sperm specific loci 
```{r}
specific_data <- length_data %>% 
  filter(genotype == "WT") %>% 
 filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap" 
           ) %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" , 
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) %>% 
  filter(length == 24) %>% 
  filter(str_detect(id, "specific|intersect")) %>% 
  filter(category == "seedling-specific\nloci" |
           category == "sperm-specific\nloci")  

head(specific_data)
```

### summary stats for seedling- and sperm- specific data 
```{r}
specific_data %>% 
  filter(category == "seedling-specific\nloci" |
           category == "sperm-specific\nloci")  %>% 
  group_by(category, sample_type) %>% 
  summarise(
    mean = mean(proportion)
  ) %>% 
  spread(category, mean)
```

* seedling specific: 
** ovary 9hr vs 0hr: 0.1 / 0.08 = 1.25x 
** zygote vs. egg: 0.078 / 0.036 = 2.16x 

* sperm specific: 
** ovary 9hr vs 0hr: 0.0072 / 0.0015 = 0.48x 
** zygote vs. egg: 0.013 / 0.036 = 0.36x  
** zygote vs. sperm: 0.013 / 0.2 = 0.065x 

### linear models 
```{r}
model_spec_24 <- lm(proportion ~ category * sample_type, data = specific_data)
anova(model_spec_24)

#plot(model_spec_24)
```

```{r}
model_spec_24_logit <- lm(logit(proportion) ~  sample_type * category, data = specific_data)
anova(model_spec_24_logit)

#plot(model_spec_24_logit)
```


```{r}
est_spec_24 <- emmeans(model_spec_24, pairwise ~ sample_type | category)
spec_results <- multcomp::cld(est_spec_24$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(grouping = str_remove_all(.group, " ")) 
 
spec_results
```
```{r}
est_spec_24$contrasts %>% 
  as.data.frame() 
  
```


```{r}
est_spec_24_logit <- emmeans(model_spec_24_logit, pairwise ~ sample_type | category)
spec_results_logit <- multcomp::cld(est_spec_24_logit$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(grouping = str_remove_all(.group, " ")) 
 
spec_results_logit
```
```{r}
est_spec_24_logit$contrasts %>% 
  as.data.frame() 
```
* seedling spec: egg vs. zygote = 3e-3
* seedling spec: ovary = NS 

* sperm spec: zygote vs sperm = 0
* sperm spec: ovary = 0.14
* sperm spec: zygote vs egg = 0.019

## plot 
```{r}
annotate_seedling_spec1 <- data.frame(
    xmin = 4,
    xmax = 5
  ) %>% 
  mutate(category = "seedling-specific\nloci") %>% 
  mutate(sample_type = "egg" ) %>% 
  mutate(proportion = 0.1) %>% 
  mutate(text = "3e-3") %>% 
  mutate(vjust = -1.5) %>% 
  mutate(hjust = -0.15)

annotate_seedling_spec2 <-  data.frame(
    xmin = 2,
    xmax = 3
  ) %>% 
  mutate(category = "seedling-specific\nloci") %>% 
  mutate(sample_type = "ovary (9hap)" ) %>% 
  mutate(proportion = 0.12) %>% 
  mutate(text = "NS") %>% 
  mutate(vjust = -1.5)%>% 
  mutate(hjust = -0.2)

annotate_sperm_spec1 <- 
  data.frame(
    xmin = 1,
    xmax = 5
  ) %>% 
  mutate(category = "sperm-specific\nloci") %>% 
  mutate(sample_type = "sperm" ) %>% 
  mutate(proportion = 0.22) %>% 
  mutate(text = "0") %>% 
  mutate(vjust = -8)%>% 
  mutate(hjust = -0.5)

annotate_sperm_spec2 <-  data.frame(
    xmin = 2,
    xmax = 3
  ) %>% 
  mutate(category = "sperm-specific\nloci") %>% 
  mutate(sample_type = "ovary (9hap)" ) %>% 
  mutate(proportion = 0.04) %>% 
  mutate(text = "NS") %>% 
  mutate(vjust = -1.5)%>% 
  mutate(hjust = -0.2)

annotate_sperm_spec3 <-  data.frame(
    xmin = 4,
    xmax = 5
  ) %>% 
  mutate(category = "sperm-specific\nloci") %>% 
  mutate(sample_type = "egg" ) %>% 
  mutate(proportion = 0.04) %>% 
  mutate(text = "0.019") %>% 
  mutate(vjust = -1.5)%>% 
  mutate(hjust = -0.1)

annotate_spec <- rbind(
  annotate_seedling_spec1,
  annotate_seedling_spec2,
  annotate_sperm_spec1,
  annotate_sperm_spec2,
  annotate_sperm_spec3
)
```

 
```{r}
length_data %>% 
  filter(genotype == "WT") %>% 
  filter(length == 24) %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap" ) %>% 
 filter(str_detect(category, "seedling-specific") |
            str_detect(category, "sperm-specific") 
           ) %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" ,
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) %>% 
  mutate(sample_type = factor(sample_type, levels = rev(c(
      "seedling", "zygote", "egg","ovary (0hr)", "ovary (9hap)", "sperm"
  )))) %>%  
  ggplot(aes(x = sample_type, y = proportion*100 )) +
  facet_grid(. ~ category, scales = "free_y") +
  geom_bar(stat = "summary", fun = mean, aes(fill = sample_type), alpha = 0.8, width =0.7) +
  geom_point(aes(fill = sample_type), color = "white", shape = 21, alpha = 0.8, 
             position = position_jitter(0.1, seed = 666), size = 2) +
  geom_text(data = spec_results_logit,
             aes(label = grouping), size = 4, fontface = "bold", y = 0,
             hjust = 0) +
  geom_errorbarh(data = annotate_spec, aes(xmin = xmin, xmax = xmax),
                  size = 1, height = 0.5) +
  geom_text(data = annotate_spec, aes(label = text, vjust = vjust, hjust = hjust), 
            size = 4, fontface = "bold") +
  scale_fill_manual(values = c("seagreen", "violetred4", "tomato1", "orangered3", "coral4", "dodgerblue2"),
                    limits = c(
      "seedling", "zygote", "egg","ovary (0hr)", "ovary (9hap)", "sperm"
  )) +  
  labs(fill = NULL,
       x = "siRNA trancriptome",
       y = "24-nt siRNA (% of total siRNA)") +
  guides(fill = guide_legend(ncol = 2)) +
  theme_minimal() +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  theme(axis.line = element_line(size = 1.1)) +
  theme(panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank()) +
  theme(legend.position = "none") +
  theme(strip.placement = "outside") +
  theme(panel.spacing = unit(1, "line")) +
  coord_flip()

ggsave(filename = "siRNA_24_spec_bar_lite.svg", width = 7, height = 4)
ggsave(filename = "siRNA_24_spec_bar_lite.png", width = 7, height = 4)
```





