---
title: "methylation at siRNA loci"
author: "Chenxin Li"
date: "12/29/2020"
output:
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(dplyr)
library(tidyr)
library(readr)
library(readxl)
library(emmeans) 
library(stringr)
library(RColorBrewer)
library(svglite) 
```

# load data 

read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/drm2_ATCGmaps/wt3_em_S47_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam.CGmap.gz.CHH.zygote-egg_intersect_loci.mtr", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 
    
    
```{r message=F, warning = F}
zy_meth_list <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_zygote_GR_revision/full_MTRs_zes_2021_10_13/zygote_zhou/", 
                        pattern = "*.mtr", full.names = T) 

zy_meth_data <- sapply(zy_meth_list, read_delim, simplify = FALSE, 
                       delim = "\t", col_names = F, escape_double = F, trim_ws = T) %>% 
  bind_rows(.id = "id")

head(zy_meth_data) 
```

```{r}
zy_meth_data %>% 
  group_by(id) %>% 
  count()
```
 

# keying data 

each locus must have 3 or more reads to be considered in the analyses 
```{r}
set.seed(666)
zy_meth_data_nice <- zy_meth_data %>%
  #group_by(id) %>% 
  #sample_n(10000) %>% 
  ungroup() %>% 
  filter(X7 >= 3) %>% 
   mutate(sample_ID = case_when(
    str_detect(id, "SRR10915153") ~ "zygote Zhou et al. WT_1", 
    str_detect(id, "SRR10915154") ~ "zygote Zhou et al. WT_2", 
  )) %>% 
  filter(is.na(sample_ID) == F) %>% 
  mutate(locus = case_when(
    str_detect(id, "egg_loci") ~ "egg siRNA loci", #1
    str_detect(id, "seedling_loci") ~ "seedling siRNA loci", #2
    str_detect(id, "embryo_loci") ~ "embryo siRNA loci", #3 
    str_detect(id, "zygote-egg_complement_loci.") ~ "Z loci - E loci", #4
    str_detect(id, "sperm_loci") &
      str_detect(id, "endo") == F ~ "sperm siRNA loci", #5 
    str_detect(id, "egg-zygote_complement_loci.") ~ "E loci - Z loci", #6 
    str_detect(id, "zygote_loci.") ~ "zygote siRNA loci" #7
  )) %>% 
  filter(is.na(locus) == F) %>% 
  mutate(context = case_when(
    str_detect(id, "gz.CG.") ~ "mCG", 
    str_detect(id, "gz.CHG.") ~ "mCHG", 
    str_detect(id, "gz.CHH.") ~ "mCHH"
  ))  

head(zy_meth_data_nice)
```
 
 

# stats
```{r}
zy_meth_data_nice %>% 
  group_by(locus, context) %>% 
  count()
```

```{r}
zy_meth_data_nice %>% 
  group_by(locus, context, sample_ID) %>% 
  count
```

## for each locus, average bioreps to the level of genotype
```{r}
zy_meth_data_s <- zy_meth_data_nice %>% 
 filter(is.na(X4) == F) %>% 
 group_by(X1, X2, X3, locus, context) %>% 
 summarise(meth = mean(X4)) %>% 
 ungroup()

head(zy_meth_data_s)
```
## summary stats - quantiles 
```{r}
zy_meth_data_ss <- zy_meth_data_s %>% 
  filter(is.na(meth) == F) %>% 
  group_by(locus, context) %>% 
  summarise(
    NN = n(),
    median = median(meth),
    mean = mean(meth),
    Q3 = quantile(meth, 0.75),
    Q1 = quantile(meth, 0.25),
    UL = quantile(meth, 0.975),
    LL = quantile(meth, 0.025)
  ) %>% 
  ungroup()

zy_meth_data_ss
```
```{r}
zy_meth_data_ss %>% 
  filter(context == "mCHH") %>% 
  arrange(mean)

write_excel_csv(zy_meth_data_ss, "zy_meth_data_ss.csv")
```
 


## linear models
```{r}
logit <- function(p){log(
  p / (1-p)
)}

logistic <- function(x){
  1/(1 + exp(-x))
}
```

```{r}
model_zy_meth <- glm(meth ~ context * locus, data = zy_meth_data_s, 
                     family = quasibinomial(link = "logit")) 

anova(model_zy_meth, test = "LRT")
```

```{r}
est_zy_meth1 <- emmeans(model_zy_meth, pairwise ~ locus | context) 
```

```{r}
zy_meth_results <- multcomp::cld(est_zy_meth1$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(grouping = str_remove_all(.group, " ")) 

zy_meth_results %>% 
  filter(context == "mCHH") %>% 
  select(locus, context, grouping)
```

```{r}
est_zy_meth1$contrasts %>% 
  as.data.frame() %>% 
  filter(context == "mCHH") %>% 
  select(contrast, p.value)
```
E-Z vs. Z-E = 0
Egg vs. zygote = 0 

# plot
## CHH 
```{r}
zy_meth_data_ss %>% 
  filter(context == "mCHH") %>%
  inner_join(zy_meth_results, by = c("locus", "context")) %>% 
  mutate(locus2 = case_when(
    str_detect(locus, "embryo siRNA loci") ~ "*embryo siRNA loci",
    T ~ locus
  )) %>% 
  mutate(locus = reorder(locus2, mean)) %>% 
  ggplot(aes(x = locus, y = Q3)) +
  #facet_grid(genotype ~., scales = "free", switch = "y") + 
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = locus, xend = locus,  y = Q3, yend = UL)) +
  geom_segment(aes(x = locus, xend = locus,  y = Q1, yend = LL)) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) +
  geom_text(aes(label = grouping, y = Q3), 
  hjust = -0.5, size = 4, fontface = "bold", vjust = -0.2) +
  labs(x = NULL, 
       y = "mCHH in WT zygote (Zhou et al.)") +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", colour = "black"),
    axis.text = element_text(colour = "black"), 
    axis.text.y = element_text(hjust = 0.5),
    strip.placement = "outside",
    strip.text.y = element_text(hjust = 0.5, angle = 0, face = "bold"), 
    panel.spacing = unit(1, "lines"),
    axis.line = element_line(size = 1),
    legend.position = "none"
  ) +
  coord_flip()  

ggsave("zy_meth.svg", height = 3.5, width = 6)
ggsave("zy_meth.png", height = 3.5, width = 6)
```
 


## CHG 
```{r}
zy_meth_data_ss %>% 
  filter(context == "mCHG") %>%
  filter(str_detect(locus, "intersect") == F) %>% 
  inner_join(zy_meth_results, by = c("locus", "context")) %>% 
  mutate(locus2 = case_when(
    str_detect(locus, "embryo siRNA loci") ~ "*embryo siRNA loci",
    T ~ locus
  )) %>% 
  mutate(locus = reorder(locus2, median)) %>% 
  # mutate(locus = factor(locus2, levels = c(
  #   "sperm siRNA loci",
  #   "zygote down loci",
  #   "zygote down loci -siren",
  #   "egg siRNA loci",
  #   "*embryo siRNA loci",
  #   "zygote up loci -siren",
  #   "zygote up loci",
  #   "NS loci",
  #   "zygote siRNA loci",
  #   "NS loci -siren",
  #   "seedling siRNA loci"
  # ))) %>% 
  ggplot(aes(x = locus, y = Q3)) +
  #facet_grid(genotype ~., scales = "free", switch = "y") + 
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = locus, xend = locus,  y = Q3, yend = UL)) +
  geom_segment(aes(x = locus, xend = locus,  y = Q1, yend = LL)) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) +
  geom_text(aes(label = grouping, y = Q3), vjust = -0.2,
   hjust = -0.5, size = 4, fontface = "bold") + 
  labs(x = NULL, 
       y = "mCHG in WT zygote (Zhou et al.)") +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", colour = "black"),
    axis.text = element_text(colour = "black"), 
    strip.placement = "outside",
    strip.text.y = element_text(hjust = 0.5, angle = 0, face = "bold"), 
    panel.spacing = unit(1, "lines"),
    axis.line = element_line(size = 1),
    legend.position = "none"
  ) +
  coord_flip()  

ggsave("zg_meth_CHG.svg", height = 3.5, width = 6)
ggsave("zg_meth_CHG.png", height = 3.5, width = 6)
```

## GG
```{r}
zy_meth_data_ss %>% 
  filter(context == "mCG") %>%
  filter(str_detect(locus, "intersect") == F) %>% 
  filter(locus != "endosperm siRNA loci") %>% 
  filter(locus != "egg siren loci") %>% 
  filter(str_detect(locus, "NOT|intersect") == F) %>% 
  inner_join(zy_meth_results, by = c("locus", "context")) %>% 
  mutate(locus2 = case_when(
    str_detect(locus, "embryo siRNA loci") ~ "*embryo siRNA loci",
    T ~ locus
  )) %>% 
  mutate(locus = reorder(locus2, median)) %>% 
  # mutate(locus = factor(locus2, levels = c(
  #   "sperm siRNA loci",
  #   "zygote down loci",
  #   "zygote down loci -siren",
  #   "egg siRNA loci",
  #   "*embryo siRNA loci",
  #   "zygote up loci -siren",
  #   "zygote up loci",
  #   "NS loci",
  #   "zygote siRNA loci",
  #   "NS loci -siren",
  #   "seedling siRNA loci"
  # ))) %>% 
  ggplot(aes(x = locus, y = Q3)) +
 # facet_grid(genotype ~., scales = "free", switch = "y") + 
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = locus, xend = locus,  y = Q3, yend = UL)) +
  geom_segment(aes(x = locus, xend = locus,  y = Q1, yend = LL)) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) +
  geom_text(aes(label = grouping, y = Q1),
   hjust = 1.5, size = 4, fontface = "bold", vjust = -0.3) + 
  labs(x = NULL, 
       y = "mCG in WT zygote (Zhou et al.)") +
  theme_minimal() +
  theme(
    text = element_text(size = 14, face = "bold", colour = "black"),
    axis.text = element_text(colour = "black"), 
    strip.placement = "outside",
    strip.text.y = element_text(hjust = 0.5, angle = 0, face = "bold"), 
    panel.spacing = unit(1, "lines"),
    axis.line = element_line(size = 1),
    legend.position = "none"
  ) +
  coord_flip()  

ggsave("zg_meth_CG.svg", height = 3.5, width = 6)
ggsave("zg_meth_CG.png", height = 3.5, width = 6)
```



 

