---
title: "sirenF_quantification"
author: "Chenxin Li"
date: "6/2/2021"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(emmeans)
library(lme4)
library(lmerTest)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(RColorBrewer)
library(viridis)
library(stringr)
library(svglite)
library(cowplot)
library(forcats)
```
# Load data 
```{r}
centro_bed <- read_table2("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/rice_centromere_genetic_mapped.bed", col_names = FALSE)

centro_bed <- centro_bed %>% 
  rename(Chr.c = X1, 
         start.c = X2,
         end.c = X3)
head(centro_bed, 12)
```
```{r}
sample_des <-  read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/sample_des_clvs12.xlsx") 

head(sample_des)
```

```{r}
lengths_summary_sirenF_sam <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/siRNA_length_counts_2021_04_16/lengths_summary.sirenF.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE) 

head(lengths_summary_sirenF_sam)
```

```{r message=F}
bin50_list <- list.files("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/50Kb_sirenF_coverages/", 
                             pattern = ".cov", full.names = T)

bin50_data <- sapply(bin50_list, read_delim, simplify = FALSE, col_names = F, delim = "\t") %>% 
  bind_rows(.id = "id")

head(bin50_data)
```

# key data
```{r}
identify_samples <- function(df){
  df %>% 
    mutate(sample_ID = case_when(
    str_detect(id, "sperm.1") ~ "BS1", #1
    str_detect(id, "Sperm2") ~ "BS2",  #2
    str_detect(id, "Sperm4a") ~ "BS4a",  #3
    str_detect(id, "sperm4b") ~ "BS4b",  #4
    str_detect(id, "sperm6a") ~ "BS6a",  #5
    str_detect(id, "sperm6b") ~ "BS6b",  #6
    str_detect(id, "sperm.5b") ~ "SP5b", #7
    str_detect(id, "sperm.5c") ~ "SP5c", #8
    str_detect(id, "Bulk.15") ~ "B15",   #9
    str_detect(id, "Bulk.20") ~ "B20",   #10
    str_detect(id, "Bulk.25") ~ "B25",   #11
    str_detect(id, "seedling4") ~ "SD4", #12
    str_detect(id, "egg.9a") ~ "E9a",    #13
    str_detect(id, "egg.9b") ~ "E9b",    #14
    str_detect(id, "egg.9c") ~ "E9c",    #15
    str_detect(id, "eggcell4") ~ "E4",   #16
    str_detect(id, "eggcell3") ~ "E3",   #17
    str_detect(id, "eggcell2") ~ "E2",   #18
    str_detect(id, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(id, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(id, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(id, "SRR57") ~ "root",              #22
    str_detect(id, "ab_control") ~ "ddm1_con",     #23
    str_detect(id, "ddm1ab") ~ "ddm1",             #24
    str_detect(id, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(id, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(id, "SRR2544786") ~ "em_dry", #27  
    str_detect(id, "SRR5049778") ~ "an_pre",  #28
    str_detect(id, "SRR5049779") ~ "an_mei",  #29
    str_detect(id, "SRR5049780") ~ "an_mic",  #30
    str_detect(id, "SRR5049781") ~ "an_bi",   #31
    str_detect(id, "SRR5049786") ~ "ov_1st",  #32
    str_detect(id, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(id, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(id, "SRR5049789") ~ "ov_4th",  #35
    str_detect(id, "S3_1") ~ "S3_1",          #36
    str_detect(id, "S3_2") ~ "S3_2",          #37
    str_detect(id, "S3_3") ~ "S3_3",          #38
    str_detect(id, "S5_1") ~ "S5_1",          #39
    str_detect(id, "S5_2") ~ "S5_2",          #40
    str_detect(id, "S5_3") ~ "S5_3",          #41
    str_detect(id, "S7_1") ~ "S7_1",          #42
    str_detect(id, "S7_2") ~ "S7_2",          #43
    str_detect(id, "S7_3") ~ "S7_3",          #44
    str_detect(id, "SRR2544787") ~ "em_12h",  #45
    str_detect(id, "SRR2544788") ~ "em_24h",  #46
    str_detect(id, "SRR771501") ~ "em_7DAF",  #47
    str_detect(id, "SRR771500") ~ "en_7DAF",  #48   
    str_detect(id, "SD7") ~ "SD7", #49
    str_detect(id, "zg1d") ~ "zg1d", #50
    str_detect(id, "zg1e") ~ "zg1e",  #51
    str_detect(id, "zg2d") ~ "zg2d",  #52
    str_detect(id, "zg2e") ~ "zg2e",  #53
    str_detect(id, "zg3a") ~ "zg3a", #54
    str_detect(id, "zg3b") ~ "zg3b" , #55
    str_detect(id, "OV9H_1") ~ "OV9_1" , #56
    str_detect(id, "OV9H_2") ~ "OV9_2" , #57
    str_detect(id, "OV9H_3") ~ "OV9_3" , #58
    str_detect(id, "SD8") ~ "SD8"  #59
    ))
} 
  
```

```{r}
identify_samples2 <- function(df){
  df %>% 
    mutate(sample_ID = case_when(
    str_detect(X1, "sperm.1") ~ "BS1", #1
    str_detect(X1, "Sperm2") ~ "BS2",  #2
    str_detect(X1, "Sperm4a") ~ "BS4a",  #3
    str_detect(X1, "sperm4b") ~ "BS4b",  #4
    str_detect(X1, "sperm6a") ~ "BS6a",  #5
    str_detect(X1, "sperm6b") ~ "BS6b",  #6
    str_detect(X1, "sperm.5b") ~ "SP5b", #7
    str_detect(X1, "sperm.5c") ~ "SP5c", #8
    str_detect(X1, "Bulk.15") ~ "B15",   #9
    str_detect(X1, "Bulk.20") ~ "B20",   #10
    str_detect(X1, "Bulk.25") ~ "B25",   #11
    str_detect(X1, "seedling4") ~ "SD4", #12
    str_detect(X1, "egg.9a") ~ "E9a",    #13
    str_detect(X1, "egg.9b") ~ "E9b",    #14
    str_detect(X1, "egg.9c") ~ "E9c",    #15
    str_detect(X1, "eggcell4") ~ "E4",   #16
    str_detect(X1, "eggcell3") ~ "E3",   #17
    str_detect(X1, "eggcell2") ~ "E2",   #18
    str_detect(X1, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(X1, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(X1, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(X1, "SRR57") ~ "root",              #22
    str_detect(X1, "ab_control") ~ "ddm1_con",     #23
    str_detect(X1, "ddm1ab") ~ "ddm1",             #24
    str_detect(X1, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(X1, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(X1, "SRR2544786") ~ "em_dry", #27  
    str_detect(X1, "SRR5049778") ~ "an_pre",  #28
    str_detect(X1, "SRR5049779") ~ "an_mei",  #29
    str_detect(X1, "SRR5049780") ~ "an_mic",  #30
    str_detect(X1, "SRR5049781") ~ "an_bi",   #31
    str_detect(X1, "SRR5049786") ~ "ov_1st",  #32
    str_detect(X1, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(X1, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(X1, "SRR5049789") ~ "ov_4th",  #35
    str_detect(X1, "S3_1") ~ "S3_1",          #36
    str_detect(X1, "S3_2") ~ "S3_2",          #37
    str_detect(X1, "S3_3") ~ "S3_3",          #38
    str_detect(X1, "S5_1") ~ "S5_1",          #39
    str_detect(X1, "S5_2") ~ "S5_2",          #40
    str_detect(X1, "S5_3") ~ "S5_3",          #41
    str_detect(X1, "S7_1") ~ "S7_1",          #42
    str_detect(X1, "S7_2") ~ "S7_2",          #43
    str_detect(X1, "S7_3") ~ "S7_3",          #44
    str_detect(X1, "SRR2544787") ~ "em_12h",  #45
    str_detect(X1, "SRR2544788") ~ "em_24h",  #46
    str_detect(X1, "SRR771501") ~ "em_7DAF",  #47
    str_detect(X1, "SRR771500") ~ "en_7DAF",  #48   
    str_detect(X1, "SD7") ~ "SD7", #49
    str_detect(X1, "zg1d") ~ "zg1d", #50
    str_detect(X1, "zg1e") ~ "zg1e",  #51
    str_detect(X1, "zg2d") ~ "zg2d",  #52
    str_detect(X1, "zg2e") ~ "zg2e",  #53
    str_detect(X1, "zg3a") ~ "zg3a", #54
    str_detect(X1, "zg3b") ~ "zg3b" , #55
    str_detect(X1, "OV9H_1") ~ "OV9_1" , #56
    str_detect(X1, "OV9H_2") ~ "OV9_2" , #57
    str_detect(X1, "OV9H_3") ~ "OV9_3" , #58
    str_detect(X1, "SD8") ~ "SD8"  #59
    ))
} 
```

```{r}
siren_counts <- lengths_summary_sirenF_sam %>% 
  identify_samples2() %>% 
  select(-X1) %>% 
  select(sample_ID, `24`) %>% 
  inner_join(sample_des, by = "sample_ID") %>% 
  filter(sample_type == "egg"|
           sample_type == "embryo 7-8 DAF" |
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap")

head(siren_counts)
```

```{r}
bin50_data_clean <- bin50_data %>%  
  identify_samples() %>% 
  select(X1, X4, X5, X9, sample_ID) %>% 
  inner_join(siren_counts, by = "sample_ID") %>% 
  mutate(RPM = X9/`24` * 10^6) 
  

head(bin50_data_clean)
```

```{r}
bin50_data_clean_s <- bin50_data_clean %>% 
  group_by(sample_type, X1, X4, X5) %>% 
  summarise(RPM = mean(RPM)) %>%
  ungroup() %>% 
  group_by(sample_type) %>% 
  mutate(z.score = (RPM - mean(RPM))/sd(RPM)) %>% 
  ungroup()

head(bin50_data_clean_s)
```

# stat 
```{r}
centro_bed %>% 
  mutate(Length = end.c - start.c) %>% 
  mutate(number.bins = Length/50/1000)

bin50_data_clean_s_cent <- bin50_data_clean_s %>% 
  inner_join(centro_bed, by = c("X1" = "Chr.c")) %>% 
  filter( 
    X4 > start.c & X5 < end.c 
  ) %>% 
  mutate(tag = paste(X1, X4, X5, sep = "-")) 

head(bin50_data_clean_s_cent)

bin50_data_clean_s_cent %>% 
  group_by(X1, sample_type) %>% 
  count()
```
```{r}
bin50_data_clean_cent <- bin50_data_clean %>% 
  inner_join(centro_bed, by = c("X1" = "Chr.c")) %>% 
  filter( 
    X4 > start.c & X5 < end.c 
  ) %>% 
  group_by(sample_ID, `24`) %>% 
  summarise(reads = sum(X9)) %>% 
  mutate(pro_24 = reads/`24` * 100)

bin50_data_clean_cent
```


## linear model
```{r}
model_cent <- lm(log(RPM + 1) ~ X1 + sample_type, data = bin50_data_clean_s_cent %>% 
                   filter(str_detect(sample_type, "ovary")==F) %>% 
                   mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "embryo") ~ "*embryo", 
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)",
    T ~ sample_type
  )))
#plot(model_cent)
anova(model_cent)
```

```{r}
est_cent <- emmeans(model_cent, pairwise ~ sample_type)
cent_results <- multcomp::cld(est_cent$emmeans, Letters = letters) %>%
  as.data.frame()  %>% 
  mutate(grouping = str_remove_all(.group, " ")) %>% 
  cbind(text.y = c(1.2, 1.3, 1.3, 1.6, 1.7))


cent_results
```

```{r}
est_cent$contrasts %>% 
  as.data.frame() %>% 
  select(contrast, p.value)
```

* egg vs. zygote = 0
* zygote vs. embryo = 0.5

# plot 
```{r}
bin50_data_clean_s_cent %>% 
                   filter(str_detect(sample_type, "ovary")==F) %>% 
                   mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "embryo") ~ "*embryo", 
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)",
    T ~ sample_type
  )) %>% 
  mutate(sample_type = forcats::fct_reorder(.f = sample_type, .x = RPM, .fun = median)) %>% 
  ggplot(aes(x = sample_type, y = log10(RPM + 1))) +
  #ggplot(aes(x = sample_type, y = RPM)) +
  ggbeeswarm::geom_quasirandom(color = "white", shape = 21, alpha = 0.8,
                              aes(fill = sample_type)) + 
  #geom_point(position = position_jitter()) +
  #geom_boxplot(width = 0.3, outlier.shape = NA, fill = NA) +
  geom_text(data = cent_results, aes(label = grouping, y = text.y), size = 3.5, fontface = "bold") +
  annotate(geom = "errorbarh", xmin = 2, xmax = 4, height = 0.05, y = 3) +
  annotate(geom = "text", x = 3, hjust = -0.1, y = 3, label = "P < 2.2e-16", size = 3.5, fontface = "bold") +
  annotate(geom = "errorbarh", xmin = 2, xmax = 3, height = 0.05, y = 2.5) +
  annotate(geom = "text", x = 2.5, hjust = -0.07, y = 2.5, label = "P = 0.5", 
           size = 3.5, fontface = "bold") +
  scale_fill_manual(values = c("seagreen", "violetred4", "grey20", "tomato1",  "dodgerblue2"),
                    limits = c("seedling", "zygote", "*embryo", "egg", "sperm")) +
  labs(x = "siRNA transcriptome", 
       y = "log10(RPM + 1)\ncentromeric non-siren 24-nt siRNA\nper million non-siren 24-nt siRNAs")+
  theme_minimal() +
  theme(legend.position = "none",
        text = element_text(size = 14, color = "black", face = "bold"),
        axis.text = element_text(color = "black"),
        axis.text.y = element_text(color = "black", hjust = 0.5),
        strip.placement = "outside",
        strip.text.y.left = element_text(hjust = 0.5, angle = 0),
        axis.line = element_line(size = 1)) +
  coord_flip() 

ggsave("non-siren_24nt_at_centromere.svg", height = 4, width = 6)
ggsave("non-siren_24nt_at_centromere.png", height = 4, width = 6)
```

```{r}
bin50_data_clean_s_cent %>% 
                   filter(str_detect(sample_type, "ovary")==F) %>% 
                   mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "embryo") ~ "*embryo", 
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)",
    T ~ sample_type
  )) %>% 
  mutate(sample_type = forcats::fct_reorder(.f = sample_type, .x = RPM, .fun = median)) %>% 
  ggplot(aes(x = sample_type, y = log10(RPM + 1))) +
  facet_wrap(~X1) +
  ggbeeswarm::geom_quasirandom(color = "white", shape = 21, alpha = 0.8,
                               aes(fill = sample_type)) + 
  scale_fill_manual(values = c("seagreen", "violetred4", "grey20", "tomato1",  "dodgerblue2"),
                    limits = c("seedling", "zygote", "*embryo", "egg", "sperm")) +
  labs(x = "siRNA transcriptome", 
       y = "log10(RPM + 1)\ncentromeric non-siren 24-nt siRNA\nper million non-siren 24-nt siRNAs")+
  theme_minimal() +
  theme(legend.position = "none",
        text = element_text(size = 14, color = "black", face = "bold"),
        axis.text = element_text(color = "black"),
        axis.text.y = element_text(color = "black", hjust = 0.5),
        strip.placement = "outside",
        strip.text.y.left = element_text(hjust = 0.5, angle = 0),
        axis.line = element_line(size = 1),
        panel.spacing = unit(1, "lines")) +
  coord_flip() 

ggsave("non-siren_24nt_at_centromere_chr.svg", height = 6, width = 7)
ggsave("non-siren_24nt_at_centromere_chr.png", height = 6, width = 7)
```

```{r}
bin50_data_clean_s_cent %>% 
                   filter(str_detect(sample_type, "ovary")==F) %>% 
                   mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "embryo") ~ "*embryo", 
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)",
    T ~ sample_type
  )) %>% 
  mutate(sample_type = forcats::fct_reorder(.f = sample_type, .x = RPM, .fun = median)) %>% 
  group_by(sample_type, X1) %>% 
  summarise(sum = sum(RPM)) %>% 
  ungroup() %>% 
  ggplot(aes(x = sample_type, y = log10(sum))) +
  geom_line(aes(group = X1, color = as.factor(X1)), size = 1.2) + 
  geom_point(color = "white", shape = 21, alpha = 0.8, size = 3,
                               aes(fill = sample_type)) + 
  scale_fill_manual(values = c("seagreen", "violetred4", "grey20", "tomato1",  "dodgerblue2"),
                    limits = c("seedling", "zygote", "*embryo", "egg", "sperm")) +
  labs(x = "siRNA transcriptome", 
       y = "log10(RPM + 1)\ncentromeric non-siren 24-nt siRNA\nper million non-siren 24-nt siRNAs",
       color= "Chr")+
  guides(fill = F) +
  theme_minimal() +
  theme(legend.position = "top",
        text = element_text(size = 14, color = "black", face = "bold"),
        axis.text = element_text(color = "black"),
        axis.text.y = element_text(color = "black", hjust = 0.5),
        strip.placement = "outside",
        strip.text.y.left = element_text(hjust = 0.5, angle = 0),
        axis.line = element_line(size = 1)) +
  coord_flip() 
```


# PCA
```{r}
bin50_wide <- bin50_data_clean %>% 
  mutate(logRPM = log10(RPM+1)) %>% 
  select(X1, X4, X5, logRPM, sample_ID) %>% 
  spread(sample_ID, logRPM)

head(bin50_wide)
```

```{r}
pc <- prcomp(t(bin50_wide[, 4:ncol(bin50_wide)]))
pc_importance <- as.data.frame(t(summary(pc)$importance))
pc_importance 
```
```{r}
PC_cent <- pc$x %>% 
  as.data.frame() %>% 
  mutate(sample_ID = row.names(.)) %>% 
  inner_join(sample_des, by = "sample_ID") %>% 
   mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "embryo") ~ "*embryo", 
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)",
    T ~ sample_type
  )) %>% 
  inner_join(bin50_data_clean_cent, by = "sample_ID") 

head(PC_cent)

cor.test(PC_cent$PC1, PC_cent$pro_24)
cor.test(PC_cent$PC1, PC_cent$pro_24)$p.value
```

```{r}
PC_cent %>% 
  ggplot(aes(x = PC1, y = pro_24)) +
  geom_point(aes(fill = sample_type), size = 3, alpha = 0.8, shape = 21, color = "white") +
  geom_smooth(method = "lm") +
  annotate(geom = "text", label = "r = 0.96\nP = 2e-19", x = -Inf, y = Inf,
           size = 3.5, hjust = -0.5, vjust = 1.5, fontface = "bold") +
  scale_fill_manual(values = c("seagreen", "violetred4", "grey20", "tomato1",  "dodgerblue2"),
                    limits = c("seedling", "zygote", "*embryo", "egg", "sperm")) +
  labs(x = paste("PC1 (", pc_importance[1, 2] %>% signif(3)*100, "% of Variance)", sep = ""), 
       y = "centromeric non-siren 24-nt siRNA\n(% of total non-siren 24-nt siRNA)",
       fill = NULL,
       color = NULL) +
  theme_minimal() +
  theme(legend.position = c(0.8, 0.2)) +
  theme(axis.line = element_line(size = 1)) +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

```

* PC1 is centromeric region siRNA relative abundance

```{r}
PC_cent %>% 
  ggplot(aes(x = PC1, y = PC2)) +
  geom_point(aes(fill = sample_type), size = 3, alpha = 0.8, shape = 21, color = "white") +
  scale_fill_manual(values = c("seagreen", "violetred4", "grey20", "tomato1",  "dodgerblue2"),
                    limits = c("seedling", "zygote", "*embryo", "egg", "sperm")) +
  labs(x = paste("PC1 (", pc_importance[1, 2] %>% signif(3)*100, "% of Variance)", sep = ""), 
       y = paste("PC2 (", pc_importance[2, 2] %>% signif(3)*100, "% of Variance)", sep = ""),
       fill = NULL,
       color = NULL) +
  theme_minimal() +
  theme(legend.position = c(0.8, 0.2)) +
  theme(axis.line = element_line(size = 1)) +
  theme(text = element_text(size= 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 
```
* PC2 is just noise. 
