---
title: "siRNA_metagene_CLVS13_sirenF"
author: "Chenxin Li"
date: "03/30/2021"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(RColorBrewer)
library(viridis)
library(stringr)
library(svglite)
library(cowplot)
library(forcats)
```

# read data 

blender_sperm_1_S30_L002_R1_001_downstream <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/metagene_coverage_sirenF/sirenF/output/blender-sperm-1_S30_L002_R1_001___downstream.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

```{r message=F, warning=F}
meta_list <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/metagene_coverage_sirenF/sirenF/output/", 
                        pattern = "*.txt", full.names = T)

meta_data <- sapply(meta_list, read_delim, simplify = FALSE, delim = "\t") %>% 
  bind_rows(.id = "id")

head(meta_data)
```
## sample des, library size 
```{r}
sample_des <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/sample_des_clvs12.xlsx")

smRNA_comp <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/codes_and_output/smRNA_comp.csv")
```
```{r}
#this is without siren reads
total_siRNA_length_sirenF <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS13/LengthCounts/lengths_summary.sirenF.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
 
# keying data and normalization 
```{r}
identify_samples <- function(df){
  df %>% 
    mutate(sample_ID = case_when(
    str_detect(id, "sperm.1") ~ "BS1", #1
    str_detect(id, "Sperm2") ~ "BS2",  #2
    str_detect(id, "Sperm4a") ~ "BS4a",  #3
    str_detect(id, "sperm4b") ~ "BS4b",  #4
    str_detect(id, "sperm6a") ~ "BS6a",  #5
    str_detect(id, "sperm6b") ~ "BS6b",  #6
    str_detect(id, "sperm.5b") ~ "SP5b", #7
    str_detect(id, "sperm.5c") ~ "SP5c", #8
    str_detect(id, "Bulk.15") ~ "B15",   #9
    str_detect(id, "Bulk.20") ~ "B20",   #10
    str_detect(id, "Bulk.25") ~ "B25",   #11
    str_detect(id, "seedling4") ~ "SD4", #12
    str_detect(id, "egg.9a") ~ "E9a",    #13
    str_detect(id, "egg.9b") ~ "E9b",    #14
    str_detect(id, "egg.9c") ~ "E9c",    #15
    str_detect(id, "eggcell4") ~ "E4",   #16
    str_detect(id, "eggcell3") ~ "E3",   #17
    str_detect(id, "eggcell2") ~ "E2",   #18
    str_detect(id, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(id, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(id, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(id, "SRR57") ~ "root",              #22
    str_detect(id, "ab_control") ~ "ddm1_con",     #23
    str_detect(id, "ddm1ab") ~ "ddm1",             #24
    str_detect(id, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(id, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(id, "SRR2544786") ~ "em_dry", #27  
    str_detect(id, "SRR5049778") ~ "an_pre",  #28
    str_detect(id, "SRR5049779") ~ "an_mei",  #29
    str_detect(id, "SRR5049780") ~ "an_mic",  #30
    str_detect(id, "SRR5049781") ~ "an_bi",   #31
    str_detect(id, "SRR5049786") ~ "ov_1st",  #32
    str_detect(id, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(id, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(id, "SRR5049789") ~ "ov_4th",  #35
    str_detect(id, "S3_1") ~ "S3_1",          #36
    str_detect(id, "S3_2") ~ "S3_2",          #37
    str_detect(id, "S3_3") ~ "S3_3",          #38
    str_detect(id, "S5_1") ~ "S5_1",          #39
    str_detect(id, "S5_2") ~ "S5_2",          #40
    str_detect(id, "S5_3") ~ "S5_3",          #41
    str_detect(id, "S7_1") ~ "S7_1",          #42
    str_detect(id, "S7_2") ~ "S7_2",          #43
    str_detect(id, "S7_3") ~ "S7_3",          #44
    str_detect(id, "SRR2544787") ~ "em_12h",  #45
    str_detect(id, "SRR2544788") ~ "em_24h",  #46
    str_detect(id, "SRR771501") ~ "em_7DAF",  #47
    str_detect(id, "SRR771500") ~ "en_7DAF",  #48   
    str_detect(id, "SD7") ~ "SD7", #49
    str_detect(id, "zg1d") ~ "zg1d", #50
    str_detect(id, "zg1e") ~ "zg1e",  #51
    str_detect(id, "zg2d") ~ "zg2d",  #52
    str_detect(id, "zg2e") ~ "zg2e",  #53
    str_detect(id, "zg3a") ~ "zg3a", #54
    str_detect(id, "zg3b") ~ "zg3b" , #55
    str_detect(id, "OV9H_1") ~ "OV9_1" , #56
    str_detect(id, "OV9H_2") ~ "OV9_2" , #57
    str_detect(id, "OV9H_3") ~ "OV9_3" , #58
    str_detect(id, "SD8") ~ "SD8"  #59
    ))
}
```

```{r}
sample_des_size <- smRNA_comp %>% 
  inner_join(sample_des, by = "sample_ID") 

total_siRNA_length_sirenF <- total_siRNA_length_sirenF %>% 
  rename(id = X1) %>% 
  identify_samples() %>% 
  filter(is.na(sample_ID) == F)
   
head(sample_des_size)
head(total_siRNA_length_sirenF)
```

```{r}
meta_data <- meta_data %>% 
  mutate(location = case_when(
    str_detect(id, "upstream") ~ "5' end",
    str_detect(id, "downstream") ~ "3' end"
  )) %>% 
  mutate(location = factor(location, levels = c("5' end", "3' end"))) %>% 
  identify_samples() %>% 
  inner_join(sample_des_size, by = "sample_ID") %>% 
  inner_join(total_siRNA_length_sirenF, by = "sample_ID") %>% 
  mutate(pro_24 = `total overlap`/ `24`)  
  
head(meta_data)
```

# stat
## finding the peak for seedling and zygote 
```{r}
meta_data %>% 
  filter(sample_type == "seedling shoot") %>% 
  filter(location == "5' end") %>% 
  group_by(`bin start`) %>% 
  summarise(pro_24 = mean(pro_24)) %>% 
  arrange(-pro_24) %>% 
  head() %>% 
  select(`bin start`, pro_24)

meta_data %>% 
  filter(sample_type == "zygote") %>% 
  filter(location == "5' end") %>% 
  group_by(`bin start`) %>% 
  summarise(pro_24 = mean(pro_24)) %>% 
  arrange(-pro_24) %>% 
  head() %>% 
  select(`bin start`, pro_24)
```
Seedling peaks at -300, 
Zygote peaks at -400.  

```{r}
stat_data <- meta_data %>% 
  filter(location == "5' end") %>%
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap") %>% 
  filter(`bin start` == -300) %>% 
  mutate(bin.f = as.factor(`bin start`))
```

```{r}
logit <- function(p){log(
  p / (1-p)
)}

logistic <- function(x){
  1/(1 + exp(-x))
}
```


```{r}
model_meta <- lm(logit(pro_24) ~ sample_type, data = stat_data)
anova(model_meta) 
```

```{r}
est_meta <- emmeans(model_meta, pairwise ~ sample_type)
multcomp::cld(est_meta$emmeans, Letters = letters) 
```
```{r}
anova_result <- multcomp::cld(est_meta$emmeans, Letters = letters) %>% 
  as.data.frame() %>% 
  mutate(grouping = str_remove_all(.group, " ")) %>% 
  mutate(location = "5' end") %>%
  mutate(location = factor(location)) %>% 
  mutate(sample_type = as.character(sample_type)) %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" , 
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) 

head(anova_result)
```

```{r}
stat_data %>% 
  group_by(sample_type) %>% 
  summarise(mean = mean(pro_24))
```

* egg vs. zygote: 451 - 347: 30%
* OV9 vs. OV0: 521 vs 437: 20%  
   

```{r}
est_meta$contrasts %>% 
  as.data.frame() 
```


zygote - egg: 4e-4
ovary 0 - ovary 9 = 0.118 

# plot 
```{r}
axis_line <- data.frame(
  `bin start` = c(-Inf),
  location = c("5' end")
) %>%  
  mutate(location = factor(location, levels = c("5' end", "3' end"))) 

axis_line
```

```{r}
meta_data %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote") %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    T ~ sample_type
  )) %>% 
  ggplot(aes(x = `bin start`, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = 0, color = NA) +
  stat_summary(geom = "ribbon",  fun.data = "mean_cl_boot", aes(fill = sample_type), alpha = 0.3) +
  stat_summary(geom = "line", fun =  mean, aes(group = sample_type, color = sample_type), size = 1.25, alpha = 0.8) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred4"),
                    limits = c("egg", "seedling", "sperm", "zygote")) +
  scale_fill_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred4"),
                    limits = c("egg", "seedling", "sperm", "zygote")) +
  scale_y_continuous(breaks = c(0, 200, 400, 600)) +
  guides(fill = F) + 
  labs(color = NULL,
       x = NULL, 
       y = "24-nt siRNA coverage") + 
  theme_minimal()+
  theme(legend.position = c(0.78, 0.8)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black")) +
  theme(panel.spacing.x = unit(1, "lines"))
```

```{r}
meta_data %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap") %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" , 
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) %>% 
  mutate(location = factor(location, levels = c("5' end", "3' end"))) %>% 
  ggplot(aes(x = `bin start`, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = 0, color = NA) +
  stat_summary(geom = "line", fun =  mean, aes(group = sample_type, color = sample_type), size = 1.25, alpha = 0.8) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2", "coral4", "violetred4"),
                    limits = c("ovary (0hr)",  "egg", "seedling", "sperm", "ovary (9hap)", "zygote")) +
  scale_y_continuous(breaks = c(0, 200, 400, 600)) +
  guides(fill = F) + 
  labs(color = NULL,
       x = NULL, 
       y = "24-nt siRNA coverage\n(per 1000 non-siren 24-nt siRNA)") + 
  theme_minimal()+
  theme(legend.position = c(0.8, 0.8)) +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black")) +
  theme(panel.spacing.x = unit(0.5, "lines"))

ggsave(filename = "24_meta_gamate_ov_sirenF.svg", height = 4, width = 4.9)
ggsave(filename = "24_meta_gamate_ov_sirenF.png", height = 4, width = 4.9)
```
```{r}
meta_data %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote" |
           sample_type == "ovary" |
           sample_type == "ovary_9hap") %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    str_detect(sample_type, "ovary_9hap") ~ "ovary (9hap)" ,
    str_detect(sample_type, "ovary") & 
      str_detect(sample_type, "9hap") == F ~ "ovary (0hr)" , 
    str_detect(sample_type, "embryo") ~ "embryo",
    T ~ sample_type
  )) %>% 
  filter(location == "5' end") %>% 
  filter(`bin start` == -300) %>% 
  group_by(sample_type, sample_ID) %>%
  summarise(
    pro_24 = mean(pro_24)
  ) %>% 
  ungroup() %>% 
  mutate(sample_type = fct_reorder(.f = sample_type, .x = pro_24, .fun = max)) %>% 
  ggplot(aes(x = sample_type, y = pro_24 * 1000)) +
  geom_vline(xintercept = -Inf, size = 1.5) + 
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = 0, color = NA) +
  geom_bar(stat = "summary", fun =  mean, aes(fill = sample_type), alpha = 0.8, width =0.7) +
  geom_point(aes(fill = sample_type), color = "white", shape = 21, alpha = 0.8, 
             position = position_jitter(0.1, seed = 666), size = 2) +
  geom_text(data = anova_result %>% 
              mutate(location == "5' end"), aes(label = grouping, y = logistic(emmean) * 600), size = 3.5, 
            fontface = "bold") +
  annotate(geom = "errorbarh", xmin = 2.1, xmax = 3.9, y = 500, 
            size = 1, height = 10) +
  annotate(geom = "errorbarh", xmin = 3.1, xmax = 4.9, y = 600, 
           size = 1, height = 10) +
  annotate(geom = "text", x = 4, y= 600, 
           label = "NS", size = 3.5, fontface = "bold", hjust = -0.2) +
  annotate(geom = "text", x = 3, y= 500, 
           label = "4e-4", size = 3.5, fontface = "bold", hjust = -0.1) +
  scale_fill_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2",  "coral4", "violetred4", "grey20"),
                    limits = c("ovary (0hr)",  "egg", "seedling", "sperm", "ovary (9hap)", "zygote", "embryo")) +
  scale_y_continuous(breaks = c(0, 200, 400, 600)) +
  labs(color = NULL,
       fill = NULL,
       x = NULL, 
       y = "mean 24-nt siRNA coverage\nat metagene curve peaks\n(evaluated at -300-bp to TSS,\nper 1000 non-siren 24-nt siRNA)") + 
  theme_minimal()+
  theme(legend.position = "none") +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black", hjust = 0.5)) +
  theme(panel.grid.major.y = element_blank()) +
  theme(panel.grid.minor.y = element_blank()) +
  coord_flip()

ggsave(filename = "24_meta_gamate_ov_bar_sirenF.svg", height = 3.8, width = 5)
ggsave(filename = "24_meta_gamate_ov_bar_sirenF.png", height = 3.8, width = 5)
```
