---
title: "siRNA lengths and TE abundances"
author: "Chenxin Li"
date: "Sep 03, 2019"
output:
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(stringr)
library(RColorBrewer)
library(svglite)
```

#smRNA composition 
```{r}
sample_des <-  read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/sample_des_clvs12.xlsx")  

smRNA_comp <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/codes_and_output/smRNA_comp.csv")
```


##overview of smRNA composition
```{r}
smRNA_comp %>% 
  full_join(sample_des, by = "sample_ID") %>% 
  select(-`minus mRNAs`, -`minus tRNAs`, -`minus NOR RNAs`, -`minus 5S RNAs`, -`minus 21nt phasi loci RNAs`) 
  
```

```{r}
smRNA_comp_L <- smRNA_comp %>%
  full_join(sample_des, by = "sample_ID") %>% 
  select(-`minus mRNAs`, -`minus tRNAs`, -`minus NOR RNAs`, -`minus 5S RNAs`, -`minus 21nt phasi loci RNAs`) %>% 
  gather("RNA_type", "count", 2:8) %>% 
  mutate(pro = count / `total read`) %>% 
  select(-count, -library) %>% 
  select(sample_ID, sample_type, genotype, pro, RNA_type) %>% 
  mutate(RNA_type1 = case_when(
    RNA_type == "general siRNA" ~ "general siRNA",
    RNA_type == "miRNA" ~ "miRNAs",
    str_detect(RNA_type, "tRNA") ~ "tRNAs", 
    str_detect(RNA_type, "NOR") ~ "NOR RNAs", 
    str_detect(RNA_type, "5S") ~ "5S rRNAs",
    str_detect(RNA_type, "21nt") ~ "phasiRNAs (21nt)",
    str_detect(RNA_type, "24nt") ~ "phasiRNAs (24nt)"
  )) %>% 
  as.data.frame() 


smRNA_comp_L
```

```{r}
write_excel_csv(smRNA_comp_L, "smRNA_composition.csv")
```

```{r}
smRNA_comp_L %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "zygote") %>%
  filter(genotype == "WT") %>% 
  filter(RNA_type != "general siRNA") %>% 
  mutate(RNA_type2 = factor(RNA_type1, levels = c(
    "miRNAs", "phasiRNAs (24nt)", "phasiRNAs (21nt)", "tRNAs", "NOR RNAs", "5S rRNAs"
  ))) %>% 
  mutate(sample_type.f = factor(sample_type, levels = c(
       "egg", "seedling shoot", "sperm", "zygote"
  ))) %>%
  ggplot(aes(x = sample_type.f, y = pro * 100)) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_bar(stat = "summary", aes(fill = sample_type), fun.y = mean, alpha = 0.8) +
  geom_point(color = "white", shape = 21, aes(fill = sample_type), alpha = 0.8, position = position_jitter(0.1, seed = 666)) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(), position = position_dodge(0.01), width = 0.3) +
  facet_wrap( ~ RNA_type2,  scales = "free_y", ncol = 2) +
  scale_fill_manual(values = c("tomato1", "seagreen",  "dodgerblue2", "violetred2"),
                      limits = c("egg", "seedling shoot", "sperm", "zygote")) + 
  guides(fill = guide_legend(ncol = 2)) + 
  labs(x = NULL, 
       y = "% small RNA",
       fill = NULL) +
  theme_minimal() +
  theme(legend.position = "bottom", legend.direction = "horizontal") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_blank()) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave("smRNA_comp_overall_essz.svg", width = 4.9, height = 6)
ggsave("smRNA_comp_overall_essz.png", width = 4.9, height = 6)
```



#length profile
```{r}
total_siRNA_length1 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/length_profiel_fig2/lengths_summary.bam.siRNAs.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

unique_siRNA_length1 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/length_profiel_fig2/lengths_summary.bam.q20.siRNAs.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

cento_siRNA_length1 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/length_profiel_fig2/lengths_summary.CentO.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

tir_siRNA_length1 <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/length_profiel_fig2/lengths_summary.TIR.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE) 

gypsy_siRNA_length1 <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/length_profiel_fig2/lengths_summary.Gypsy.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE) 
```

```{r}
CLVS12_cento <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.bam.CentO.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

CLVS12_gypsy <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.bam.GYPSY.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

CLVS12_tir <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.bam.TIR.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

CLVS12_total <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.siRNAs.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

CLVS12_unique <- CLVS12_total %>% 
  filter(str_detect(X1, "q20"))

CLVS12_total <- CLVS12_total %>% 
  filter(str_detect(X1, "q20") == F)
```
```{r}
total_siRNA_length <- rbind(total_siRNA_length1, CLVS12_total)
unique_siRNA_length <- rbind(unique_siRNA_length1, CLVS12_unique)
cento_siRNA_length <- rbind(cento_siRNA_length1, CLVS12_cento)
gypsy_siRNA_length <- rbind(gypsy_siRNA_length1, CLVS12_gypsy)
tir_siRNA_length <- rbind(tir_siRNA_length1, CLVS12_tir)
```


```{r}
colnames(total_siRNA_length) <- c("library", "20", "21", "22", "23", "24", "25")
colnames(unique_siRNA_length) <- c("library", "20", "21", "22", "23", "24", "25")
colnames(cento_siRNA_length) <- c("library", "20", "21", "22", "23", "24", "25")
colnames(tir_siRNA_length) <- c("library", "20", "21", "22", "23", "24", "25")
colnames(gypsy_siRNA_length) <- c("library", "20", "21", "22", "23", "24", "25")
```

```{r}
sample_des_size <- smRNA_comp %>% 
  full_join(sample_des, by = c("sample_ID")) 

sample_des_size
```
```{r}
head(total_siRNA_length)
```

```{r}
norm_tidy <- function(df){
  df %>% 
    mutate(sample_ID = case_when(
    str_detect(library, "sperm.1") ~ "BS1", #1
    str_detect(library, "Sperm2") ~ "BS2",  #2
    str_detect(library, "Sperm4a") ~ "BS4a",  #3
    str_detect(library, "sperm4b") ~ "BS4b",  #4
    str_detect(library, "sperm6a") ~ "BS6a",  #5
    str_detect(library, "sperm6b") ~ "BS6b",  #6
    str_detect(library, "sperm.5b") ~ "SP5b", #7
    str_detect(library, "sperm.5c") ~ "SP5c", #8
    str_detect(library, "Bulk.15") ~ "B15",   #9
    str_detect(library, "Bulk.20") ~ "B20",   #10
    str_detect(library, "Bulk.25") ~ "B25",   #11
    str_detect(library, "seedling4") ~ "SD4", #12
    str_detect(library, "egg.9a") ~ "E9a",    #13
    str_detect(library, "egg.9b") ~ "E9b",    #14
    str_detect(library, "egg.9c") ~ "E9c",    #15
    str_detect(library, "eggcell4") ~ "E4",   #16
    str_detect(library, "eggcell3") ~ "E3",   #17
    str_detect(library, "eggcell2") ~ "E2",   #18
    str_detect(library, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(library, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(library, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(library, "SRR57") ~ "root",              #22
    str_detect(library, "ab_control") ~ "ddm1_con",     #23
    str_detect(library, "ddm1ab") ~ "ddm1",             #24
    str_detect(library, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(library, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(library, "SRR2544786") ~ "em_dry", #27  
    str_detect(library, "SRR5049778") ~ "an_pre",  #28
    str_detect(library, "SRR5049779") ~ "an_mei",  #29
    str_detect(library, "SRR5049780") ~ "an_mic",  #30
    str_detect(library, "SRR5049781") ~ "an_bi",   #31
    str_detect(library, "SRR5049786") ~ "ov_1st",  #32
    str_detect(library, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(library, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(library, "SRR5049789") ~ "ov_4th",  #35
    str_detect(library, "S3_1") ~ "S3_1",          #36
    str_detect(library, "S3_2") ~ "S3_2",          #37
    str_detect(library, "S3_3") ~ "S3_3",          #38
    str_detect(library, "S5_1") ~ "S5_1",          #39
    str_detect(library, "S5_2") ~ "S5_2",          #40
    str_detect(library, "S5_3") ~ "S5_3",          #41
    str_detect(library, "S7_1") ~ "S7_1",          #42
    str_detect(library, "S7_2") ~ "S7_2",          #43
    str_detect(library, "S7_3") ~ "S7_3",          #44
    str_detect(library, "SRR2544787") ~ "em_12h",  #45
    str_detect(library, "SRR2544788") ~ "em_24h",  #46
    str_detect(library, "SRR771501") ~ "em_7DAF",  #47
    str_detect(library, "SRR771500") ~ "en_7DAF",  #48
    str_detect(library, "SD7") ~ "SD7", #49
    str_detect(library, "zg1d") ~ "zg1d", #50
    str_detect(library, "zg1e") ~ "zg1e",  #51
    str_detect(library, "zg2d") ~ "zg2d",  #52
    str_detect(library, "zg2e") ~ "zg2e",  #53
    str_detect(library, "zg3a") ~ "zg3a", #54
    str_detect(library, "zg3b") ~ "zg3b"  #55
  )) %>% 
    full_join(sample_des_size %>% 
                 select(`general siRNA`, sample_ID, sample_type, genotype), by = "sample_ID") %>% 
    gather("length", "count", 2:7) %>% 
    mutate(proportion = count / `general siRNA`) %>% 
    select(-count) 
}
```

```{r}
total_siRNA_length1 <- norm_tidy(total_siRNA_length) %>% 
  mutate(profile = rep("total siRNA")) 
unique_siRNA_length1 <- norm_tidy(unique_siRNA_length) %>% 
  mutate(profile = rep("uniquely mapped")) 
cento_siRNA_length1 <- norm_tidy(cento_siRNA_length)%>% 
  mutate(profile = rep("CentO")) 
gypsy_siRNA_length1 <- norm_tidy(gypsy_siRNA_length)%>% 
  mutate(profile = rep("Gypsy")) 
tir_siRNA_length1 <- norm_tidy(tir_siRNA_length)%>% 
  mutate(profile = rep("TIR")) 
```

```{r}
siRNA_length <- rbind(
  total_siRNA_length1,
  unique_siRNA_length1,
  cento_siRNA_length1,
  gypsy_siRNA_length1,
  tir_siRNA_length1
)

siRNA_length %>% head(6)
```
```{r}
write_excel_csv(siRNA_length, "siRNA_length.csv")
```



```{r}
#siRNA_length
siRNA_length %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm" |
           sample_type == "seedling shoot"|
           sample_type == "zygote") %>% 
  mutate(profile1 = case_when(
    profile == "total siRNA" ~ "general siRNA",
    profile != "total siRNA" ~ profile
    )) %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    T ~ sample_type
  )) %>% 
  mutate(profile1 = factor(profile1, levels = c("TIR", "Gypsy", "CentO", "uniquely mapped", "general siRNA"))) %>% 
  ggplot(aes(x = length, y = proportion*100 )) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  facet_wrap( ~ profile1, scales = "free_y", ncol = 2) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type), position = position_dodge(0.01)) + 
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2"),
                     limits = c("egg", "seedling", "sperm", "zygote")) +
  labs(color = NULL,
       x = "length (nt)",
       y = "% siRNA") +
  theme_minimal() +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(legend.position = c(0.77, 0.12))

ggsave(filename = "siRNA profile gamemtes.svg", width = 4.9, height = 6)
ggsave(filename = "siRNA profile gamemtes.png", width = 4.9, height = 6)
```


##stat 
```{r}
anova_data <- siRNA_length %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm" |
           sample_type == "seedling shoot"|
           sample_type == "zygote") %>% 
  filter(length == 24)
```
 

```{r}
model_24 <- lm(proportion ~ profile * sample_type, data = anova_data)
anova(model_24)
```

```{r}
est_24 <- emmeans(model_24, pairwise ~ sample_type | profile)
multcomp::cld(est_24$emmeans, Letters = letters)
```





##scatter plot
```{r}
head(sample_des_size)

siRNA_comp_l <- sample_des_size %>% 
  gather("RNA_type", "count", 8:13) %>% 
  mutate(pro = count / `total read`) %>% 
  select(-count) %>% 
  select(sample_ID, sample_type, genotype, pro, RNA_type) %>% 
  filter(RNA_type != "total.siRNA") %>% 
  spread(RNA_type, pro)

siRNA_comp_l
```
```{r}
levels <- c(
    "leaf", "root", "seedling shoot", 
    "ovary", 
    "megasporocyte ovary", "megasporocyte meiosis ovary", 
    "functional megaspore ovary", "8-nuclei embryo sac ovary", 
    "egg",
    "spikelets, anther stage 3",
    "spikelets, anther stage 5",
    "spikelets, anther stage 7",
    "pre-meiotic anther",
    "meiosis anther",
    "microspore anther",
    "bicellular pollen anther",
    "sperm",
    "embryo 7-8 DAF",
    "endosperm 7-8 DAF",
    "mature embryo", 
    "12h imbibed embryo",
    "24h imbibed embryo",
    "zygote"
  )
```

```{r}
siRNA_TE_pro <- siRNA_length %>% 
  filter(profile != "total siRNA") %>% 
  group_by(sample_ID, sample_type, genotype, profile) %>% 
  summarise(pro = sum(proportion)) %>% 
  mutate(RNA_type = profile) %>% 
  select(-profile) %>% 
  spread("RNA_type", "pro") %>%
  inner_join(siRNA_comp_l, by = c("sample_ID", "sample_type", "genotype")) %>% 
  mutate(type = case_when(
      sample_type == "egg" ~ "egg", 
      str_detect(sample_type, "sperm" ) ~ "sperm",
      str_detect(sample_type, "zygote") ~ "zygote" 
  )) %>% 
  as.data.frame()



row.names(siRNA_TE_pro) <- paste(siRNA_TE_pro$sample_ID, 
                                 siRNA_TE_pro$sample_type, 
                                 siRNA_TE_pro$genotype, 
                                 siRNA_TE_pro$type,
                                 sep = ":")


siRNA_TE_pro 
```


```{r}
siRNA_TE_pro %>%
  filter(str_detect(sample_ID, "drm2") == F) %>% 
  #filter(sample_type == "zygote") %>% 
  ggplot(aes(y = Gypsy * 100, x = TIR * 100)) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_ellipse(aes(group = type, fill = type), alpha = 0.3, geom = "polygon", level = 0.8, type = "t") +
  geom_point(aes(color = sample_type, shape = genotype, alpha = genotype, size = genotype)) +
  labs(x = "TIR siRNA (% siRNA)",
       y = "Gypsy siRNA (% siRNA)",
       fill = NULL,
       color = NULL, 
       shape = NULL) +
  scale_color_manual(values = c("darkgreen", "grey40", "seagreen",
                               "orangered3", "orchid1", "brown2", "brown3", "brown4", "tomato1", 
                               "turquoise1", "turquoise3", "turquoise4",
                               "tan", "tan2", "tan3", "tan4", "dodgerblue2",
                               "magenta", "lightgoldenrod4",
                               "magenta4", "yellow3", "yellow4", "violetred2"), 
                    limits = levels) + 
  scale_fill_manual(values = c("tomato1", "dodgerblue2", "violetred2", NA),
                   limits = c("egg","sperm", "zygote", NA)) +
  scale_size_manual(values = c(4, 2), guide = F) +
  scale_alpha_manual(values = c(1, 0.8), guide = F) +
  scale_shape_manual(values = c(15, 16)) +
  theme_minimal() +
  theme(legend.position = c(0.8, 0.8)) +
  guides(color = F) + 
  guides(fill = F) + 
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black")) 
```

```{r}
library(cowplot)
```
#phasi plus scatter
```{r}
p0 <- get_legend(
  smRNA_comp_L %>% 
  filter(genotype == "WT") %>% 
  filter(RNA_type != "general siRNA") %>% 
  mutate(RNA_type2 = factor(RNA_type1, levels = c(
    "miRNAs", "phasiRNAs (24nt)", "phasiRNAs (21nt)", "tRNAs", "NOR RNAs", "5S rRNAs"
  ))) %>% 
  mutate(sample_type.f = factor(sample_type, levels = levels)) %>%
  ggplot(aes(x = sample_type.f, y = pro * 100)) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_bar(stat = "summary", aes(fill = sample_type.f), fun.y = mean, alpha = 0.8) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", 
               aes(group = sample_type.f), position = position_dodge(0.01), width = 0.3) +
  facet_wrap( ~ RNA_type2,  scales = "free_y", ncol = 3) +
  scale_fill_manual(values = c("darkgreen", "grey40", "seagreen",
                               "orangered3", "orchid1", "brown2", "brown3", "brown4", "tomato1", 
                               "turquoise1", "turquoise3", "turquoise4",
                               "tan", "tan2", "tan3", "tan4", "dodgerblue2",
                               "magenta", "lightgoldenrod4",
                               "magenta4", "yellow3", "yellow4", "violetred2"), 
                    limits = levels) + 
  guides(fill = guide_legend(ncol = 2)) +
  labs(x = NULL, 
       y = "proportion (%)",
       fill = NULL) +
  theme_minimal() +
  theme(legend.position = "bottom", legend.direction = "horizontal") +
  theme(text = element_text(size= 18, face="bold")) +
  #theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.x=element_blank()) +
  theme(axis.text.y=element_text(colour = "black"))
)

p0 %>% plot_grid()
```

```{r}
p1 <- smRNA_comp_L %>% 
  filter(genotype == "WT") %>% 
  filter(RNA_type != "general siRNA") %>% 
  mutate(RNA_type2 = factor(RNA_type1, levels = c(
    "miRNAs", "phasiRNAs (24nt)", "phasiRNAs (21nt)", "tRNAs", "NOR RNAs", "5S rRNAs"
  ))) %>% 
  mutate(sample_type.f = factor(sample_type, levels = levels)) %>%
  ggplot(aes(x = sample_type.f, y = pro * 100)) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_bar(stat = "summary", aes(fill = sample_type.f), fun.y = mean, alpha = 0.8) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", 
               aes(group = sample_type.f), position = position_dodge(0.01), width = 0.3) +
  facet_wrap( ~ RNA_type2,  scales = "free_y", ncol = 3) +
  scale_fill_manual(values = c("darkgreen", "grey40", "seagreen",
                               "orangered3", "orchid1", "brown2", "brown3", "brown4", "tomato1", 
                               "turquoise1", "turquoise3", "turquoise4",
                               "tan", "tan2", "tan3", "tan4", "dodgerblue2",
                               "magenta", "lightgoldenrod4",
                               "magenta4", "yellow3", "yellow4", "violetred2"), 
                    limits = levels) + 
  guides(fill = guide_legend(ncol = 1)) +
  labs(x = NULL, 
       y = "% small RNA",
       fill = NULL) +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size= 18, face="bold")) +
  #theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.x=element_blank()) +
  theme(axis.text.y=element_text(colour = "black")) 

p1
```

```{r}
p2 <- siRNA_TE_pro %>%
  filter(str_detect(sample_ID, "drm2") == F) %>% 
  ggplot(aes(y = Gypsy * 100, x = TIR * 100)) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_ellipse(aes(group = type, fill = type), alpha = 0.3, geom = "polygon", level = 0.8, type = "t") +
  geom_point(aes(color = sample_type, shape = genotype, alpha = genotype, size = genotype)) +
  labs(x = "TIR siRNA (% siRNA)",
       y = "Gypsy siRNA (% siRNA)",
       fill = NULL,
       color = NULL, 
       shape = NULL) +
  scale_color_manual(values = c("darkgreen", "grey40", "seagreen",
                               "orangered3", "orchid1", "brown2", "brown3", "brown4", "tomato1", 
                               "turquoise1", "turquoise3", "turquoise4",
                               "tan", "tan2", "tan3", "tan4", "dodgerblue2",
                               "magenta", "lightgoldenrod4",
                               "magenta4", "yellow3", "yellow4", "violetred2"), 
                    limits = levels) + 
  scale_fill_manual(values = c("tomato1", "dodgerblue2", "violetred2", NA),
                    limits = c("egg","sperm", "zygote", NA)) +
  scale_size_manual(values = c(4, 2), guide = F) +
  scale_alpha_manual(values = c(1, 0.8), guide = F) +
  scale_shape_manual(values = c(15, 16)) +
  xlim(c(5, 42)) +
  theme_minimal() +
  theme(legend.position = c(0.75, 0.75)) +
  guides(color = F) + 
  guides(fill = F) + 
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))

p2
```
```{r}
left <- plot_grid(
  p1, NULL, p0, ncol = 1, rel_heights = c(1, 0.01, 1)
)

left
```
```{r}
plot_grid(left, NULL, p2,
          rel_widths = c(1, 0, 0.55), 
          nrow = 1,
          labels = c("A", "B"),
          label_size = 24)

ggsave("phasi_and_scatter.svg", height = 6, width = 10)
ggsave("phasi_and_scatter.png", height = 6, width = 10)
```



Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
