---
title: "miR"
author: "Chenxin Li"
date: "Aug 15, 2020"
output:
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(edgeR)
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(vegan)
library(dynamicTreeCut)
library(RColorBrewer)
library(stringr)
library(svglite)
library(ggdendro)
library(ggalluvial)
library(cowplot)
```


#load data
```{r}
sample_des <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/sample_des_clvs12.xlsx") 

sample_des

smRNA_comp <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/codes_and_output/smRNA_comp.csv")  

sample_des_lib <- sample_des %>% 
  select(-library)%>% 
  inner_join(smRNA_comp, by = "sample_ID") %>% 
  arrange(sample_ID)

sample_des_lib 
```
```{r}
#counts from gametes
miR_CLVS11 <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS11/miR_cts.csv")
```
```{r}
miR_CLVS8 <- miR_CLVS11 %>% 
  select(-SD5, -zg1c, -zg2c)
```

```{r}
#load CLVS12 
SD7 <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/mir_cov/SD7_S55_L003_R1_001.fastq.gz.trimmed.dedup.fastq.gz.miR.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

zg1d <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/mir_cov/zg1d_S49_L003_R1_001.fastq.gz.trimmed.dedup.fastq.gz.miR.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

zg1e <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/mir_cov/zg1e_S51_L003_R1_001.fastq.gz.trimmed.dedup.fastq.gz.miR.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

zg2d <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/mir_cov/zg2d_S50_L003_R1_001.fastq.gz.trimmed.dedup.fastq.gz.miR.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

zg2e <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/mir_cov/zg2e_S52_L003_R1_001.fastq.gz.trimmed.dedup.fastq.gz.miR.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

zg3a <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/mir_cov/zg3a_S53_L003_R1_001.fastq.gz.trimmed.dedup.fastq.gz.miR.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

zg3b <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/mir_cov/zg3b_S54_L003_R1_001.fastq.gz.trimmed.dedup.fastq.gz.miR.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
```

```{r}
head(SD7)
```




#data arrangement
```{r}
cts_clvs12 <- SD7[, 9:10] %>% 
  full_join(zg1d[, 9:10], by = "X9") %>% 
  full_join(zg1e[, 9:10], by = "X9") %>% 
  full_join(zg2d[, 9:10], by = "X9") %>% 
  full_join(zg2e[, 9:10], by = "X9") %>% 
  full_join(zg3a[, 9:10], by = "X9") %>% 
  full_join(zg3a[, 9:10], by = "X9") %>%
  separate(X9, c("A", "B", "C", "D"), sep = ";", remove = T) %>% 
  dplyr::select(-A, -B, -D) %>% 
  separate(C, c("C1", "miR"), remove = T, sep = "=") %>% 
  dplyr::select(-C1)

colnames(cts_clvs12)[2:ncol(cts_clvs12)] <- c(
  "SD7", "zg1d", "zg1e", "zg2d", "zg2e", "zg3a", "zg3b"
  )

head(cts_clvs12)
write_excel_csv(cts_clvs12, "miR_cts_clvs12.csv") 
```

```{r}
cts <- miR_CLVS8 %>% 
  full_join(cts_clvs12, by = "miR") %>% 
  filter(str_detect(miR, "miR159a.1") == F) %>% 
  filter(str_detect(miR, "miR159b") == F) 

head(cts)
```

##miR of same sequence? 
```{r}
osa_mirBase22 <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/osa_mirBase22.txt", header=FALSE) 

osa_mir <- osa_mirBase22[, 1:2]
colnames(osa_mir) <- c("tag", "sequence") 
osa_mir <- osa_mir%>% 
  separate(tag, c("T1", "T2", "T3", "T4", "T5"), sep = " ", remove = T) %>% 
  mutate(miR = paste("osa", T5, sep = "-")) %>% 
  select(-T1, -T3, -T4, -T5) %>% 
  arrange(T2)

osa_mir <- osa_mir[osa_mir$miR %in% cts$miR, ]
head(osa_mir, 10)
```
```{r}
osa_mir_unique <- distinct(osa_mir, sequence, .keep_all = T)
osa_mir_dup <- anti_join(osa_mir, osa_mir_unique, by = c("miR", "sequence"))
dim(osa_mir_unique)

head(osa_mir_unique, 10)

write.csv(osa_mir, "osa_mir_seq.csv")
write.csv(osa_mir_dup, "osa_mir_seq_duplicated.csv")
```

```{r}
cts_seq <- cts %>% 
  full_join(osa_mir, by = "miR") %>% 
  #mutate(n.ch = nchar(miR)) %>% 
  #mutate(fam = substr(miR, 1, (n.ch - 1))) %>% 
  gather("sample_ID", "counts", 2:25) %>% 
  group_by(sequence, sample_ID) %>%
  summarise(sum.count = sum(counts)) %>% 
  spread(sample_ID, sum.count) %>% 
  full_join(osa_mir_unique, by = "sequence") %>% 
  arrange(T2) %>% 
  dplyr::select(-T2)

cts_seq <- cts_seq[complete.cases(cts_seq), ]
cts_seq %>% head(10)  


dim(cts_seq)[1] == dim(osa_mir_unique)[1]

#cts_seq %>% filter(str_detect(miR, "159"))
```

#EdgeR
```{r}
sample_des_lib_pca <- sample_des_lib %>% 
  filter(sample_type == "sperm"|
           sample_type == "egg"|
           sample_type == "seedling shoot"|
           sample_type == "ovary"| 
           sample_type == "zygote")

sample_des_lib_pca
```

```{r}
#sample_des_lib_pca$sample_ID
cts_matrix <- cts%>% 
  select(B15, B20, B25, BS1, BS2, BS4a, BS4b, BS6a, BS6b,
         E2, E3, E4, E9a, E9b, E9c,
         OV1a, OV1b, OV2,
         SD4, SD7, SP5b,
         SP5c, zg1d, zg1e, zg2d, zg2e, zg3a, zg3b) %>%
  as.data.frame()
row.names(cts_matrix) <- cts$miR

y <- DGEList(counts=cts_matrix, group=sample_des_lib_pca$sample_type, lib.size = sample_des_lib_pca$`total read`)    
y$samples

keep <- rowSums(cpm(y)> 1) >= 3
#keep <- filterByExpr(y)
y <- y[keep, , keep.lib.sizes = T]
y <- calcNormFactors(y, method = "none")
y$samples

design <- model.matrix(~ 0 + sample_des_lib_pca$sample_type)
y <- estimateDisp(y, design)
plotBCV(y)
```
#PCA plots
```{r}
cpm <- cpm(y, log =F) %>% as.data.frame()
logcpm <- cpm(y, log =T) %>% as.data.frame()     
adonis((t(logcpm)) ~ sample_type, sample_des_lib_pca, method= "e")
```

```{r}
pc <- prcomp(t(logcpm))
pc_importance <- as.data.frame(t(summary(pc)$importance))
pc_importance

ggplot(pc_importance, aes(x = 1:length( `Proportion of Variance`), y = `Proportion of Variance`) ) +
  geom_bar(stat = 'identity') +
  xlab('dimension') +
  theme_classic()

ggplot(pc_importance, aes(x = 1:length(`Cumulative Proportion`), y = `Cumulative Proportion`) ) +
  geom_point() +
  xlab('dimension') +
  ylab('cumulative variance') +
  ylim(c(0, 1.1))+
  theme_classic()

pc_nice <- cbind(sample_des_lib_pca, pc$x)
pc_nice <- pc_nice %>% 
  mutate(type = case_when(
    sample_type == "seedling shoot" |
      sample_type == "root" |
      sample_type == "leaf" ~ "vegetative",
    sample_type == "egg" |
      str_detect(sample_type, "ovary") ~ "female",
    sample_type == "sperm" |
      str_detect(sample_type, "spikelet") |
      str_detect(sample_type, "anther") ~ "male",
    str_detect(sample_type, "zygote") ~ "zygote"
  )) %>% 
  mutate(type = factor(type, levels = c("female", "vegetative", "male", "zygote")))
```

```{r}
pc_nice %>% 
  filter(sample_type == "seedling shoot"|
           sample_type == "egg"|
           sample_type == "sperm"|
           #sample_type == "ovary"|
           sample_type == "zygote") %>% 
  mutate(sample_type1 = case_when(
    sample_type == "seedling shoot" ~ "seedling",
    sample_type != "seedling shoot" ~ sample_type
  )) %>% 
  ggplot(aes(PC1, PC2)) +
  stat_ellipse(aes(fill = type), alpha = 0.3, level = 0.9, type = "t", geom = "polygon") +
  geom_point(aes(color = sample_type1), size = 3, alpha = 0.8) +
  labs(x = paste("PC1 (", pc_importance[1, 2] %>% signif(3)*100, "% of Variance)", sep = ""), 
       y = paste("PC2 (", pc_importance[2, 2] %>% signif(3)*100, "% of Variance)", "  ", sep = ""),
       fill = NULL,
       color = NULL) +
  scale_color_manual(values = c(  "tomato1", "seagreen", "dodgerblue2", "violetred2"),
                   limits = c("egg", "seedling", "sperm", "zygote")) +
  guides(fill = F) +
  theme_bw() + 
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave(filename = "miR_PCA_no_159a1_b.svg", width = 6, height = 5)
ggsave(filename = "miR_PCA_no_159a1_b.png", width = 6, height = 5)
```

```{r}
pc_nice %>% 
  filter(sample_type == "seedling shoot"|
           sample_type == "egg"|
           sample_type == "sperm"|
           #sample_type == "ovary"|
           sample_type == "zygote") %>% 
  mutate(sample_type1 = case_when(
    sample_type == "seedling shoot" ~ "seedling\nshoot",
    sample_type != "seedling shoot" ~ sample_type
  )) %>% 
  ggplot(aes(PC1, PC2)) +
  #stat_ellipse(aes(fill = type), alpha = 0.3, level = 0.9, type = "t", geom = "polygon") +
  geom_point(aes(color = sample_type1), size = 3, alpha = 0.8) +
  geom_label(aes(label = sample_ID), nudge_x = 2, nudge_y = 2) +
  labs(x = paste("PC1 (", pc_importance[1, 2] %>% signif(3)*100, "% of Variance)", sep = ""), 
       y = paste("PC2 (", pc_importance[2, 2] %>% signif(3)*100, "% of Variance)", "  ", sep = ""),
       fill = NULL,
       color = NULL) +
  scale_color_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2", "violetred2"),
                   limits = c("ovary", "egg", "seedling\nshoot", "sperm", "zygote")) +
  theme_bw()+
# theme(legend.position = "bottom", legend.box = "vertical",legend.spacing.y = unit(0.2, "lines"))+
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black"))
```



#differential expressions
```{r}
fit <- glmQLFit(y, design, robust = T) 
z_e <- glmTreat(fit,  contrast = c(-1, 0, 0, 0, 1), lfc = 1) 
z_ov <- glmTreat(fit, contrast = c(0, -1, 0, 0, 1), lfc = 1) 
z_sd <- glmTreat(fit, contrast = c(0, 0, -1, 0, 1), lfc = 1) 
z_sp <- glmTreat(fit, contrast = c(0, 0, 0, -1, 1), lfc = 1) 
```


```{r}
z.vs.e <- z_e$table %>% 
  mutate(FDR = p.adjust(PValue, method = "fdr")) %>% 
  mutate("sig"=ifelse(FDR < 0.05,"s","ns")) %>%
  mutate("miR" = row.names(z_e$table))

z.vs.e %>% 
  filter(sig == "s") %>% 
  filter(logFC < 0)
```
```{r}
z.vs.ov <- z_ov$table %>% 
  mutate(FDR = p.adjust(PValue, method = "fdr")) %>% 
  mutate("sig"=ifelse(FDR < 0.05,"s","ns")) %>%
  mutate("miR" = row.names(z_sd$table))

z.vs.ov %>% 
  filter(sig == "s")# %>% 
 # filter(logFC <0) %>% 
 # head()
```


```{r}
z.vs.sd <- z_sd$table %>% 
  mutate(FDR = p.adjust(PValue, method = "fdr")) %>% 
  mutate("sig"=ifelse(FDR < 0.05,"s","ns")) %>%
  mutate("miR" = row.names(z_sd$table))

z.vs.sd %>% 
  filter(sig == "s") %>% 
  head()
```
```{r}
z.vs.sp <- z_sp$table %>% 
  mutate(FDR = p.adjust(PValue, method = "fdr")) %>% 
  mutate("sig"=ifelse(FDR < 0.05,"s","ns")) %>%
  mutate("miR" = row.names(z_sd$table))

z.vs.sp %>% 
  filter(sig == "s") %>% 
  head()
```


#visualization on per million miRNA scale
```{r}
a <- DGEList(counts = cts_matrix ,group=sample_des_lib_pca$sample_type)    
a$samples

keep <- rowSums(cpm(a)> 0) >= 3 
#keep <- filterByExpr(y)
a <- a[keep, , keep.lib.sizes = F]
a <- calcNormFactors(a, method = "none")

a$samples
```
```{r}
counts_per_M_miR <- cpm(a, log =F) %>% as.data.frame()
```

```{r}
cpm_l <- counts_per_M_miR %>% 
  mutate(miR = row.names(counts_per_M_miR)) %>% 
  gather("sample_ID", "CPM", 1:ncol(counts_per_M_miR)) %>% 
   mutate(sample_type = case_when(
    sample_ID == "E2" |
      sample_ID == "E3" |
      sample_ID == "E4" |
      sample_ID == "E9a" |
      sample_ID == "E9b" |
      sample_ID == "E9c" ~ "egg",
      sample_ID == "SP5b" |
      sample_ID == "SP5c" |
      sample_ID == "BS1" |
      sample_ID == "BS2" |
      sample_ID == "BS4a" |
      sample_ID == "BS4b" |
      sample_ID == "BS6a" |
      sample_ID == "BS6b"  ~ "sperm", 
      sample_ID == "B15" |
      sample_ID == "B20" |
      sample_ID == "B25" |
      sample_ID == "SD4" |
      sample_ID == "SD7" ~ "seedling shoot", 
      sample_ID == "OV1a" |
      sample_ID == "OV1b" |
      sample_ID == "OV2" ~ "ovary",
      str_detect(sample_ID, "zg") ~ "zygote"
    )) 

head(cpm_l)
```


##high egg low zygote 
```{r}
 z.vs.e %>% 
  filter(sig == "s") %>% 
  filter(logFC < 0) %>% 
  inner_join(cpm_l, by = "miR") %>% 
  filter(sample_type == "egg" |
           sample_type == "zygote"|
           sample_type == "ovary") %>% 
  mutate(sample_type = factor(sample_type, levels = c(
    "ovary", "egg", "zygote"
  ))) %>% 
  mutate(header = substr(miR, start = 5, stop = 100)) %>% 
  ggplot(aes(x = sample_type, y = CPM)) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  # geom_point(position = position_jitter(0.2, seed = 666), shape = 21, color = "black", alpha = 0.8) +
  stat_summary(geom = "line", aes(group = miR, color = -log10(FDR)), 
               fun.y = mean, size = 2, alpha = 0.8, lineend = "round") +
  stat_summary(geom = "errorbar", fun.data = "mean_cl_boot", width = 0.3) +
  facet_wrap(~ header, scales = "free_y", ncol = 4) +
  labs(x = NULL,
       y = "counts per million miR reads",
       color = "-log10(FDR)\nzygote vs. egg ") +
  scale_color_gradientn(colours = brewer.pal(9, "YlOrRd")[c(6:9)], breaks = c(2, 3)) +
  theme_minimal() +
  theme(legend.position = c(0.75, 0.05), legend.direction = "horizontal") +
  theme(legend.key.width = unit(1, "lines")) +
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 14, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

```
 


```{r}
 z.vs.e %>% 
  filter(sig == "s") %>% 
  filter(logFC < 0) %>% 
  inner_join(cpm_l, by = "miR") %>% 
  filter(sample_type == "egg" |
         sample_type == "zygote") %>% 
  mutate(header = substr(miR, start = 5, stop = 100)) %>% 
  ggplot(aes(x = sample_type, y = CPM)) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_bar(stat = "summary", aes(fill = sample_type), alpha = 0.8, fun.y = mean) + 
  #stat_summary(geom = "line", aes(group = miR, color = -log10(FDR)), 
  #             fun.y = mean, size = 2, alpha = 0.8, lineend = "round") +
  geom_point(position = position_jitter(0.1, seed = 666), 
             shape = 21, color = "white", alpha = 0.8,
             aes(fill = sample_type)) + 
  stat_summary(geom = "errorbar", fun.data = "mean_cl_boot", width = 0.3) +
  facet_wrap(~ header, scales = "free_y", ncol = 4) +
  labs(x = NULL,
       y = "counts per million miR reads",
       fill = NULL) +
  #scale_color_gradientn(colours = brewer.pal(9, "YlOrRd")[c(6:9)], breaks = c(2, 3)) +
  scale_fill_manual(values = c("tomato1",  "violetred2"),
                   limits = c("egg", "zygote")) +
  theme_minimal() +
  theme(legend.position = c(0.75, 0.05), legend.direction = "horizontal") +
  theme(legend.key.width = unit(1, "lines")) +
  theme(axis.text.x = element_text(angle = 45, hjust= 1)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) 

```

 


