---
title: "siRNA_abundance_at_24nt_loci"
author: "Chenxin Li"
date: "Feb 19, 2020"
output:
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(RColorBrewer)
library(stringr)
library(svglite)
```

#load data
```{r}
egg_spec0 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/siRNA_length_counts/lengths_summary.egg_specific.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

seedling_spec0 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/siRNA_length_counts/lengths_summary.seedling_specific.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

sperm_spec0 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/siRNA_length_counts/lengths_summary.sperm_specific.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

intersection0 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/siRNA_length_counts/lengths_summary.intersection.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```


```{r}
egg_spec12 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.egg_specific.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

seedling_spec12 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.seedling_specific.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

sperm_spec12 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.sperm_specific.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

intersection12 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.intersection.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
```{r}
egg_spec <- rbind(egg_spec0, egg_spec12)
seedling_spec <- rbind(seedling_spec0, seedling_spec12)
sperm_spec <- rbind(sperm_spec0, sperm_spec12)
intersection <- rbind(intersection0, intersection12)
```

```{r}
colnames(egg_spec) <- c("library", "20", "21", "22", "23", "24", "25")
colnames(seedling_spec) <- c("library", "20", "21", "22", "23", "24", "25")
colnames(sperm_spec) <- c("library", "20", "21", "22", "23", "24", "25")
colnames(intersection) <- c("library", "20", "21", "22", "23", "24", "25")

```

```{r}
sample_des <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/sample_des_clvs12.xlsx")

smRNA_comp <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/codes_and_output/smRNA_comp.csv")
```
```{r}
sample_des_size <- smRNA_comp %>% 
  inner_join(sample_des, by = c("sample_ID")) %>% 
  select(-2, -3, -4, -5, -6)
sample_des_size
```
```{r}
total_siRNA_length0 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/length_profiel_fig2/lengths_summary.bam.siRNAs.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE) 

CLVS12_total <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.siRNAs.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

total_siRNA_length <- total_siRNA_length0 %>%
  rbind(CLVS12_total %>% 
          filter(str_detect(X1, "q20") == F))


colnames(total_siRNA_length) <- c("library", "20", "21", "22", "23", "24", "25")
```

```{r}
identify_samples <- function(df){
  df %>% 
    mutate(sample_ID = case_when(
    str_detect(library, "sperm.1") ~ "BS1", #1
    str_detect(library, "Sperm2") ~ "BS2",  #2
    str_detect(library, "Sperm4a") ~ "BS4a",  #3
    str_detect(library, "sperm4b") ~ "BS4b",  #4
    str_detect(library, "sperm6a") ~ "BS6a",  #5
    str_detect(library, "sperm6b") ~ "BS6b",  #6
    str_detect(library, "sperm.5b") ~ "SP5b", #7
    str_detect(library, "sperm.5c") ~ "SP5c", #8
    str_detect(library, "Bulk.15") ~ "B15",   #9
    str_detect(library, "Bulk.20") ~ "B20",   #10
    str_detect(library, "Bulk.25") ~ "B25",   #11
    str_detect(library, "seedling4") ~ "SD4", #12
    str_detect(library, "egg.9a") ~ "E9a",    #13
    str_detect(library, "egg.9b") ~ "E9b",    #14
    str_detect(library, "egg.9c") ~ "E9c",    #15
    str_detect(library, "eggcell4") ~ "E4",   #16
    str_detect(library, "eggcell3") ~ "E3",   #17
    str_detect(library, "eggcell2") ~ "E2",   #18
    str_detect(library, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(library, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(library, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(library, "SRR57") ~ "root",              #22
    str_detect(library, "ab_control") ~ "ddm1_con",     #23
    str_detect(library, "ddm1ab") ~ "ddm1",             #24
    str_detect(library, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(library, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(library, "SRR2544786") ~ "em_dry", #27  
    str_detect(library, "SRR5049778") ~ "an_pre",  #28
    str_detect(library, "SRR5049779") ~ "an_mei",  #29
    str_detect(library, "SRR5049780") ~ "an_mic",  #30
    str_detect(library, "SRR5049781") ~ "an_bi",   #31
    str_detect(library, "SRR5049786") ~ "ov_1st",  #32
    str_detect(library, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(library, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(library, "SRR5049789") ~ "ov_4th",  #35
    str_detect(library, "S3_1") ~ "S3_1",          #36
    str_detect(library, "S3_2") ~ "S3_2",          #37
    str_detect(library, "S3_3") ~ "S3_3",          #38
    str_detect(library, "S5_1") ~ "S5_1",          #39
    str_detect(library, "S5_2") ~ "S5_2",          #40
    str_detect(library, "S5_3") ~ "S5_3",          #41
    str_detect(library, "S7_1") ~ "S7_1",          #42
    str_detect(library, "S7_2") ~ "S7_2",          #43
    str_detect(library, "S7_3") ~ "S7_3",          #44
    str_detect(library, "SRR2544787") ~ "em_12h",  #45
    str_detect(library, "SRR2544788") ~ "em_24h",  #46
    str_detect(library, "SRR771501") ~ "em_7DAF",  #47
    str_detect(library, "SRR771500") ~ "en_7DAF",   #48
    str_detect(library, "SD7") ~ "SD7", #49
    str_detect(library, "zg1d") ~ "zg1d", #50
    str_detect(library, "zg1e") ~ "zg1e",  #51
    str_detect(library, "zg2d") ~ "zg2d",  #52
    str_detect(library, "zg2e") ~ "zg2e",  #53
    str_detect(library, "zg3a") ~ "zg3a", #54
    str_detect(library, "zg3b") ~ "zg3b"  #55
  )) %>% 
    inner_join(sample_des_size %>% 
                 select(`total read`, `general siRNA`, sample_ID, sample_type, genotype), by = "sample_ID")
}
```

```{r}
total_siRNA_length1 <- identify_samples(total_siRNA_length) 
head(total_siRNA_length1)
```

 

```{r}
egg_spec1 <- identify_samples(egg_spec) %>% 
  mutate(bin = "egg-\nspecific\nloci") %>% 
  filter(sample_type == "egg" | 
           sample_type == "seedling shoot" |
           sample_type == "sperm"|
           sample_type == "zygote")

seedling_spec1 <- identify_samples(seedling_spec) %>% 
  mutate(bin = "seedling-\nspecific\nloci") %>% 
  filter(sample_type == "egg" | 
           sample_type == "seedling shoot" |
           sample_type == "sperm"|
           sample_type == "zygote")

sperm_spec1 <- identify_samples(sperm_spec) %>% 
  mutate(bin = "sperm-\nspecific\nloci") %>% 
  filter(sample_type == "egg" | 
           sample_type == "seedling shoot" |
           sample_type == "sperm"|
           sample_type == "zygote")

intersection1 <- identify_samples(intersection) %>% 
  mutate(bin = "intersect.") %>% 
  filter(sample_type == "egg" | 
           sample_type == "seedling shoot" |
           sample_type == "sperm"|
           sample_type == "zygote")
```


#calculate proportion of 24nt siRNA in sample-specific bins
```{r}
loci_table_spec_inter <- rbind(egg_spec1, 
                             seedling_spec1, 
                             sperm_spec1,
                         intersection1) %>% 
  inner_join(total_siRNA_length1 %>% 
               select(-library), by = c("sample_ID", "genotype", "sample_type", "total read")) %>% 
  select(-genotype, -library) %>% 
  mutate(proportion_in_bin = `24.x` / `24.y`) %>% 
  select(sample_ID, `total read`, sample_type, bin, proportion_in_bin)

loci_table_spec_inter


loci_table_spec_inter_s <- loci_table_spec_inter %>% 
  group_by(sample_type, bin) %>% 
  summarise(mean.porportion.in.bin = mean(proportion_in_bin)) %>% 
  ungroup() %>% 
  spread(bin, mean.porportion.in.bin)

colnames(loci_table_spec_inter_s) <- c("sample type", "egg-specific", "intersect.", "seedling-specific", "sperm-specific")

loci_table_spec_inter_s <- loci_table_spec_inter_s[, c(1, 2, 4, 5, 3)]
loci_table_spec_inter_s
```



#plot normalized to total siRNA
```{r}
loci_table <- rbind(egg_spec1, seedling_spec1, sperm_spec1, intersection1) %>% 
  inner_join(total_siRNA_length1 %>% 
               select(-library), by = c("sample_ID", "genotype", "sample_type", "total read", "general siRNA")) %>% 
  select(-genotype, -library) 

head(loci_table)
```


```{r}
loci_2 <- loci_table %>% 
  select(-`20.y`, -`21.y`, -`22.y`, -`23.y`, -`24.y`, -`25.y`) %>% 
  gather("length", "counts", 1:6) %>% 
  mutate(proportion = counts / `general siRNA`) %>% 
  mutate(length = substr(length, start = 0, stop = 2)) %>% 
  select(-counts) %>% 
  mutate(norm = "per total siRNA")
  
  
loci_2
```


```{r}
axis_line <- data.frame(
  length = c(-Inf),
  bin.f = c("egg-\nspecific\nloci")
)

axis_line
```



```{r}
loci_2 %>% 
    mutate(bin2 = case_when(
      bin == "intersect." ~ "intersect.\nloci",
      bin != "intersect." ~ bin
    )) %>% 
    mutate(bin.f = factor(bin2, levels = c(
    "egg-\nspecific\nloci", 
    "seedling-\nspecific\nloci",
    "sperm-\nspecific\nloci",
    "intersect.\nloci"
  ))) %>%
  mutate(sample_type = case_when(
    str_detect(sample_type, "seedling") ~ "seedling",
    T ~ sample_type
  )) %>% 
  ggplot(aes(x = length, y = proportion*100 )) +
  facet_grid( norm ~ bin.f, scales = "free_y", switch = "y") +
  geom_vline(data = axis_line, aes(xintercept = length), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type), position = position_dodge(0.01)) + 
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2"),
                     limits = c("egg", "seedling", "sperm", "zygote")) +
  labs(color = "siRNA transcriptome",
       x = "length (nt)",
       y = "% siRNA") +
  guides(color = guide_legend(nrow = 2)) + 
  theme_minimal() +
  theme(panel.spacing.y = unit(1, "lines"), strip.placement = "outside") +
  theme(legend.position = "bottom") +
  theme(strip.text.y = element_blank()) +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(legend.position = "bottom")

ggsave("siRNA_in_24nt_loci.png", width = 8, height = 3.8)
ggsave("siRNA_in_24nt_loci.svg", width = 8, height = 3.8)
```
```{r}
loci_2 %>% 
  filter(str_detect(bin, "seedling-")) %>% 
  filter(length == 24) %>% 
    mutate(bin2 = case_when(
      bin == "intersect." ~ "intersect.\nloci",
      bin != "intersect." ~ bin
    )) %>% 
    mutate(bin.f = factor(bin2, levels = c(
    "egg-\nspecific\nloci", 
    "seedling-\nspecific\nloci",
    "sperm-\nspecific\nloci",
    "intersect.\nloci"
  ))) %>%
  ggplot(aes(x = sample_type, y = proportion*100 )) +
  facet_grid( norm ~ bin.f, scales = "free_y", switch = "y") +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type), 
               position = position_dodge(0.01), width = 0.3) + 
  geom_point(aes(fill = sample_type), color = "white", alpha = 0.8, 
             position = position_jitter(0.1, seed = 666), shape = 21, size = 2) +
  geom_text(aes(label = sample_ID), position = position_dodge2(0.5)) + 
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2"),
                     limits = c("egg", "seedling shoot", "sperm", "zygote")) +
  scale_fill_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2"),
                     limits = c("egg", "seedling shoot", "sperm", "zygote")) +
  labs(fill = "siRNA transcriptome",
       color = NULL, 
       x = "length (nt)",
       y = "% siRNA") +
  guides(fill = guide_legend(nrow = 2)) + 
  guides(color = F) + 
  theme_minimal() +
  theme(panel.spacing.y = unit(1, "lines"), strip.placement = "outside") +
  theme(legend.position = "bottom") +
  theme(strip.text.y = element_blank()) +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_blank()) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(legend.position = "bottom")
```

```{r}
anova_df <- loci_2 %>% 
  filter(str_detect(bin, "seedling-")) %>% 
  filter(length == 24) 
```

```{r}
model_sd_24 <- lm(proportion ~ sample_type, anova_df)
anova(model_sd_24)
```
```{r}
est_sd24 <- emmeans(model_sd_24, pairwise ~ sample_type) 
multcomp::cld(est_sd24$emmean, Letters = letters)
```





#other samples
```{r}
egg_spec2 <- identify_samples(egg_spec) %>% 
  mutate(bin = "egg-\nspecific\nloci") %>% 
  filter(sample_type == "seedling shoot" |
          str_detect(sample_ID, "ov_") |
          str_detect(sample_ID, "an_") |
          str_detect(sample_type, "spik")|
          str_detect(sample_type, "DAF"))

seedling_spec2 <- identify_samples(seedling_spec) %>% 
  mutate(bin = "seedling-\nspecific\nloci") %>% 
  filter(sample_type == "seedling shoot" |
          str_detect(sample_ID, "ov_") |
          str_detect(sample_ID, "an_") |
          str_detect(sample_type, "spik")|
          str_detect(sample_type, "DAF"))

sperm_spec2 <- identify_samples(sperm_spec) %>% 
  mutate(bin = "sperm-\nspecific\nloci") %>% 
  filter(sample_type == "seedling shoot" |
          str_detect(sample_ID, "ov_") |
          str_detect(sample_ID, "an_") |
          str_detect(sample_type, "spik")|
          str_detect(sample_type, "DAF"))

intersection2 <- identify_samples(intersection) %>% 
  mutate(bin = "intersect.") %>% 
  filter(sample_type == "seedling shoot" |
          str_detect(sample_ID, "ov_") |
          str_detect(sample_ID, "an_") |
          str_detect(sample_type, "spik")|
          str_detect(sample_type, "DAF"))
```

```{r}
loci_table2 <- rbind(egg_spec2, seedling_spec2, sperm_spec2, intersection2) %>% 
  inner_join(total_siRNA_length1 %>%
               select(-library), by = c("sample_ID", "genotype", "sample_type", "total read", "general siRNA")) %>%
  select(-genotype, -library)

head(loci_table2)
```
```{r}
loci_2_sup <- loci_table2 %>% 
  select(-`20.y`, -`21.y`, -`22.y`, -`23.y`, -`24.y`, -`25.y`) %>% 
  gather("length", "counts", 1:6) %>% 
  mutate(proportion = counts / `general siRNA`) %>% 
  mutate(length = substr(length, start = 0, stop = 2)) %>% 
  select(-counts) %>% 
  mutate(norm = "per total siRNA")
  
  
loci_2_sup
```
```{r}
#female
loci_2_sup %>% 
  mutate(bin2 = case_when(
      bin == "intersect." ~ "intersect.\nloci",
      bin != "intersect." ~ bin
    )) %>% 
  mutate(bin.f = factor(bin2, levels = c(
    "egg-\nspecific\nloci", 
    "seedling-\nspecific\nloci",
    "sperm-\nspecific\nloci",
    "intersect.\nloci"
  ))) %>% 
  filter(sample_type == "seedling shoot" |
          str_detect(sample_ID, "ov_")) %>% 
   mutate(sample_type2 = case_when(
    str_detect(sample_ID, "ov_1st") ~ "pre-meiotic ovary",
    str_detect(sample_ID, "ov_2nd") ~ "meiotic ovary",
    str_detect(sample_ID, "ov_3rd") ~ "functional\nmegaspore ovary",
    str_detect(sample_ID, "ov_4th") ~ "8-nuclei\nembryo sac\novary",
    str_detect(sample_ID, "ov_") == F ~ sample_type, 
  )) %>%
  ggplot(aes(x = length, y = proportion*100 )) +
  facet_grid( norm ~ bin.f, scales = "free_y", switch = "y") +
  geom_vline(data = axis_line, aes(xintercept = length), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type2, color = sample_type2), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type2), position = position_dodge(0.01)) + 
  scale_color_manual(values = c( "seagreen",
                                "orchid1", "brown2", "brown3", "brown4"),
                     limits = c( "seedling shoot", 
    "pre-meiotic ovary", "meiotic ovary", 
    "functional\nmegaspore ovary", "8-nuclei\nembryo sac\novary")) +
  labs(color = NULL,
       x = "length (nt)",
       y = "% siRNA") +
  guides(color = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(panel.spacing.y = unit(1, "lines"), strip.placement = "outside") +
  theme(strip.text.y = element_blank()) +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(legend.margin = margin(0, 0, 0, 0)) +
  theme(legend.box.margin = margin(-12, 0, 0, 0)) + 
  theme(legend.position = "bottom")

ggsave("siRNA_in_24nt_loci_fe.png", width = 7, height = 3.5)
ggsave("siRNA_in_24nt_loci_fe.svg", width = 7, height = 3.5)
```
```{r}
#male
loci_2_sup %>% 
  mutate(bin2 = case_when(
      bin == "intersect." ~ "intersect.\nloci",
      bin != "intersect." ~ bin
    )) %>% 
  mutate(bin.f = factor(bin2, levels = c(
    "egg-\nspecific\nloci", 
    "seedling-\nspecific\nloci",
    "sperm-\nspecific\nloci",
    "intersect.\nloci"
  ))) %>% 
  filter(sample_type == "seedling shoot" |
          str_detect(sample_ID, "an_")) %>% 
   mutate(sample_type2 = case_when(
    sample_type == "bicellular pollen anther" ~ "bicellular\npollen\nanther",
    sample_type == "meiosis anther" ~ "meiotic anther",
    sample_type != "bicellular pollen anther" &
    sample_type != "meiosis anther" ~ sample_type
  )) %>% 
  ggplot(aes(x = length, y = proportion*100 )) +
  facet_grid( norm ~ bin.f, scales = "free_y", switch = "y") +
  geom_vline(data = axis_line, aes(xintercept = length), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type2, color = sample_type2), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type2), position = position_dodge(0.01)) + 
  scale_color_manual(values = c("seagreen",
                               "tan", "tan2", "tan3", "tan4"),
                     limits = c("seedling shoot",
    "pre-meiotic anther",
    "meiotic anther",
    "microspore anther",
    "bicellular\npollen\nanther")) +
  labs(color = NULL,
       x = "length (nt)",
       y = "% siRNA") +
  guides(color = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(panel.spacing.y = unit(1, "lines"), strip.placement = "outside") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(strip.text.y = element_blank()) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(legend.margin = margin(0, 0, 0, 0)) +
  theme(legend.box.margin = margin(-12, 0, 0, 0)) + 
  theme(legend.position = "bottom")

ggsave("siRNA_in_24nt_loci_anther.png", width = 7, height = 3.5)
ggsave("siRNA_in_24nt_loci_anther.svg", width = 7, height = 3.5)
```
```{r}
#spikelets
loci_2_sup %>% 
  mutate(bin2 = case_when(
      bin == "intersect." ~ "intersect.\nloci",
      bin != "intersect." ~ bin
    )) %>% 
  mutate(bin.f = factor(bin2, levels = c(
    "egg-\nspecific\nloci", 
    "seedling-\nspecific\nloci",
    "sperm-\nspecific\nloci",
    "intersect.\nloci"
  ))) %>% 
  filter(str_detect(sample_type, "spike") |
           sample_type == "seedling shoot"
           ) %>%  
   mutate(sample_type2 = case_when(
    str_detect(sample_ID, "S3") ~ "spikelets\nanther stage 3", 
    str_detect(sample_ID, "S5") ~ "spikelets\nanther stage 5",
    str_detect(sample_ID, "S7") ~ "spikelets\nanther stage 7",
    str_detect(sample_type, "spike") == F ~ sample_type
  )) %>% 
  ggplot(aes(x = length, y = proportion*100 )) +
  facet_grid( norm ~ bin.f, scales = "free_y", switch = "y") +
  geom_vline(data = axis_line, aes(xintercept = length), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type2, color = sample_type2), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type2), position = position_dodge(0.01)) + 
   scale_color_manual(values = c("seagreen", 
                                "turquoise1", "turquoise3", "turquoise4"),
                     limits = c("seedling shoot",
    "spikelets\nanther stage 3",
    "spikelets\nanther stage 5",
    "spikelets\nanther stage 7")) +
  labs(color = NULL,
       x = "length (nt)",
       y = "% siRNA") +
  guides(color = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(panel.spacing.y = unit(1, "lines"), strip.placement = "outside") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(strip.text.y = element_blank()) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(legend.margin = margin(0, 0, 0, 0)) +
  theme(legend.box.margin = margin(-12, 0, 0, 0)) + 
  theme(legend.position = "bottom") 

ggsave("siRNA_in_24nt_loci_spik.png", width = 7, height = 3.5)
ggsave("siRNA_in_24nt_loci_spik.svg", width = 7, height = 3.5)
```
```{r}
#em_en
loci_2_sup %>% 
  mutate(bin2 = case_when(
      bin == "intersect." ~ "intersect.\nloci",
      bin != "intersect." ~ bin
    )) %>% 
  mutate(bin.f = factor(bin2, levels = c(
    "egg-\nspecific\nloci", 
    "seedling-\nspecific\nloci",
    "sperm-\nspecific\nloci",
    "intersect.\nloci"
  ))) %>% 
  filter(sample_type == "seedling shoot"|
           sample_type == "embryo 7-8 DAF"|
           sample_type == "endosperm 7-8 DAF") %>%
  mutate(sample_type2 = case_when(
    sample_type == "seedling shoot" ~ "seedling\nshoot",
    str_detect(sample_type, "emb") ~ "embryo\n7-8 DAF",
    str_detect(sample_type, "end") ~ "endosperm\n7-8 DAF",
  )) %>% 
  ggplot(aes(x = length, y = proportion*100 )) +
  facet_grid(norm ~ bin.f, scales = "free_y", switch = "y") +
  geom_vline(data = axis_line, aes(xintercept = length), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type2, color = sample_type2), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type2), position = position_dodge(0.01)) + 
  scale_color_manual(values = c("seagreen", "magenta", "lightgoldenrod4"),
                     limits = c("seedling\nshoot", "embryo\n7-8 DAF", "endosperm\n7-8 DAF")) +
  labs(color = NULL,
       x = "length (nt)",
       y = "% siRNA") +
  guides(color = guide_legend(nrow = 1)) +
  theme_minimal() +
  theme(panel.spacing.y = unit(1, "lines"), strip.placement = "outside") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(strip.text.y = element_blank()) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(legend.margin = margin(0, 0, 0, 0)) +
  theme(legend.box.margin = margin(-12, 0, 0, 0)) + 
  theme(legend.position = "bottom") 

ggsave("siRNA_in_24nt_loci_em_en.png", width = 7, height = 3.5)
ggsave("siRNA_in_24nt_loci_em_en.svg", width = 7, height = 3.5)
```




Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
