---
title: "Shortstack_loci"
author: "Chenxin Li"
date: "3/3/2020"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(RColorBrewer)
library(viridis)
library(stringr)
library(svglite)
library(cowplot)
library(forcats)
library(ggbeeswarm)
```

#load data
```{r}
egg <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_03_03_2020/egg.bam.shortstack/Results.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
```{r}
ovary <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_03_03_2020/ovary.bam.shortstack/Results.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
```{r}
seedling <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_03_03_2020/seedling.bam.shortstack/Results.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
```{r}
sperm <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_03_03_2020/sperm.bam.shortstack/Results.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
```{r}
endosperm <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_03_03_2020/SRR771500.fastq.gz.trimmed.fastq.gz.sorted.bam.shortstack/Results.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
```{r}
zygote <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_03_03_2020/zygote.bam.shortstack/Results.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
```{r}
osa_mirBase22 <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/miRNA/osa_mirBase22.txt", header=FALSE) 
head(osa_mirBase22)
```

#clean data
```{r}
head(zygote)
```
```{r}
clean <- function(df){
  df %>% 
    filter(RPM > 2) %>%  #larger than 2 RPM
    filter(DicerCall == "24") %>%  #24nt siRNA 
    mutate(RPKM = RPM/Length * 1000) %>% 
    separate(`#Locus`, c("Chr", "Start_End"), sep = ":", remove = F) %>% 
    separate(Start_End, c("start", "end"), sep = "-") %>% 
    filter(MIRNA != "Y") %>% #not a miRNA by shortstack
    filter(MajorRNA %in% osa_mirBase22$V2 == F) #major RNA not in miRBase22 
} 
```

```{r}
egg1 <- egg %>% 
  clean() %>% 
  mutate(sample_type = "egg") 

ovary1 <- ovary %>% 
  clean() %>% 
  mutate(sample_type = "ovary") 

seedling1 <- seedling %>% 
  clean() %>% 
  mutate(sample_type = "seedling") 

sperm1 <- sperm %>% 
  clean() %>% 
  mutate(sample_type = "sperm") 

zygote1 <- zygote %>% 
  clean() %>% 
  mutate(sample_type = "zygote") 

endo.1 <- endosperm %>% 
  clean() %>% 
  mutate(sample_type = "endosperm") 

```

```{r}
siRNA_24_loci <- rbind(egg1, ovary1, seedling1, sperm1, zygote1, endo.1) 
  
head(siRNA_24_loci)
```


#cumulative distribution
```{r}
siRNA_24_loci_cummu <- siRNA_24_loci %>% 
  group_by(sample_type) %>% 
  arrange(-RPM) %>% 
  mutate(rank = rank(-RPM, ties.method = "first")) %>%
  mutate(rank = rank/max(rank)) %>% 
  mutate(dist = cumsum(RPM)) %>% 
  mutate(perc_cum_rpm = dist/max(dist) * 100) %>% 
  ungroup() 
```

```{r}
siren <- siRNA_24_loci_cummu %>% 
  filter(sample_type == "ovary") %>% 
  #filter(PhaseScore < 30) %>% 
  filter(perc_cum_rpm <= 75)  

head(siren)
``` 


```{r}
total <- c(nrow(siRNA_24_loci %>% 
                  filter(sample_type == "egg")), 
           nrow(siRNA_24_loci %>% 
                  filter(sample_type == "ovary")),
           nrow(siRNA_24_loci %>% 
                  filter(sample_type == "seedling")),
           nrow(siRNA_24_loci %>% 
                  filter(sample_type == "sperm")),
           nrow(siRNA_24_loci %>% 
                  filter(sample_type == "zygote")),
           nrow(siRNA_24_loci %>% 
                  filter(sample_type == "endosperm"))
           )%>% 
  as.data.frame() %>% 
  mutate(sample_type = c("egg", "ovary", "seedling", "sperm", "zygote", "endosperm"))

total
```
```{r}
x_label <- siRNA_24_loci_cummu %>% 
  group_by(sample_type) %>% 
  #filter(PhaseScore < 30) %>% 
  filter(perc_cum_rpm <= 75) %>% 
  count() %>% 
  full_join(total, by = "sample_type") %>% 
  mutate(rank = n/`.`) 

x_label
```





```{r}
set.seed(666) 
siRNA_24_loci_cummu %>% 
  group_by(sample_type) %>% 
  sample_n(2000) %>% 
  ungroup() %>% 
  #filter(PhaseScore < 30) %>% 
  ggplot(aes(x = rank* 100, y = perc_cum_rpm)) +
  geom_vline(xintercept = 0, size = 1) +
  geom_hline(yintercept = 0, size = 1) +
  geom_line(aes(color = sample_type, group = sample_type), size = 1.25, alpha = 0.8) +
  geom_hline(yintercept = 75, linetype = 2, color = "black", size = 1.1) +
  scale_color_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2", "violetred2", "lightgoldenrod4"),
                     limits = c("ovary", "egg", "seedling", "sperm", "zygote", "endosperm")) +
  labs(x = "rank (%)",
       y = "cumulative\nexpression (%)",
       color = NULL) +
  guides(color = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(legend.position = "bottom") +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) +
  coord_fixed()

ggsave("minus_miR.png", width = 5, height = 5)
ggsave("minus_miR.svg", width = 5, height = 5)
```


```{r}
endo_siren <- siRNA_24_loci_cummu %>% 
  filter(sample_type == "endosperm") %>% 
  filter(perc_cum_rpm <= 75) 

head(endo_siren)
```

#write data
```{r}
write_excel_csv(egg1, "egg_siRNA_loci.csv")
write_excel_csv(sperm1, "sperm_siRNA_loci.csv")
write_excel_csv(seedling1, "seedling_siRNA_loci.csv")
write_excel_csv(ovary1, "ovary_siRNA_loci.csv")
write_excel_csv(endo.1, "endosperm_siRNA_loci.csv")
write_excel_csv(zygote1, "zygote_siRNA_loci.csv")

write_excel_csv(siren, "ovary_siren_loci.csv")
write_excel_csv(endo_siren, "endosperm_siren_loci.csv")
```


#corresponding siRNA loci of sirens
```{r}
siren_simplify <- siren %>% 
  #separate(Name, c("T1", "Name2"), sep = "_") %>% 
  select(Chr, start, end) %>% 
  as.matrix()

colnames(siren_simplify) <- NULL
head(siren_simplify)

match_location <- function(df, r){
  df %>% 
  filter(Chr == r[1]) %>% 
  mutate(anchor = (start + end)/2) %>% 
  filter(anchor > r[2] &
           anchor < r[3]) %>% 
  summarise(
    RPM = sum(RPM),
    Length = sum(Length),
    Chr = min(Chr),
    start = min(start),
    end = max(end) 
  )  
}
```

```{r}
fixer <- function(df){
  df %>% 
    mutate(Chr = as.numeric(Chr)) %>% 
    mutate(start = as.numeric(start)) %>% 
    mutate(end = as.numeric(end)) %>% 
    filter(is.na(Chr) == F) 
}
```

```{r}
egg2 <- fixer(egg1)
zygote2 <- fixer(zygote1)
sperm2 <- fixer(sperm1)
seedling2 <- fixer(seedling1)
endo.2 <- fixer(endo.1)
siren <- fixer(siren)
ovary2 <- fixer(ovary1)
```

```{r}
# apply(FUN = match_location, df = egg2, X = siren_simplify[1:10, ], MARGIN = 1) %>% 
#   bind_rows()

egg2 %>% 
  filter(Chr == 5) %>% 
  mutate(anchor = (start + end)/2) %>%
  filter(anchor > 16430824 &
           anchor < 16431405) 
```

```{r}
ovary_exp <- ovary1 %>% 
  filter(Name %in% siren$Name) %>% 
  select(RPM, Length, Chr, start, end, `#Locus`, Name) 

egg_exp <- apply(FUN = match_location, df = egg2, siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(siren %>% 
              select(`#Locus`, Name)) 

zygote_exp <- apply(FUN = match_location, df = zygote2, siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(siren %>% 
              select(`#Locus`, Name))

sperm_exp <- apply(FUN = match_location, df = sperm2, siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(siren %>% 
              select(`#Locus`, Name))

seedling_exp <- apply(FUN = match_location, df = seedling2, siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(siren %>% 
              select(`#Locus`, Name))

endosperm_exp <- apply(FUN = match_location, df = endo.2, siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(siren %>% 
              select(`#Locus`, Name))
```

```{r}
ovary_siren_cors <- rbind(
  ovary_exp %>% mutate(sample_type = "ovary"), 
  egg_exp %>% mutate(sample_type = "egg"),
  zygote_exp %>% mutate(sample_type = "zygote") ,
  sperm_exp %>% mutate(sample_type = "sperm"),
  seedling_exp %>% mutate(sample_type = "seedling"),
  endosperm_exp %>% mutate(sample_type = "endosperm")
) %>% 
  mutate(RPKM = RPM/Length * 1000) %>% 
  mutate(RPKM = case_when(
    Length == 0 ~ 0,
    T ~ RPKM
  ))

head(ovary_siren_cors)
```
```{r}
ovary_siren_cors %>% 
  filter(sample_type == "egg" |
           sample_type == "zygote") %>% 
  filter(RPM == 0) %>% 
  group_by(sample_type) 
```

 

```{r}
ovary_siren_cors_s <- ovary_siren_cors %>% 
  group_by(sample_type) %>% 
  summarise(
    median = median(RPM + 1),
    IQR = IQR(RPM + 1),
    Q3 = quantile(RPM + 1, 0.75),
    Q1 = quantile(RPM + 1, 0.25),
    UL = quantile(RPM + 1, 0.95),
    LL = quantile(RPM + 1, 0.05)
  ) %>% 
  mutate(sample_type = factor(sample_type, levels = c(
                              "ovary", "egg", 
                              "seedling", "sperm", 
                              "zygote", "endosperm")))  

head(ovary_siren_cors_s) 
```

```{r}
ovary_siren_cors_s_RPKM <- ovary_siren_cors %>% 
  group_by(sample_type) %>% 
  summarise(
    median = median(RPKM),
    IQR = IQR(RPKM),
    Q3 = quantile(RPKM, 0.75),
    Q1 = quantile(RPKM, 0.25),
    UL = quantile(RPKM, 0.975),
    LL = quantile(RPKM, 0.025)
  )

head(ovary_siren_cors_s_RPKM)
```
```{r}
pairwise.wilcox.test(x = ovary_siren_cors$RPM, g = ovary_siren_cors$sample_type, paired = F, p.adjust.method = "bon")
```
```{r}
model_siren <- lm(log10(RPM + 1) ~ sample_type, data = ovary_siren_cors)
anova(model_siren)
est_siren <- emmeans(model_siren, pairwise ~ sample_type)
multcomp::cld(est_siren$emmeans)
est_siren$contrasts
```

```{r}
ovary_siren_cors_s %>% 
  #filter(sample_type != "endosperm") %>% 
  #mutate(sample_type = reorder(sample_type, median)) %>% 
  ggplot(aes(x = sample_type)) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median, fill = sample_type), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = sample_type, xend = sample_type,  y = Q3, yend = UL)) +
  geom_segment(aes(x = sample_type, xend = sample_type,  y = Q1, yend = LL)) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) +
  geom_text(data = data.frame(
    "text" = c("d", "a", "e", "b", "c", "d"),
    "sample_type" = ovary_siren_cors_s$sample_type,
    "UL" = ovary_siren_cors_s$UL
  ), aes(label = text, y = UL), size = 5, fontface = "bold", vjust = -0.3) +
  scale_fill_manual(values = c("orangered3", "tomato1", 
                               "seagreen", "dodgerblue2", 
                               "violetred2", "lightgoldenrod3"),
                   limits = c("ovary", "egg", 
                              "seedling", "sperm", 
                              "zygote", "endosperm")) +
  scale_x_discrete(labels = NULL) + 
  scale_y_log10(breaks = c(10, 100, 1000)) +
  labs(x = NULL,
       y = "ovary siren\nRPM",
       fill = NULL,
       color = NULL) +  
   guides(fill = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(legend.position = "bottom") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black")) +
  theme(panel.grid = element_blank()) +
  annotation_logticks(side = "l") 

ggsave("ovary_siren_RPM.svg", height = 5, width = 5)
ggsave("ovary_siren_RPM.png", height = 5, width = 5)
```
 
##endo. siren euivalents  
```{r}
endo.siren_simplify <- endo_siren %>% 
  select(Chr, start, end) %>% 
  as.matrix()

colnames(endo.siren_simplify) <- NULL
head(endo.siren_simplify)
```

```{r}
endo_siren_exp <- endo.1 %>% 
  filter(Name %in% endo_siren$Name) %>% 
  select(RPM, Length, Chr, start, end, `#Locus`, Name)  

endo_egg_exp <- apply(FUN = match_location, df = egg2, endo.siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(endo_siren %>% 
              select(`#Locus`, Name)) 

endo_zygote_exp <- apply(FUN = match_location, df = zygote2, endo.siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(endo_siren %>% 
              select(`#Locus`, Name))

endo_sperm_exp <- apply(FUN = match_location, df = sperm2, endo.siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(endo_siren %>% 
              select(`#Locus`, Name))

endo_seedling_exp <- apply(FUN = match_location, df = seedling2, endo.siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(endo_siren %>% 
              select(`#Locus`, Name))


endo_ovary_exp <- apply(FUN = match_location, df = ovary2, endo.siren_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(endo_siren %>% 
              select(`#Locus`, Name))
```

```{r}
summary(endo_siren_exp$RPM)
```

```{r}
endo_siren_cors <- rbind(
  endo_ovary_exp %>% mutate(sample_type = "ovary"), 
  endo_egg_exp %>% mutate(sample_type = "egg"),
  endo_zygote_exp %>% mutate(sample_type = "zygote") ,
  endo_sperm_exp %>% mutate(sample_type = "sperm"),
  endo_seedling_exp %>% mutate(sample_type = "seedling"),
  endo_siren_exp %>% mutate(sample_type = "endosperm")
) %>% 
  mutate(RPKM = RPM/Length * 1000) %>% 
  mutate(RPKM = case_when(
    Length == 0 ~ 0,
    T ~ RPKM
  ))

head(endo_siren_cors)
```
```{r}
endo_siren_cors_s <- endo_siren_cors %>% 
  group_by(sample_type) %>% 
  summarise(
    median = median(RPM + 1),
    IQR = IQR(RPM + 1),
    Q3 = quantile(RPM + 1, 0.75),
    Q1 = quantile(RPM + 1, 0.25),
    UL = quantile(RPM + 1, 0.95),
    LL = quantile(RPM + 1, 0.05)
  ) %>% 
  mutate(sample_type = factor(sample_type, levels = c(
                              "ovary", "egg", 
                              "seedling", "sperm", 
                              "zygote", "endosperm")))  

head(endo_siren_cors_s)
```
```{r}
model_endo_siren <- lm(log10(RPM + 1) ~ sample_type, data = endo_siren_cors)
#plot(model_endo_siren)
anova(model_endo_siren)
```

```{r}
est_endo_siren <- emmeans(model_endo_siren, pairwise ~ sample_type)
multcomp::cld(est_endo_siren$emmeans)
est_endo_siren$contrasts
```

```{r}
endo_siren_cors_s %>% 
  ggplot(aes(x = sample_type)) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median, fill = sample_type), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = sample_type, xend = sample_type,  y = Q3, yend = UL)) +
  geom_segment(aes(x = sample_type, xend = sample_type,  y = Q1, yend = LL)) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) +
  geom_text(data = data.frame(
    "text" = c("b", "c", "a", "b", "b", "a"),
    "sample_type" = endo_siren_cors_s$sample_type,
    "UL" = endo_siren_cors_s$UL
  ), aes(label = text, y = UL), size = 5, fontface = "bold", vjust = -0.3) +
  scale_fill_manual(values = c("orangered3", "tomato1", 
                               "seagreen", "dodgerblue2", 
                               "violetred2", "lightgoldenrod3"),
                   limits = c("ovary", "egg", 
                              "seedling", "sperm", 
                              "zygote", "endosperm")) +
  scale_x_discrete(labels = NULL) + 
  scale_y_log10(breaks = c(10, 100, 1000)) +
  labs(x = NULL,
       y = "endosperm siren\nRPM",
       fill = NULL,
       color = NULL) +  
   guides(fill = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(legend.position = "bottom") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black")) +
  theme(panel.grid = element_blank()) +
  annotation_logticks(side = "l") 

ggsave("endo_siren_RPM.svg", height = 5, width = 5)
ggsave("endo_siren_RPM.png", height = 5, width = 5)
```

```{r}
ovary_siren_cors_s %>% 
  mutate(type = "ovary siren") %>% 
  mutate(letter = case_when(
    sample_type == "ovary" ~ "e",
    sample_type == "egg" ~ "d",
    sample_type == "zygote" ~ "d",
    sample_type == "sperm" ~ "c",
    sample_type == "seedling" ~ "b",
    sample_type == "endosperm" ~ "a",
  )) %>% 
  rbind(endo_siren_cors_s %>% 
          mutate(type = "endosperm siren") %>% 
          mutate(letter = case_when(
    sample_type == "ovary" ~ "a",
    sample_type == "egg" ~ "b",
    sample_type == "zygote" ~ "a",
    sample_type == "sperm" ~ "b",
    sample_type == "seedling" ~ "b",
    sample_type == "endosperm" ~ "c",
          ))) %>% 
  ggplot(aes(x = sample_type)) +
  facet_grid(. ~ type) + 
  # geom_vline(xintercept = -Inf, size = 1.2) +
  # geom_hline(yintercept = 0, size = 1.2) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median, fill = sample_type), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = sample_type, xend = sample_type,  y = Q3, yend = UL)) +
  geom_segment(aes(x = sample_type, xend = sample_type,  y = Q1, yend = LL)) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) +
  geom_text(aes(label = letter, y = UL), size = 5, fontface = "bold", vjust = -0.3) +
  scale_fill_manual(values = c("orangered3", "tomato1", 
                               "seagreen", "dodgerblue2", 
                               "violetred2", "lightgoldenrod3"),
                   limits = c("ovary", "egg", 
                              "seedling", "sperm", 
                              "zygote", "endosperm")) +
  scale_x_discrete(labels = NULL) + 
  scale_y_log10(breaks = c(10, 100, 1000)) +
  labs(x = NULL,
       y = "RPM",
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(legend.position = "bottom") + 
  theme(axis.line = element_line(size = 1)) +
  theme(panel.spacing.x = unit(1, "lines")) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black")) +
  theme(panel.grid = element_blank())  

ggsave("siren_RPM.svg", height = 5, width = 5)
ggsave("siren_RPM.png", height = 5, width = 5)
```



#phase score distribution
```{r}
head(siren)
```

```{r}
all <- siRNA_24_loci_cummu %>% 
  rbind(siren %>% mutate(sample_type = "ovary siren")) %>% 
  rbind(endo_siren %>% mutate(sample_type = "endo. siren"))

all_s <- all %>%  
  group_by(sample_type) %>% 
  summarise(median.PS = median(PhaseScore),
            Q3 = quantile(PhaseScore, 0.75),
            Q1 = quantile(PhaseScore, 0.25),
            UL = quantile(PhaseScore, 0.975),
            LL = quantile(PhaseScore, 0.025), 
            n = n())

all_s
```
```{r}
siRNA_24_loci_cummu %>% 
  ggplot(aes(x = PhaseScore, y = RPM)) +
  facet_wrap(~ sample_type, scales = "free") +
  geom_point(alpha = 0.5) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1.2)) 
```
 

```{r}
all_s %>% 
  mutate(sample_type = factor(sample_type, levels = c(
    "ovary", "egg", "seedling", "sperm", "zygote", "endo.", "ovary siren", "endo. siren"
  ))) %>% 
  ggplot(aes(x = sample_type)) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median.PS, fill = sample_type), alpha = 0.8, width = 0.7) +
  geom_segment(aes(y = Q3, yend = UL, x = sample_type, xend = sample_type), size = 0.75) +
  geom_segment(aes(y = Q1, yend = LL, x = sample_type, xend = sample_type), size = 0.75) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) + 
  geom_hline(yintercept = 30, linetype = 2, size = 1) +
  scale_fill_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2", "violetred2", "lightgoldenrod4", "orangered3", "lightgoldenrod4"),
                    limits = c( "ovary", "egg", "seedling", "sperm", "zygote", "endo.", "ovary siren", "endo. siren")) +
  labs(x = NULL,
       y = "phasing score") +
  theme_minimal() +
  theme(axis.line = element_line(size = 1.25)) +
  theme(legend.position = "none") +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black", hjust = 0.5)) +
  coord_flip()   

ggsave("PhaseScore.svg", width = 5, height = 5)
ggsave("PhaseScore.png", width = 5, height = 5)
```
```{r}
all_length_s <- all %>% 
  group_by(sample_type) %>% 
  summarise(median = median(Length),
            Q3 = quantile(Length, 0.75),
            Q1 = quantile(Length, 0.25),
            UL = quantile(Length, 0.975),
            LL = quantile(Length, 0.025), 
            n = n()) %>% 
  mutate(sample_type = factor(sample_type, levels = c(
    "ovary", "egg", "seedling", "sperm", "zygote", "endo.", "ovary siren", "endo. siren"
  ))) %>% 
  mutate(text.y = case_when(
    str_detect(sample_type, "endo. siren") ~ Q3,
    T ~ UL
  ))

all_length_s
```

```{r}
all_length_s %>% 
  ggplot(aes(x = sample_type %>% reorder(median))) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = sample_type), alpha = 0.8, width = 0.7) +
  geom_segment(aes(y = Q3/1000, yend = UL/1000, x = sample_type, xend = sample_type), size = 0.75) +
  geom_segment(aes(y = Q1/1000, yend = LL/1000, x = sample_type, xend = sample_type), size = 0.75) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) +
  geom_text(aes(label = paste("median = ", signif(median/1000, 2), " kb", sep = ""), 
                x = sample_type, y = text.y/1000),
                vjust = -0.2,  hjust = -0.05, size = 5, fontface = "bold") +
  scale_fill_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2", "violetred2", "lightgoldenrod4", "orangered3", "lightgoldenrod4"),
                    limits = c( "ovary", "egg", "seedling", "sperm", "zygote", "endo.", "ovary siren", "endo. siren")) +
  labs(x = "siRNA loci",
       y = "length (kb)") +
  theme_minimal() +
  theme(axis.line = element_line(size = 1.25)) +
  theme(legend.position = "none") +
  theme(text = element_text(size = 18, face="bold")) + 
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("siren_length_box.svg", height = 5, width = 6)
ggsave("siren_length_box.png", height = 5, width = 6)
```



##adjusting for length 
```{r}
endosperm_2 <- endo.1 %>% 
  mutate(type2 = case_when(
    Name %in% endo_siren$Name ~ "endo.\nsiren",
    T ~ "endo.\nnon-siren"
    ))
```

```{r}
ovary_2 <- ovary1 %>% 
  mutate(type2 =case_when(
    Name %in% siren$Name ~ "ovary\nsiren",
    T ~ "ovary\nnon-siren"
    ))  
```

```{r}
ov.en.2 <- rbind(endosperm_2, ovary_2)
head(ov.en.2)
```
```{r}
ov.en.2.s <- ov.en.2 %>% 
  group_by(type2, sample_type) %>% 
  mutate(logRPKM = log10(RPKM + 1)) %>% 
  summarise(
    median = median(logRPKM),
            Q3 = quantile(logRPKM, 0.75),
            Q1 = quantile(logRPKM, 0.25),
            UL = quantile(logRPKM, 0.975),
            LL = quantile(logRPKM, 0.025), 
            n = n(),
    mean2 = mean(RPKM)
  ) %>%
  ungroup() %>% 
  mutate(type2 = factor(type2, levels = c(
    "endo.\nsiren",
    "endo.\nnon-siren",
    "ovary\nsiren",
    "ovary\nnon-siren"
  )))

ov.en.2.s
```

```{r}
ov.en.2.s %>% 
  ggplot(aes(x = type2)) +
   geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median, fill = sample_type), alpha = 0.8, width = 0.7) +
  geom_segment(aes(y = Q3, yend = UL, x = type2, xend = type2), size = 0.75) +
  geom_segment(aes(y = Q1, yend = LL, x = type2, xend = type2), size = 0.75) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) +
   scale_fill_manual(values = c("orangered3",  "lightgoldenrod4"),
                    limits = c( "ovary", "endo.")) +
  labs(x = NULL,
       y = "siRNA\nlog10(RPKM + 1)") +
  theme_minimal() +
  theme(axis.line = element_line(size = 1.25)) +
  theme(legend.position = "none") +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("siren_RPKM_box.svg", height = 5, width = 4)
ggsave("siren_RPKM_box.png", height = 5, width = 4)
```



#miRNAs
```{r}
osa_mir <- osa_mirBase22[, 1:2] 
colnames(osa_mir) <- c("tag", "sequence")  
osa_mir <- osa_mir%>% 
  separate(tag, c("T1", "T2", "T3", "T4", "T5"), sep = " ", remove = T) %>% 
  mutate(miR = paste("osa", T5, sep = "-")) %>% 
  select(-T1, -T3, -T4, -T5) %>% 
  arrange(T2) 

osa_mir 
```

```{r}
find_miR <- function(df){
  df %>% 
    separate(`#Locus`, c("Chr", "Start_End"), sep = ":", remove = F) %>% 
    separate(Start_End, c("start", "end"), sep = "-") %>% 
    filter(MIRNA == "Y") %>%
    left_join(osa_mir, by = c("MajorRNA" = "sequence")) %>% 
    distinct(Name, .keep_all = T)
}
```

 

```{r}
egg_miR <- find_miR(egg)
ovary_miR <- find_miR(ovary)
seedling_miR <- find_miR(seedling)
sperm_miR <- find_miR(sperm)
zygote_miR <- find_miR(zygote)
endo_miR <- find_miR(endosperm)
```
```{r}
egg_miR
```

##find specific
```{r}
data1 <- egg_miR
data2 <- seedling_miR
data3 <- sperm_miR
data4 <- zygote_miR
```

```{r}
find_specific <- function(set1, against){
  set1 %>% 
    filter(MajorRNA %in% against$MajorRNA == F) #In your data, the col name won't be "element". Change it here. 
}
```

```{r}
data1_specific <- data1 %>% 
  find_specific(against = rbind(data2, data3, data4))
data2_specific <- data2 %>% 
  find_specific(against = rbind(data1, data3, data4))
data3_specific <- data3 %>% 
  find_specific(against = rbind(data1, data2, data4))
data4_specific <- data4 %>% 
  find_specific(against = rbind(data1, data2, data3))
```

```{r}
ndata1_spec <- data1_specific %>% nrow()
ndata2_spec <- data2_specific %>% nrow()
ndata3_spec <- data3_specific %>% nrow()
ndata4_spec <- data4_specific %>% nrow()
```

##find overlaps
```{r}
find_2way <- function(set1, set2){
  set1 %>% 
    filter(set1$MajorRNA %in% set2$MajorRNA)  #In your data, the col name won't be "element". Change it here. 
  
} 
```

```{r}
data1by2 <- find_2way(data1, data2)
data1by3 <- find_2way(data1, data3)
data1by4 <- find_2way(data1, data4)
data2by3 <- find_2way(data2, data3)
data2by4 <- find_2way(data2, data4)
data3by4 <- find_2way(data3, data4)
```


```{r}
ndata1by2 <- nrow(data1by2)
ndata1by3 <- nrow(data1by3)
ndata1by4 <- nrow(data1by4)
ndata2by3 <- nrow(data2by3)
ndata2by4 <- nrow(data2by4)
ndata3by4 <- nrow(data3by4)
```

```{r}
find_3way <- function(set1, set2, set3){
  set1 %>% 
    filter(set1$MajorRNA %in% set2$MajorRNA) %>%  #In your data, the col name won't be "element". Change it here. 
    filter(MajorRNA %in% set3$MajorRNA) 
}
```

```{r}
data1by2by3 <- find_3way(data1, data2, data3)
data2by3by4 <- find_3way(data2, data3, data4)
data3by4by1 <- find_3way(data3, data4, data1)
data4by1by2 <- find_3way(data4, data1, data2)
```

```{r}
ndata1by2by3 <- data1by2by3 %>% nrow()
ndata2by3by4 <- data2by3by4 %>% nrow()
ndata3by4by1 <- data3by4by1 %>% nrow()
ndata4by1by2 <- data4by1by2 %>% nrow()
```

```{r}
data1by2by3by4 <- data1 %>% 
  filter(MajorRNA %in% data2$MajorRNA) %>% 
  filter(MajorRNA %in% data3$MajorRNA) %>% 
  filter(MajorRNA %in% data4$MajorRNA)

ndata1by2by3by4 <- data1by2by3by4 %>% nrow()
```

##total
```{r}
total <- data.frame(
  "sets" = c("data1", "data2", "data3", "data4"),
  "sizes" = c(nrow(data1), nrow(data2), nrow(data3), nrow(data4))
)
total #this will be the basis for the upperleft bar plot 
```

```{r}
upperleft <- total %>% 
  ggplot(aes(x = sets, y= sizes)) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_bar(stat = "identity", aes(fill = sets), alpha = 0.8) +
  geom_text(aes(label = as.character(sizes)), size = 5, angle = 90, hjust = 0, y = 1, fontface = "bold") +
  scale_fill_manual(values = c( "tomato1", "seagreen", "dodgerblue2", "violetred2"), #this is my own custom palette  
                     limits = c("data1", "data2", "data3", "data4")) +              #feel free to use something else     
  scale_x_discrete(labels = NULL) +
  scale_y_continuous(labels = NULL) +
  labs(x = NULL,
       y = "set size") +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(panel.grid = element_blank()) 

upperleft
```

##overlap matrix
```{r}
sets <- c("1", "2", "3", "4")
category <- c(
    "data1_specific",
    "data2_specific",
    "data3_specific",
    "data4_specific",
    "data1by2", 
    "data1by3",
    "data1by4",
    "data2by3",
    "data2by4",
    "data3by4",
    "data1by2by3",
    "data2by3by4",
    "data3by4by1",
    "data4by1by2",
    "data1by2by3by4" 
    )
overlap_matrix <- expand.grid(  #generate all the combinations 
  "sets" = sets,
  "category" = category
) %>% 
  as.data.frame()  %>%    #determine the intersection: a character col of Y or N. 
  mutate(intersect = case_when(
    str_detect(category, sets %>% as.character()) ~ "Y",
    T ~ "N"
  ))  
overlap_matrix
```
```{r}
lowerleft <- overlap_matrix %>% 
  mutate(category = factor(rev(category))) %>% #in the colored matrix the first y value apprears in the bottom, 
                                               #so the order need to be reversed 
  ggplot(aes(x = sets, y = category))+
  geom_tile(aes(fill = sets, alpha = intersect), color = "black", size = 1.5) +
  scale_fill_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2"),
                    limits = c("1", "2", "3", "4")) +
  scale_alpha_manual(values = c(0.8, 0),  #color the grid for Y, don't color for N. 
                     limits = c("Y", "N")) +
  scale_y_discrete(labels = NULL) +
  scale_x_discrete(labels = rep(" ", length(sets))) + #I left white space here for better alignment w/ extended plots 
  labs(x = " ", #white space for better alignment w/ right side plots 
       y = "overlap") +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(panel.grid = element_blank())

lowerleft
```
##legend 
```{r}
upperright <- get_legend(
  total %>% 
    mutate(sample_type = case_when( #make a new column call sample type 
      sets == "data1" ~ "ovary",     #make dataset back to its cell type 
      sets == "data2" ~ "egg",
      sets == "data3" ~ "seedling",
      sets == "data3" ~ "sperm"
    )) %>% 
    mutate(sample_type = factor(sample_type, levels = c(
      "ovary", "egg", "seedling", "sperm"   #order the cell types so that they will line up w/ other plots 
    ))) %>% 
  ggplot(aes(x = sample_type, y= sizes)) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_bar(stat = "identity", aes(fill = sets), alpha = 0.8) +
  geom_text(aes(label = as.character(sizes)), size = 5, angle = 90, hjust = 0, y = 1, fontface = "bold") +
  scale_fill_manual(values = c( "tomato1", "seagreen", "dodgerblue2", "violetred2"), #this is my own custom palette  
                     limits = c("egg", "seedling", "sperm", "zygote")) +              #feel free to use something else     
  scale_x_discrete(labels = NULL) +
  scale_y_continuous(labels = NULL) +
  labs(x = NULL,
       y = "set size",
       fill = "miRNA sets") +
  theme_minimal() +
  theme(legend.position = "right") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(panel.grid = element_blank()) 
)

plot_grid(upperright)
```
##lowerright
```{r}
overlap_sizes <- data.frame(
  "overlap_sizes" = c(ndata1_spec, ndata2_spec, ndata3_spec, ndata4_spec, 
                      ndata1by2, ndata1by3, ndata1by4, ndata2by3, ndata2by4, ndata3by4, 
                      ndata1by2by3, ndata2by3by4, ndata3by4by1, ndata4by1by2,
                      ndata1by2by3by4),
  "category" = category 
) %>% 
  mutate(category = factor(category, levels = rev(category)))   #again order needs to be reversed 
overlap_sizes #this will be the basis for bar plot at the lower right
```

```{r}
lowerright <- overlap_sizes %>% 
  ggplot(aes(x = category, y = overlap_sizes)) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_bar(stat = "identity", fill = "grey80", color = NA, alpha = 0.8) +
  geom_text(aes(label = overlap_sizes, y = 0), size = 5, hjust = 0, vjust = 0.5, fontface = "bold") +
  scale_y_continuous(breaks = c(0, max(overlap_sizes$overlap_sizes)) ,
                     labels = rep(" ", 2)) + #I left white space here for better alignment w/ extended plots
  scale_x_discrete(labels = NULL) +
  labs(y = "intersect. sizes",
       x = NULL) +
  theme_minimal() +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(panel.grid = element_blank()) + 
  coord_flip()

lowerright
```
##all upset
```{r}
plot_grid(upperleft, upperright, lowerleft, lowerright, 
          nrow = 2, 
          ncol = 2,
          rel_heights = c(1, 4), #the more rows in the lower part, the longer it should be
          rel_widths = c(1, 0.8))

ggsave("upset_miRNA_shortstack.svg", height = 7, width = 3.5)  
ggsave("upset_miRNA_shortstack.png", height = 7, width = 3.5)
```
```{r}
seedling_zygote_only <- data2by4 %>% 
  filter(MajorRNA %in% data1$MajorRNA == F) %>% 
  filter(MajorRNA %in% data3$MajorRNA == F)


```


Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
