---
title: "ovary siren vs non ovary siren"
author: "Chenxin Li"
date: "5/18/2020"
output: html_notebook
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(emmeans)
library(RColorBrewer)
library(stringr)
library(svglite)
library(forcats)
library(patchwork)
library(cowplot)
```

```{r}
set.seed(666)
```

#data
##siren loci 
```{r}
temp2 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/ovary_siren_meth/48_26em_S48_L005_R1_001.fastq.gz.trimmed.fastq.gz.filtered.fastq.gz.sorted.bam_ovary_siren_loci.meth", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE, skip = 1)

head(temp2)

temp2 %>% 
  mutate(length = X5 - X4) %>% 
  summarise(length = sum(length))

3717028 / 1463750
```


```{r}
siren_list <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/ovary_siren_meth_ovary_siren", 
                         pattern = "*.meth", full.names = T)
```

```{r}
siren_meth <- sapply(siren_list, read_delim, simplify = FALSE, 
                    delim = "\t", 
                    col_names = FALSE, 
                    escape_double = FALSE, 
                    trim_ws = TRUE,
                    skip = 1) %>% 
  bind_rows(.id = "id")  

head(siren_meth)
```

##reduced non ovary siren siren 
```{r}
non_siren_list <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/methylation/reduced_non_ovary_siren_meth", 
                         pattern = "*.meth", full.names = T)
```

```{r}
non_siren_meth <- sapply(non_siren_list, read_delim, simplify = FALSE, 
                    delim = "\t", 
                    col_names = FALSE, 
                    escape_double = FALSE, 
                    trim_ws = TRUE,
                    skip = 1) %>% 
  bind_rows(.id = "id")  

head(non_siren_meth)
```


#arrange data 
##ovary siren 
```{r}
siren_meth_clean <- siren_meth %>% 
  mutate(Chr = X1, 
         start = X4, 
         end = X5, 
         mCG = X10, 
         mCHG = X11, 
         mCHH = X12) %>% 
  dplyr::select(id, Chr, start, end, mCG, mCHG, mCHH, X18, X19) %>% 
  filter(X18 > 0.5) %>% 
  filter(X19 > 2) %>% 
  # mutate(siren_type = case_when(
  #   str_detect(id, "endosperm_siren") ~ "endo. siren",
  #   str_detect(id, "ovary_siren") ~ "ovary siren"
  # )) %>% 
  mutate(sample_type = case_when(
    str_detect(id, "26en|35en|36en|38en|39en|40en|wt1_en|wt2_en|wt3_en") ~ "endo.",
    str_detect(id, "em_") ~ "embryo", 
    str_detect(id, "br1|br2|br3|br4|br5|br6") ~ "bract",
    str_detect(id, "ov1|ov2|ov3|ov4|ov5|ov6") ~ "ovary",
    str_detect(id, "egg100_1") ~ "egg_1",
    str_detect(id, "egg100_2") ~ "egg_2",
    str_detect(id, "sperm1") ~ "sperm",
    str_detect(id, "EC") ~ "egg\nPark et al",
    str_detect(id, "SRR35031") ~ "leaf",
    str_detect(id, "SRR8594858|SRR8594860") ~ "sperm\nKim et al",
    str_detect(id, "SRR8594859|SRR8594861") ~ "pollen VC\nKim et al"
  )) %>% 
  mutate(genotype = case_when(
    str_detect(id, "48_") ~ "drm2-48", 
    str_detect(id, "52_38|52_39|52_40") ~ "drm2",
    str_detect(id, "SRR8594860|SRR8594861") ~ "ros1a/+",
    str_detect(id, "SRR3503142") ~ "drm2",
    str_detect(id, "SRR3503134") ~ "ddm1",
    T ~ "WT"
  )) %>% 
  gather("context", "methylation", c(mCG, mCHG, mCHH)) %>% 
  filter(is.na(methylation) == F) 

head(siren_meth_clean)
```
```{r}
siren_meth_clean %>% 
  group_by(sample_type, genotype) %>% 
  count()
```

```{r}
##average across reps for conventional libs
siren_meth_clean_s <- siren_meth_clean %>% 
  group_by(sample_type, genotype, Chr, start, context) %>% 
  summarise(methylation = mean(methylation)) %>% 
  ungroup() 

head(siren_meth_clean_s)
```
```{r}
siren_meth_clean_s %>% 
  group_by(sample_type, genotype) %>% 
  count()
```
```{r}
##calculate the quartiles
siren_meth_clean_ss <- siren_meth_clean_s %>% 
  group_by(sample_type, genotype, context) %>% 
  summarise(median = median(methylation),
            Q1 = quantile(methylation, 0.25),
            Q3 = quantile(methylation, 0.75),
            UL = quantile(methylation, 0.975),
            LL = quantile(methylation, 0.025)) %>% 
  ungroup() %>% 
  mutate(class = case_when(
    genotype == "WT" ~ sample_type,
    genotype != "WT" ~ paste(genotype, sample_type, sep = "\n") 
  )) %>% 
  filter(genotype != "drm2-48") 
  

head(siren_meth_clean_ss)
```

##reduced non_siren
```{r}
non_siren_meth_clean <- non_siren_meth %>% 
  mutate(Chr = X1, 
         start = X4, 
         end = X5, 
         mCG = X9, 
         mCHG = X10, 
         mCHH = X11) %>% 
  dplyr::select(id, Chr, start, end, mCG, mCHG, mCHH, X17, X18) %>% 
  filter(X17 > 0.5) %>% 
  filter(X18 > 2) %>% 
  # mutate(siren_type = case_when(
  #   str_detect(id, "endosperm_siren") ~ "endo. siren",
  #   str_detect(id, "ovary_siren") ~ "ovary siren"
  # )) %>% 
  mutate(sample_type = case_when(
    str_detect(id, "26en|35en|36en|38en|39en|40en|wt1_en|wt2_en|wt3_en") ~ "endo.",
    str_detect(id, "em_") ~ "embryo", 
    str_detect(id, "br1|br2|br3|br4|br5|br6") ~ "bract",
    str_detect(id, "ov1|ov2|ov3|ov4|ov5|ov6") ~ "ovary",
    str_detect(id, "egg100_1") ~ "egg_1",
    str_detect(id, "egg100_2") ~ "egg_2",
    str_detect(id, "sperm1") ~ "sperm",
    str_detect(id, "EC") ~ "egg\nPark et al",
    str_detect(id, "SRR35031") ~ "leaf",
    str_detect(id, "SRR8594858|SRR8594860") ~ "sperm\nKim et al",
    str_detect(id, "SRR8594859|SRR8594861") ~ "pollen VC\nKim et al"
  )) %>% 
  mutate(genotype = case_when(
    str_detect(id, "48_") ~ "drm2-48", 
    str_detect(id, "52_38|52_39|52_40") ~ "drm2",
    str_detect(id, "SRR8594860|SRR8594861") ~ "ros1a/+",
    str_detect(id, "SRR3503142") ~ "drm2",
    str_detect(id, "SRR3503134") ~ "ddm1",
    T ~ "WT"
  )) %>% 
  gather("context", "methylation", c(mCG, mCHG, mCHH)) %>% 
  filter(is.na(methylation) == F) 

head(non_siren_meth_clean)
```

```{r}
non_siren_meth_clean_s <- non_siren_meth_clean %>% 
  group_by(sample_type, genotype, Chr, start, context) %>% 
  summarise(methylation = mean(methylation)) %>% 
  ungroup() 

head(non_siren_meth_clean_s)
```
 



```{r}
non_siren_meth_clean_ss <- non_siren_meth_clean_s %>% 
  #rbind(leaf_clean) %>% 
  group_by(sample_type, genotype, context) %>% 
  summarise(median = median(methylation),
            Q1 = quantile(methylation, 0.25),
            Q3 = quantile(methylation, 0.75),
            UL = quantile(methylation, 0.975),
            LL = quantile(methylation, 0.025)) %>% 
  ungroup() %>% 
  mutate(class = case_when(
    genotype == "WT" ~ sample_type,
    genotype != "WT" ~ paste(genotype, sample_type, sep = "\n") 
  )) %>% 
  filter(genotype != "drm2-48") 
  

head(non_siren_meth_clean_ss)
```


#stat
##across siren
```{r}
logit <- function(p){log(
  p / (1-p)
)}
invlogit <- function(x){
  1/(1 + exp(-x))
}
```

```{r}
model_siren <- glm(methylation ~ context * class, family = binomial(link = "logit"), data = siren_meth_clean_s %>% 
                     filter(str_detect(genotype, "ros") == F) %>% 
                     filter(str_detect(genotype, "48") == F) %>% 
                     mutate(class = case_when(
    genotype == "WT" ~ sample_type,
    genotype != "WT" ~ paste(genotype, sample_type, sep = "\n") 
  )))

```
```{r}
summary(model_siren)
```

```{r}
est_siren <- emmeans(model_siren, pairwise ~ class | context)
multcomp::cld(est_siren$emmeans, Letters = letters)
```

##siren vs. non siren
```{r}
data_comb <- siren_meth_clean_s %>%
  mutate(siren_type = "ovary siren") %>% 
  rbind(non_siren_meth_clean_s %>% 
          mutate(siren_type = "other"))
```

```{r}
model_paired <- glm(methylation ~ context * siren_type* class, family = binomial(link = "logit"), data = data_comb %>% 
                     filter(str_detect(genotype, "ros") == F) %>% 
                     filter(str_detect(genotype, "48") == F) %>% 
                     mutate(class = case_when(
    genotype == "WT" ~ sample_type,
    genotype != "WT" ~ paste(genotype, sample_type, sep = "\n") 
  )))

summary(model_paired)
```
```{r}
est_paired <- emmeans(model_paired, pairwise ~  siren_type | class | context)
multcomp::cld(est_paired$emmeans, Letters = letters)
```
```{r}
paired_P <- est_paired$contrasts %>% 
  as.data.frame() %>% 
  mutate(adjusted.p = p.adjust(p.value, method = "bon"))

paired_P %>% 
  filter(context == "mCHH") %>% 
  arrange(p.value)
```
```{r}
ss_wide <- non_siren_meth_clean_ss %>% 
  full_join(siren_meth_clean_ss, by = c("sample_type", "genotype", "context", "class")) %>% 
  mutate(diff = median.y - median.x) %>%
  mutate(FC = median.y / median.x) %>% 
  group_by(context) %>% 
  arrange(-diff) %>% 
  ungroup()

ss_wide %>% filter(context == "mCHH")
```


#plot
```{r}
#remove ros1a stuff 
levels_1 <- c("leaf", "ddm1\nleaf", "drm2\nleaf",
            "bract", "ovary", 
            "egg_1", "egg_2", "egg\nPark et al",
            "sperm", "sperm\nKim et al", "pollen VC\nKim et al", 
            "embryo", "drm2\nembryo", 
            "endo.", "drm2\nendo.") 
```

```{r}
ss_comb <- siren_meth_clean_ss %>%
  mutate(siren_type = "ovary siren") %>% 
  rbind(non_siren_meth_clean_ss %>% 
          mutate(siren_type = "other"))

head(ss_comb)
```


```{r}
ss_comb %>% 
  filter(str_detect(genotype, "ros") == F) %>% 
  mutate(class = factor(class, levels = levels_1)) %>% 
  filter(context == "mCHH") %>% 
  mutate(sample_type2 = case_when(
    str_detect(sample_type, "egg") ~ "egg",
    str_detect(sample_type, "sperm") ~ "sperm", 
    str_detect(sample_type, "pollen") ~ "pollen",
    T ~ sample_type
  )) %>%  
  ggplot(aes(x = siren_type, y = median)) + 
  facet_wrap( ~ class) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, fill = sample_type2), width = 0.7, alpha = 0.8) +
  geom_segment(aes(y = Q3, yend = UL, x = siren_type, xend = siren_type)) +
  geom_segment(aes(y = Q1, yend = LL, x = siren_type, xend = siren_type)) +
  geom_errorbar(aes(ymax = LL, ymin = LL), width = 0.3) + 
  geom_errorbar(aes(ymax = UL, ymin = UL), width = 0.3) +
  scale_fill_manual(values = c("seagreen", "green4", "orangered3", "tomato1", "dodgerblue2", 
                               "orange", "magenta4", "lightgoldenrod4"),
                     limits = c("leaf", "bract", "ovary", "egg", "sperm", "pollen",
                                "embryo", "endo.")) +
  scale_y_continuous(breaks = c(0, 0.4, 0.8)) +
  labs(x = NULL,
       y = "mCHH") +
  theme_minimal() +
  theme(legend.position = "none",
        strip.placement = "outside",
        panel.spacing.y = unit(1, "lines"),
        text = element_text(size = 18, face = "bold", color = "black"), 
        axis.text = element_text(size = 14, face = "bold", color = "black"),
        axis.text.x = element_text(angle = 45, hjust = 1)) 

ggsave("paired_siren_meth.png", height = 6.5, width = 4.7)
ggsave("paired_siren_meth.svg", height = 6.5, width = 4.7)
```

```{r}
ss_comb %>% 
  filter(str_detect(genotype, "ros") == F) %>% 
  mutate(class = factor(class, levels = levels_1)) %>% 
  filter(context == "mCG") %>% 
  mutate(sample_type2 = case_when(
    str_detect(sample_type, "egg") ~ "egg",
    str_detect(sample_type, "sperm") ~ "sperm", 
    str_detect(sample_type, "pollen") ~ "pollen",
    T ~ sample_type
  )) %>%  
  ggplot(aes(x = siren_type, y = median)) + 
  facet_wrap( ~ class) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, fill = sample_type2), width = 0.7, alpha = 0.8) +
  geom_segment(aes(y = Q3, yend = UL, x = siren_type, xend = siren_type)) +
  geom_segment(aes(y = Q1, yend = LL, x = siren_type, xend = siren_type)) +
  geom_errorbar(aes(ymax = LL, ymin = LL), width = 0.3) + 
  geom_errorbar(aes(ymax = UL, ymin = UL), width = 0.3) +
  scale_fill_manual(values = c("seagreen", "green4", "orangered3", "tomato1", "dodgerblue2", 
                               "orange", "magenta4", "lightgoldenrod4"),
                     limits = c("leaf", "bract", "ovary", "egg", "sperm", "pollen",
                                "embryo", "endo.")) +
  scale_y_continuous(breaks = c(0, 0.4, 0.8)) +
  labs(x = NULL,
       y = "mCG") +
  theme_minimal() +
  theme(legend.position = "none",
        strip.placement = "outside",
        panel.spacing.y = unit(1, "lines"),
        text = element_text(size = 18, face = "bold", color = "black"), 
        axis.text = element_text(size = 14, face = "bold", color = "black"),
        axis.text.x = element_text(angle = 45, hjust = 1)) 

ggsave("paired_siren_meth_CG.png", height = 6.5, width = 4.7)
ggsave("paired_siren_meth_CG.svg", height = 6.5, width = 4.7)
```
```{r}
ss_comb %>% 
  filter(str_detect(genotype, "ros") == F) %>% 
  mutate(class = factor(class, levels = levels_1)) %>% 
  filter(context == "mCHG") %>% 
  mutate(sample_type2 = case_when(
    str_detect(sample_type, "egg") ~ "egg",
    str_detect(sample_type, "sperm") ~ "sperm", 
    str_detect(sample_type, "pollen") ~ "pollen",
    T ~ sample_type
  )) %>%  
  ggplot(aes(x = siren_type, y = median)) + 
  facet_wrap( ~ class) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, fill = sample_type2), width = 0.7, alpha = 0.8) +
  geom_segment(aes(y = Q3, yend = UL, x = siren_type, xend = siren_type)) +
  geom_segment(aes(y = Q1, yend = LL, x = siren_type, xend = siren_type)) +
  geom_errorbar(aes(ymax = LL, ymin = LL), width = 0.3) + 
  geom_errorbar(aes(ymax = UL, ymin = UL), width = 0.3) +
  scale_y_continuous(breaks = c(0, 0.4, 0.8))+ 
  scale_fill_manual(values = c("seagreen", "green4", "orangered3", "tomato1", "dodgerblue2", 
                               "orange", "magenta4", "lightgoldenrod4"),
                     limits = c("leaf", "bract", "ovary", "egg", "sperm", "pollen",
                                "embryo", "endo.")) +
  labs(x = NULL,
       y = "mCHG") +
  theme_minimal() +
  theme(legend.position = "none",
        strip.placement = "outside",
        panel.spacing.y = unit(1, "lines"),
        text = element_text(size = 18, face = "bold", color = "black"), 
        axis.text = element_text(size = 14, face = "bold", color = "black"),
        axis.text.x = element_text(angle = 45, hjust = 1)) 

ggsave("paired_siren_meth_CHG.png", height = 6.5, width = 4.7)
ggsave("paired_siren_meth_CHG.svg", height = 6.5, width = 4.7)
```




```{r}
ss_comb %>% 
  filter(str_detect(genotype, "ros") == F) %>% 
  filter(str_detect(sample_type, "egg|ovary")) %>%   
  mutate(class = factor(class, levels = levels_1)) %>% 
  filter(context == "mCHH") %>% 
  mutate(sample_type2 = case_when(
    str_detect(sample_type, "egg") ~ "egg",
    str_detect(sample_type, "sperm") ~ "sperm", 
    str_detect(sample_type, "pollen") ~ "pollen",
    T ~ sample_type
  )) %>%  
  ggplot(aes(x = siren_type, y = median)) + 
  facet_wrap( ~ class) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, fill = sample_type2), width = 0.7, alpha = 0.8) +
  geom_segment(aes(y = Q3, yend = UL, x = siren_type, xend = siren_type)) +
  geom_segment(aes(y = Q1, yend = LL, x = siren_type, xend = siren_type)) +
  geom_errorbar(aes(ymax = LL, ymin = LL), width = 0.3) + 
  geom_errorbar(aes(ymax = UL, ymin = UL), width = 0.3) +
  scale_fill_manual(values = c( "orangered3", "tomato1"),
                     limits = c( "ovary", "egg")) +
  labs(x = NULL,
       y = "mCHH") +
  theme_minimal() +
  theme(legend.position = "none",
        strip.placement = "outside",
        panel.spacing.y = unit(1, "lines"),
        text = element_text(size = 18, face = "bold", color = "black"), 
        axis.text = element_text(size = 14, face = "bold", color = "black"),
        axis.text.x = element_text(angle = 45, hjust = 1)) 

ggsave("paired_siren_meth_CHH_lite.png", height = 5.5, width = 4)
ggsave("paired_siren_meth_CHH_lite.svg", height = 5.5, width = 4)
```




 



 

 











 
 
 
 
 
 


