---
title: "zygote_24nt_siRNA_loci"
author: "Chenxin Li"
date: "3/10/2020"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(ggridges)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(RColorBrewer)
library(viridis)
library(stringr)
library(svglite)
library(cowplot)
library(patchwork)
library(forcats)
library(purrr)
```

#load data
##zygote subsets
```{r}
z.e.inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gamete_zygote_siRNA_loci/zygote-egg_intersect_siRNA_loci.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
```
```{r}
z.e.comp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gamete_zygote_siRNA_loci/zygote-egg_complement_siRNA_loci.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
```
```{r}
z.sp.inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gamete_zygote_siRNA_loci/zygote-sperm_intersect_siRNA_loci.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
```
```{r}
z.sp.comp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gamete_zygote_siRNA_loci/zygote-sperm_complement_siRNA_loci.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
```
 
```{r}
z.g.comp <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gamete_zygote_siRNA_loci/zygote-gamete_complement_siRNA_loci.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
```
```{r}
e.sp.inter <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gamete_zygote_siRNA_loci/egg-sperm_intersect_siRNA_loci.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
```



##sample total 24nt siRNA loci 
```{r}
# egg_siRNA_loci <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_codes_and_output/egg_siRNA_loci.csv")
# 
# seedling_siRNA_loci <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_codes_and_output/seedling_siRNA_loci.csv")
# 
# sperm_siRNA_loci <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_codes_and_output/sperm_siRNA_loci.csv")
# 
zygote_siRNA_loci <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_codes_and_output/zygote_siRNA_loci.csv")
```

 


#plot
```{r}
colnames(z.e.inter) <- colnames(zygote_siRNA_loci)[2:26]
colnames(z.sp.inter) <- colnames(zygote_siRNA_loci)[2:26]
colnames(z.g.comp) <- colnames(zygote_siRNA_loci)[2:26]
colnames(z.e.comp) <- colnames(zygote_siRNA_loci)[2:26]
#colnames(e.sp.inter) <- colnames(zygote_siRNA_loci)[2:26] 
```

```{r}
temp <- rbind(z.e.inter %>% 
                mutate(set = "zygote/egg intersect"),
              z.sp.inter %>% 
                mutate(set = "zygote/sperm intersect"),
              z.g.comp %>% 
                mutate(set = "zygote not gamete"),
              z.e.comp %>% 
                mutate(set = "zygote de novo loci")
              #e.sp.inter %>% 
                #mutate(set = "egg/sperm intersect")
              )
```

```{r}
head(temp)
```
```{r}
model_RPM <- lm(log10(RPM) ~ set, data = temp)
```
```{r}
est_RPM <- emmeans(model_RPM, pairwise ~ set)
multcomp::cld(est_RPM$emmeans, Letters = letters) 
est_RPM$contrasts
```
```{r}
pairwise.wilcox.test(x = temp$RPM, g = temp$set)$p.value
```

```{r}
temp_s <- temp %>% 
  group_by(set) %>% 
  summarise(sum.RPM = sum(RPM),
            NN = n(),
            median = median(RPM),
            IQR = IQR(RPM),
            Q3 = quantile(RPM, 0.75),
            Q1 = quantile(RPM, 0.25),
            UL = quantile(RPM, 0.975),
            LL = quantile(RPM, 0.025)) %>% 
  ungroup() %>% 
  mutate(set = factor(set, levels = c(
    "zygote not gamete",
    "zygote de novo loci",
    "zygote/egg intersect",
    "zygote/sperm intersect"
  )))

temp_s
```
```{r}
temp_s %>% 
  mutate(set = case_when(
    str_detect(set %>% as.character(), "not") ~ "zygote NOT gametes",
    T ~ set %>% as.character() 
  )) %>%  
  ggplot(aes(x = set)) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median, fill = set), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = set, xend = set,  y = Q3, yend = UL)) +
  geom_segment(aes(x = set, xend = set,  y = Q1, yend = LL)) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) + 
  geom_text(aes(label = paste(NN, "\nloci"), y = UL), size = 5, fontface = "bold", vjust = -0.1) + 
  scale_fill_manual(values = c("violetred4", "violetred2", "tomato1", "dodgerblue2"), 
                    limits = c("zygote de novo loci", "zygote NOT gametes", 
                               "zygote/egg intersect", "zygote/sperm intersect")) +  
  scale_x_discrete(labels = NULL) + 
  scale_y_log10(breaks = c(10, 100, 1000), limits = c(2, 2000)) +
  labs(x = NULL,
       y = "RPM",
       fill = NULL,
       color = NULL) +  
  theme_minimal() +
  theme(legend.position = "bottom", legend.direction = "vertical") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black")) +
  theme(panel.grid = element_blank())  

ggsave("zygote_loci.svg", width = 4.5, height = 4.5)
ggsave("zygote_loci.png", width = 4.5, height = 4.5)
```

 
 


```{r}
temp24_s <- temp %>% 
  group_by(set) %>% 
  summarise(sum.24 = sum(`24`),
            NN = n(),
            median = median(`24`),
            IQR = IQR(`24`),
            Q3 = quantile(`24`, 0.75),
            Q1 = quantile(`24`, 0.25),
            UL = quantile(`24`, 0.975),
            LL = quantile(`24`, 0.025))

temp24_s
```
```{r}
de_novo_stat <- cbind(
  z.e.comp %>% 
  summarise(N.de.novo = n(),
            Mb.de.novo = sum(Length)/10^6,
            Reads.de.novo = sum(Reads)),
  zygote_siRNA_loci %>% 
  summarise(NN = n(),
            Mb = sum(Length)/10^6,
            Reads = sum(Reads))
) %>% 
  mutate(percent.N = N.de.novo / NN * 100,
         percent.Reads = Reads.de.novo / Reads * 100,
         percent.Mb = Mb.de.novo / 430 *100)

head(de_novo_stat)
```


#gene distances
```{r}
files <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gene_distances", pattern = "*.txt", full.names = T)

gene_distance  <- sapply(files, read_delim, simplify = FALSE, col_names = F, delim = "\t") %>% 
  bind_rows(.id = "id")

head(gene_distance)
```

```{r}
colnames(gene_distance) <- c("id", "loci.Chr", "loci.start", "loci.end", "Name",
                            "Length", "Reads", "RPM", "UniqueReads", "FracTop",
                            "Strand", "MajorRNA", "MajorRNAReads", "Complexity", 
                            "DicerCall", "MIRNA", "PhaseScore", "Short", "Long", 
                            "20", "21", "22", "23", "24", "RPKM", "sample_type", 
                            "gene.Chr", "dff", "feature", "gene.start", "gene.end",
                            "X31", "gene.Strand", "X33", "tag", "distance")
```

```{r}
gene_distance <- gene_distance %>% 
  mutate(ID = case_when(
    str_detect(id, "egg_siRNA_loci") ~ "egg siRNA loci",
    str_detect(id, "endosperm_siRNA_loci") ~ "endosperm siRNA loci",
    str_detect(id, "endosperm_siren_loci") ~ "endosperm siren",
    str_detect(id, "ovary_siren_loci") ~ "ovary siren",
    str_detect(id, "ovary_siRNA_loci") ~ "ovary siRNA loci",
    str_detect(id, "seedling_siRNA_loci") ~ "seedling siRNA loci",
    str_detect(id, "sperm_siRNA_loci") &
      str_detect(id, "endo") == F ~ "sperm siRNA loci",
    str_detect(id, "zygote_siRNA_loci") ~ "zygote siRNA loci",
    str_detect(id, "zygote-egg_inter") ~ "zygote/egg intersect",
    str_detect(id, "zygote-egg_com") ~ "zygote de novo loci",
    str_detect(id, "zygote-gamete_com") ~ "zygote not gamete",
    str_detect(id, "zygote-sperm_com") ~ "zygote/sperm set diff.",
    str_detect(id, "zygote-sperm_inter") ~ "zygote/sperm intersect",
  ))

head(gene_distance)
```

 
```{r}
gene_distance_lite <- gene_distance %>% 
  filter(str_detect(ID, "endo") == F) %>% 
  filter(str_detect(ID, "de novo loci") |
           str_detect(ID, "seedling") | 
           str_detect(ID, "egg siRNA loci") |
           str_detect(ID, "sperm siRNA loci") |
           str_detect(ID, "zygote siRNA loci")) 
  
gene_dist_wilcox <- pairwise.wilcox.test(x = gene_distance_lite$distance, g = gene_distance_lite$ID)$p.value %>% 
  as.data.frame() 
gene_dist_wilcox
write.csv(gene_dist_wilcox, "gene_dist_wilcox.csv") 
```
 

```{r}
gene_distance %>% 
  group_by(ID) %>% 
  count()
```

```{r}
gene_distance_s <- gene_distance %>% 
  group_by(ID) %>% 
   summarise( 
            NN = n(),
            median = median(distance),
            Q3 = quantile(distance, 0.75),
            Q1 = quantile(distance, 0.25),
            UL = quantile(distance, 0.975),
            LL = quantile(distance, 0.025))

gene_distance_s
```
 


```{r}
##zygote sets
gene_distance_s %>% 
  filter(str_detect(ID, "endo|not") == F) %>% 
  filter(str_detect(ID, "de novo loci") |
           str_detect(ID, "seedling") | 
           str_detect(ID, "egg siRNA loci") |
           str_detect(ID, "sperm siRNA loci") |
           str_detect(ID, "zygote siRNA loci")) %>% 
  mutate(ID = factor(ID, levels = c(
    "egg siRNA loci", "seedling siRNA loci", "sperm siRNA loci", 
    "zygote siRNA loci", "zygote de novo loci" 
  ))) %>% 
  mutate(text.pos = case_when(
           str_detect(ID, "seedling") ~ 16,  
           str_detect(ID, "egg siRNA loci") ~ 19,
           str_detect(ID, "sperm siRNA loci") ~ 20, 
           str_detect(ID, "zygote siRNA loci") ~ 18,
           str_detect(ID, "zygote de novo loci") ~ 17.5,
  )) %>% 
  mutate(ID = reorder(ID, median)) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("median = ", median/1000, " kb", sep = ""), 
                x = ID, y = text.pos),
                vjust = -0.2,  hjust = 0, size = 5, fontface = "bold") +
  scale_fill_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred4", "violetred1"),
                    limits = c(
                      "egg siRNA loci", "seedling siRNA loci", "sperm siRNA loci", "zygote de novo loci",
                      "zygote siRNA loci"
                    )) +
  scale_y_continuous(breaks = c(0, 20, 40)) +
  labs(x = NULL,
       y = "kilobases to nearest gene",
       fill = NULL,
       color = NULL) +  
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("distance_to_gene.svg", width = 7, height = 3.25)
ggsave("distance_to_gene.png", width = 7, height = 3.25)
```


```{r}
gene_distance_s %>% 
  filter(str_detect(ID, "ovary|sperm|egg|zygote|seedling|endo") &
           str_detect(ID, "not|diff") == F &
           str_detect(ID, "intersect") == F &
           str_detect(ID, "de novo loci") == F) %>% 
  mutate(ID = case_when(
    str_detect(ID, "endosperm siren") ~ "endosperm siren loci",
    str_detect(ID, "ovary siren") ~ "ovary siren loci",
    T ~ ID
  )) %>% 
  mutate(ID = reorder(ID, median)) %>% 
  mutate(text.pos = case_when(
    UL < 20000 ~ UL + 1000,
    T ~ Q3 + 1000
  )) %>% 
  ggplot(aes(x = ID)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = ID), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = ID, xend = ID,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = ID, xend = ID,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("median = ", median/1000, " kb", sep = ""), 
                x = ID, y = text.pos/1000),
                vjust = -0.2,  hjust = 0, size = 5, fontface = "bold") + 
  scale_fill_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2",
                               "violetred2", "lightgoldenrod4", "orangered3", "lightgoldenrod4",
                               "violetred4", "tomato1", "dodgerblue2"),
                   limits = c(
    "ovary siRNA loci", "egg siRNA loci", "seedling siRNA loci", "sperm siRNA loci", 
    "zygote siRNA loci", "endosperm siRNA loci", 
    "ovary siren loci", "endosperm siren loci", 
    "zygote de novo loci", "zygote/egg intersect", "zygote/sperm intersect"
  )) +
  scale_y_continuous(breaks = c(0, 20, 40)) +
  labs(x = NULL,
       y = "kilobases to nearest gene",
       fill = NULL,
       color = NULL) +  
  guides(fill = guide_legend(nrow = 4)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("distance_near_gene.svg", width = 7, height = 4.5)
ggsave("distance_near_gene.png", width = 7, height = 4.5)
```
 


##lm with log(+1) transformed data
```{r}
model_distance_lite_log <- lm(log10(distance + 1) ~ ID, data = gene_distance_lite)
anova(model_distance_lite_log)
```
```{r}
est_dis_lite_log <- emmeans(model_distance_lite_log, pairwise ~ ID)
multcomp::cld(est_dis_lite_log$emmeans, Letters = letters)
```
```{r}
gene_distance_tukey <- est_dis_lite_log$contrasts %>% 
  as.data.frame()

gene_distance_tukey
write_excel_csv(gene_distance_tukey,  "gene_distance_tukey.csv") 
```


#TE overlaps
##data

read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/Loci_TE_cov/egg_siRNA_loci.bed.CentO.cov", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

```{r}
overlap_files <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/siren_loci_TE_coverage", 
                            pattern = "*.cov", full.names = T)

overlap_data <- sapply(overlap_files, read_delim, simplify = FALSE, 
                       col_names = F, trim_ws = T, escape_double = FALSE, delim = "\t") %>% 
  sapply(mutate, Chr = as.numeric(X1), simplify = F) %>% 
  sapply(filter, is.na(Chr) == F, simplify = F) %>% 
  sapply(select, -X1, simplify = F) %>% 
  bind_rows(.id = "id")

head(overlap_data) 
```

```{r}
overlap_data <- overlap_data %>% 
  mutate(ID = case_when(
    str_detect(id, "egg_siRNA_loci") ~ "egg siRNA loci",   
    str_detect(id, "egg-sperm inter") ~ "egg/sperm intersect", 
    str_detect(id, "endosperm_siRNA_loci") ~ "endosperm siRNA loci",
    str_detect(id, "endosperm_siren_loci") ~ "endosperm siren",  
    str_detect(id, "ovary_siren_loci") ~ "ovary siren",     
    str_detect(id, "ovary_siRNA_loci") ~ "ovary siRNA loci",
    str_detect(id, "seedling_siRNA_loci") ~ "seedling siRNA loci",
    str_detect(id, "sperm_siRNA_loci") &
      str_detect(id, "endo") == F ~ "sperm siRNA loci",
    str_detect(id, "sperm-seedling-inter") ~ "seedling/sperm intersect",
    str_detect(id, "zygote_siRNA_loci") ~ "zygote siRNA loci",
    str_detect(id, "zygote-egg_com") ~ "don't include",
    str_detect(id, "zygote-egg_inter") ~ "zygote/egg intersect",
    str_detect(id, "zygote-gamete_com") ~ "zygote not gamete",
    str_detect(id, "zygote-sperm_com") ~ "zygote/sperm set diff.",
    str_detect(id, "zygote-sperm_inter") ~ "zygote/sperm intersect",
  )) %>% 
  mutate(TE = case_when(
   str_detect(id, "TIR") ~ "TIR",     #1
   str_detect(id, "CentO") ~ "CentO",  #2
   str_detect(id, "Gypsy") ~ "Gypsy",  #3
   str_detect(id, "Copia") ~ "Copia",   #4
   str_detect(id, "LINE") ~ "LINE",    #5
   str_detect(id, "SINE") ~ "SINE",     #6
   str_detect(id, "CACTA") ~ "CACTA",    #7
   str_detect(id, "Helitron") ~ "Helitron",  #8
   str_detect(id, "genes") ~ "genes"  #9
  )) %>% 
  filter(ID != "don't include") 


head(overlap_data, 10)
```
```{r}
overlap_data_total <- overlap_data %>%  
  group_by(ID, TE) %>%
  filter(TE == "genes") %>% 
  count() %>% 
  ungroup() %>% 
  select(-TE) %>% 
  rename(total = n)

head(overlap_data_total, 10)
```
```{r}
overlap_data_TE <- overlap_data %>% 
  group_by(ID, TE) %>% 
  filter(X8 > 0) %>% 
  count() %>% 
  ungroup() %>% 
  full_join(overlap_data_total, by = "ID") %>% 
  mutate(percent = n/total * 100)

head(overlap_data_TE, 20)
```


```{r}
overlap_data_TE_s <- overlap_data %>%
  group_by(ID, TE) %>%
  summarise(
    NN = n(),
    mean = mean(X8 * 100)
  ) %>% 
  ungroup()  

overlap_data_TE_s
```


#plot total
```{r}
##total overlap
overlap_data_TE_s %>% 
  filter(str_detect(ID, "ovary|sperm|egg|zygote|seedling|endo") &
           str_detect(ID, "diff") == F &
           str_detect(ID, "inter") == F & 
           str_detect(ID, "de novo loci") == F) %>%
  mutate(ID = case_when(
    str_detect(ID, "endosperm siren") ~ "endosperm siren loci",
    str_detect(ID, "ovary siren") ~ "ovary siren loci",
    T ~ ID
  )) %>% 
  mutate(ID = factor(ID, levels = c(
    "endosperm siren loci", 
    "seedling siRNA loci",
    "ovary siRNA loci",
    "zygote siRNA loci",
    "endosperm siRNA loci",
    "egg siRNA loci",
    #"zygote/egg intersect",
    "ovary siren loci",
    #"zygote/sperm intersect",
    "sperm siRNA loci"
  ))) %>% 
  filter(is.na(ID) == F) %>% 
  filter(TE != "genes") %>% 
  mutate(TE = factor(TE, levels = c(
    "Helitron", "SINE", "LINE", "TIR", "CentO", "Copia", "CACTA", "Gypsy" 
  ))) %>% 
  ggplot(aes(x = ID, y = mean)) +
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  scale_fill_manual(values = c("indianred1", "indianred3", "indianred4", 
                               "orange2", "seagreen", "skyblue2", "skyblue4", "grey70"),
                   limits = c("Helitron", "SINE", "LINE", "TIR", "CentO", "Copia", "CACTA", "Gypsy")) +
  labs(x = NULL,
       y = "mean fraction of locus length\noverlapped by TEs (%)",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(legend.position = "bottom") +
  theme(text = element_text(size = 18, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("TE_overlap_stackbar.svg", height = 4, width = 7)
ggsave("TE_overlap_stackbar.png", height = 4, width = 7)
```

 
##zygote de novo overlaps 
```{r}
TE_cov_list <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_special_TE_cov", pattern = "*.cov", full.names = T)

de_novo_TE_overlaps <- sapply(TE_cov_list, read_table2, simplify = FALSE, col_names = F) %>% 
  bind_rows(.id = "id")

head(de_novo_TE_overlaps)
```
```{r}
de_novo_TE_overlaps_clean <- de_novo_TE_overlaps %>% 
 mutate(TE = case_when(
   str_detect(id, "TIR") ~ "TIR",      
   str_detect(id, "CentO") ~ "CentO",
   str_detect(id, "Gypsy") ~ "Gypsy",
   str_detect(id, "Copia") ~ "Copia",
   str_detect(id, "LINE") ~ "LINE",
   str_detect(id, "SINE") ~ "SINE",
   str_detect(id, "CACTA") ~ "CACTA",
   str_detect(id, "Helitron") ~ "Helitron",
   str_detect(id, "gene") ~ "genes"
  )) %>% 
  mutate(tag = paste(X1, ":", X2, "-", X3, sep ="")) %>% 
  group_by(TE) %>% 
  distinct(tag, .keep_all = T) %>% 
  ungroup()  

head(de_novo_TE_overlaps_clean)
```

 
```{r}
de_novo_TE_overlaps_percent <- de_novo_TE_overlaps_clean %>% 
  group_by(TE) %>% 
  filter(X29 >0) %>% 
  count() %>% 
  ungroup() %>% 
  mutate(percent = n/18467 * 100) 

de_novo_TE_overlaps_percent

head(overlap_data_TE)
```


##plot
```{r}
#egg, seedling, sperm, zygote, zygote de novo
overlap_data_TE %>% 
  select(-total) %>% 
  rbind(de_novo_TE_overlaps_percent %>% 
          mutate(ID = "zygote de novo loci")) %>% 
  filter(str_detect(ID, "endo") == F) %>% 
  filter(str_detect(ID, "de novo loci") |
           str_detect(ID, "seedling") | 
           str_detect(ID, "egg siRNA loci") |
           str_detect(ID, "sperm siRNA loci") |
           str_detect(ID, "zygote siRNA loci")) %>% 
  mutate(ID = factor(ID, levels = c(
   "sperm siRNA loci", "egg siRNA loci", "zygote siRNA loci", "zygote de novo loci", "seedling siRNA loci"
  ))) %>% 
  filter(TE == "TIR"|
           TE == "Gypsy") %>% 
  ggplot(aes(x = TE, y = percent)) +
  facet_grid(ID ~ ., switch = "y") + 
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  geom_text(aes(label = paste(
    signif(percent, 2), "%", sep = ""
  )), 
            size = 5, fontface = "bold", hjust = 1) +
  scale_fill_manual(values = c("grey70", "orange2")) +
  scale_x_discrete(labels = NULL) +
  labs(x = NULL,
       y = "overlapping loci (%)",
       fill = NULL) + 
  theme_minimal() +
  theme(strip.placement = "outside", legend.position = c(0.875, 0.55)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(strip.text.y = element_text(angle = 180, face = "bold")) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black")) +
  theme(panel.grid = element_blank()) +
  coord_flip()

ggsave("TE_overlap_lite.svg", width = 7.5, height = 3.25)
ggsave("TE_overlap_lite.png", width = 7.5, height = 3.25)
```

##cutoff at 33%
```{r}
overlap_data_TE33 <- overlap_data %>% 
  group_by(ID, TE) %>% 
  filter(X8 >= 0.33) %>% 
  count() %>% 
  ungroup() %>% 
  full_join(overlap_data_total, by = "ID") %>% 
  mutate(percent = n/total * 100)

head(overlap_data_TE33, 20)
```
```{r}
de_novo_TE_overlaps_percent33 <- de_novo_TE_overlaps_clean %>% 
  group_by(TE) %>% 
  filter(X29 >= 0.33) %>% 
  count() %>% 
  ungroup() %>% 
  mutate(percent = n/18467 * 100) 

de_novo_TE_overlaps_percent33
```
```{r}
overlap_data_TE33 %>% 
  select(-total) %>% 
  rbind(de_novo_TE_overlaps_percent33 %>% 
          mutate(ID = "zygote de novo loci")) %>% 
  filter(str_detect(ID, "endo") == F) %>% 
  filter(str_detect(ID, "de novo loci") |
           str_detect(ID, "seedling") | 
           str_detect(ID, "egg siRNA loci") |
           str_detect(ID, "sperm siRNA loci") |
           str_detect(ID, "zygote siRNA loci")) %>% 
  mutate(ID = factor(ID, levels = c(
   "sperm siRNA loci", "egg siRNA loci", "zygote siRNA loci", "zygote de novo loci", "seedling siRNA loci"
  ))) %>% 
  filter(TE == "TIR"|
           TE == "Gypsy") %>% 
  ggplot(aes(x = TE, y = percent)) +
  facet_grid(ID ~ ., switch = "y") + 
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  geom_text(aes(label = paste(
    signif(percent, 2), "%", sep = ""
  )), 
            size = 5, fontface = "bold", hjust = 1) +
  scale_fill_manual(values = c("grey70", "orange2")) +
  scale_x_discrete(labels = NULL) +
  labs(x = NULL,
       y = "overlapping loci (%)",
       fill = NULL) + 
  theme_minimal() +
  theme(strip.placement = "outside", legend.position = c(0.875, 0.55)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(strip.text.y = element_text(angle = 180, face = "bold")) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(axis.text.x = element_text(colour = "black")) +
  theme(axis.text.y = element_text(colour = "black")) +
  theme(panel.grid = element_blank()) +
  coord_flip()

ggsave("TE_overlap_lite33.svg", width = 7.5, height = 3.25)
ggsave("TE_overlap_lite33.png", width = 7.5, height = 3.25)
```

 


#Gypsy overlap vs not Gypsy overlap RPKM
##data
```{r}
ovary_siren_gypsy <- overlap_data %>% 
  filter(TE == "Gypsy") %>% 
  filter(ID == "ovary siren")

head(ovary_siren_gypsy)
```
```{r}
ovary_siren_loci <- read_csv("ovary_siren_loci.csv")
head(ovary_siren_loci)
```

```{r}
ovary_siren_gypsy2 <- ovary_siren_gypsy %>% 
  full_join(ovary_siren_loci, by = c("X4"= "Name")) %>% 
  mutate(overlap = case_when(
    X8 >= 0.33 ~ "Y",
    T ~ "N"
  ))

head(ovary_siren_gypsy2)
```

##stat
```{r}
model_ovary_siren <- lm(log10(RPKM) ~ overlap, data = ovary_siren_gypsy2 %>% 
                                      mutate(Chr = as.factor(Chr.x)))
anova(model_ovary_siren)

est_ovary_siren <- emmeans(model_ovary_siren, pairwise ~ overlap)
est_ovary_siren$contrasts %>% 
  as.data.frame()
```


##summary and plot
```{r}
ovary_siren_gypsy_s <- ovary_siren_gypsy2 %>% 
  group_by(overlap) %>% 
  summarise(
     NN = n(),
     median = median(RPKM),
     IQR = IQR(RPKM),
     Q3 = quantile(RPKM, 0.75),
     Q1 = quantile(RPKM, 0.25),
     UL = quantile(RPKM, 0.975),
     LL = quantile(RPKM, 0.025),
     mean = mean(RPKM)
  )

ovary_siren_gypsy_s
```

```{r}
ovary_siren_gypsy_s %>% 
  ggplot(aes(x = overlap)) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = overlap, xend = overlap,  y = Q3, yend = UL)) +
  geom_segment(aes(x = overlap, xend = overlap,  y = Q1, yend = LL)) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) + 
  geom_text(aes(label = paste(
    "n = ", NN, sep = ""
  ), y= Q3), size = 5, fontface = "bold", hjust = -0.2, vjust = -0.2) +
  annotate(geom = "text", y = 2000, x = 1.55, label = "P = 6.4e-42",
           fontface = "bold", size = 5, hjust = 1) +
  scale_y_log10(breaks = c(10, 100, 1000)) +
  scale_x_discrete(label = c("Gypsy overlap\n< 33% of locus length", 
                             "Gypsy overlap\n>= 33% of locus length")) +
  labs(y = "RPKM ovary siRNA\nat ovary siren loci",
       x = "") +
  theme_classic() +
  theme(
    #axis.line = element_line(size = 1.2),
    text = element_text(size = 18, color = "black", face = "bold"),
    axis.text = element_text(color = "black", face = "bold"),
    axis.text.y = element_text(hjust = 0.5)
    ) +
  coord_flip()

ggsave("ovary_siren_gypsy.svg", height = 3, width = 5.5)
ggsave("ovary_siren_gypsy.png", height = 3, width = 5.5)
```
```{r}
ovary_siren_gypsy2 %>%
  ggplot(aes(x = X8, y = RPKM)) +
  geom_point()
```
```{r}
ovary_siren_gypsy2 %>% 
  filter(X8 ==0) %>% 
  count()

ovary_siren_gypsy2 %>% 
  filter(X8 ==1) %>% 
  count()
```


#length profile
##data
```{r}
length_files <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/length_counts", pattern = "*.txt", full.names = T)

length_data  <- sapply(length_files, read_delim, simplify = FALSE, delim = "\t") %>% 
  bind_rows(.id = "id")
```

```{r}
length_data %>% 
  head(10)
```
```{r}
sample_des <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/sample_des_clvs12.xlsx")

smRNA_comp <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/codes_and_output/smRNA_comp.csv")

sample_des_size <- smRNA_comp %>% 
  inner_join(sample_des, by = c("sample_ID")) %>% 
  select(-2, -3, -4, -5, -6)
sample_des_size
```


```{r}
length_data <- length_data %>%
  mutate(library = X1) %>% 
    mutate(sample_ID = case_when(
    str_detect(library, "sperm.1") ~ "BS1", #1
    str_detect(library, "Sperm2") ~ "BS2",  #2
    str_detect(library, "Sperm4a") ~ "BS4a",  #3
    str_detect(library, "sperm4b") ~ "BS4b",  #4
    str_detect(library, "sperm6a") ~ "BS6a",  #5
    str_detect(library, "sperm6b") ~ "BS6b",  #6
    str_detect(library, "sperm.5b") ~ "SP5b", #7
    str_detect(library, "sperm.5c") ~ "SP5c", #8
    str_detect(library, "Bulk.15") ~ "B15",   #9
    str_detect(library, "Bulk.20") ~ "B20",   #10
    str_detect(library, "Bulk.25") ~ "B25",   #11
    str_detect(library, "seedling4") ~ "SD4", #12
    str_detect(library, "egg.9a") ~ "E9a",    #13
    str_detect(library, "egg.9b") ~ "E9b",    #14
    str_detect(library, "egg.9c") ~ "E9c",    #15
    str_detect(library, "eggcell4") ~ "E4",   #16
    str_detect(library, "eggcell3") ~ "E3",   #17
    str_detect(library, "eggcell2") ~ "E2",   #18
    str_detect(library, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(library, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(library, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(library, "SRR57") ~ "root",              #22
    str_detect(library, "ab_control") ~ "ddm1_con",     #23
    str_detect(library, "ddm1ab") ~ "ddm1",             #24
    str_detect(library, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(library, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(library, "SRR2544786") ~ "em_dry", #27  
    str_detect(library, "SRR5049778") ~ "an_pre",  #28
    str_detect(library, "SRR5049779") ~ "an_mei",  #29
    str_detect(library, "SRR5049780") ~ "an_mic",  #30
    str_detect(library, "SRR5049781") ~ "an_bi",   #31
    str_detect(library, "SRR5049786") ~ "ov_1st",  #32
    str_detect(library, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(library, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(library, "SRR5049789") ~ "ov_4th",  #35
    str_detect(library, "S3_1") ~ "S3_1",          #36
    str_detect(library, "S3_2") ~ "S3_2",          #37
    str_detect(library, "S3_3") ~ "S3_3",          #38
    str_detect(library, "S5_1") ~ "S5_1",          #39
    str_detect(library, "S5_2") ~ "S5_2",          #40
    str_detect(library, "S5_3") ~ "S5_3",          #41
    str_detect(library, "S7_1") ~ "S7_1",          #42
    str_detect(library, "S7_2") ~ "S7_2",          #43
    str_detect(library, "S7_3") ~ "S7_3",          #44
    str_detect(library, "SRR2544787") ~ "em_12h",  #45
    str_detect(library, "SRR2544788") ~ "em_24h",  #46
    str_detect(library, "SRR771501") ~ "em_7DAF",  #47
    str_detect(library, "SRR771500") ~ "en_7DAF",   #48
    str_detect(library, "SD7") ~ "SD7", #49
    str_detect(library, "zg1d") ~ "zg1d", #50
    str_detect(library, "zg1e") ~ "zg1e",  #51
    str_detect(library, "zg2d") ~ "zg2d",  #52
    str_detect(library, "zg2e") ~ "zg2e",  #53
    str_detect(library, "zg3a") ~ "zg3a", #54
    str_detect(library, "zg3b") ~ "zg3b"  #55
  )) %>% 
    inner_join(sample_des_size %>% 
                 select(`total read`, `general siRNA`, sample_ID, sample_type, genotype), by = "sample_ID")


```

```{r}
head(length_data)
```



##siRNA in different loci - length profile
zygote, zygote/egg, zygote/egg comp, zygote de novo 
```{r}
length_data1 <- length_data %>% 
  filter(str_detect(id, "zygote")) %>% 
  mutate(category = case_when(
    str_detect(id, "zygote-egg_inter") ~ "zygote/egg\nintersect",
    str_detect(id, "zygote-egg_comp") ~ "zygote\nde novo loci", 
    str_detect(id, "zygote_siRNA_loci") ~ "zygote\nsiRNA loci", 
    str_detect(id, "zygote-gamete") ~ "zygote/gamete\nset difference"
  )) %>% 
  mutate(category = factor(category, levels = c(
     "zygote\nsiRNA loci", 
     "zygote/egg\nintersect",
     "zygote\nde novo loci",
     "zygote/gamete\nset difference" 
  ))) %>% 
  filter(is.na(category) == F) %>% 
  filter(sample_type == "egg"|
           sample_type == "seedling shoot" |
           sample_type == "sperm" |
           sample_type == "zygote"|
           str_detect(sample_type, "DAF")) %>%
  filter(str_detect(sample_type, "endo") == F) %>% 
  gather("length", "count", c(`20`, `21`, `22`, `23`, `24`, `25`)) %>% 
  select(category, sample_type, sample_ID, length, count, `general siRNA`) %>% 
  mutate(proportion = count / `general siRNA`)

head(length_data1)
```
```{r}
f4D <- length_data1 %>% 
  filter(str_detect(category, "gamete") == F) %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "embryo") ~ "embryo",
    str_detect(sample_type, "seedling") ~ "seedling",
    T ~ sample_type
  )) %>% 
  ggplot(aes(x = length, y = proportion * 100)) +
  facet_grid(. ~ category, switch = "y") +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type), 
               position = position_dodge(0.01), width = 0.5) + 
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2", "grey20"),
                    limits = c("egg", "seedling", "sperm", "zygote", "embryo")) +
  labs(color = "siRNA transcriptome",
       x = "length (nt)",
       y = "% siRNA") +
  guides(color = guide_legend(nrow = 2)) + 
  theme_minimal() +
  theme(strip.placement = "outside") +
  theme(legend.position = "bottom") +
  theme(strip.text.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) 

f4D
ggsave("length_profile_siRNA_loci.svg", height = 4, width = 7)
ggsave("length_profile_siRNA_loci.png", height = 4, width = 7)
```


```{r}
f4E <- length_data1 %>% 
  filter(str_detect(category, "de novo loci")) %>% 
  ggplot(aes(x = length, y = proportion * 100)) +
  facet_grid(. ~ category, switch = "y") +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type), 
               position = position_dodge(0.01), width = 0.5) + 
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2", "grey20"),
                    limits = c("egg", "seedling shoot", "sperm", "zygote", "embryo 7-8 DAF")) +
  labs(color = "siRNA transcriptome",
       x = NULL,
       y = NULL) +
  guides(color = guide_legend(nrow = 2)) + 
  theme_minimal() +
  theme(strip.placement = "outside") +
  theme(legend.position = "none") +
  theme(strip.text.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(color = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black"))  

f4E
ggsave("length_profile_siRNA_loci_denovo.svg", height = 3, width = 2.1)
ggsave("length_profile_siRNA_loci_denovo.png", height = 3, width = 2.1)
```
```{r}
plot_grid(f4D, f4E, nrow = 1, axis = "tb", rel_widths = c(0.76, 0.24), align = "h")
ggsave("length_profile_zygote_loci.svg", width = 10, height = 4) 
```


```{r}
length_data1 %>% 
  filter(str_detect(category, "diff")) %>% 
  ggplot(aes(x = length, y = proportion * 100)) +
  facet_grid(. ~ category, switch = "y") +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type), 
               position = position_dodge(0.01), width = 0.5) + 
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2", "magenta3"),
                    limits = c("egg", "seedling shoot", "sperm", "zygote", "embryo 7-8 DAF")) +
  labs(color = "siRNA transcriptome",
       x = NULL,
       y = NULL) +
  guides(color = guide_legend(nrow = 2)) + 
  theme_minimal() +
  theme(strip.placement = "outside") +
  theme(legend.position = "none") +
  theme(strip.text.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(color = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black"))  

ggsave("length_profile_siRNA_loci_zygote_egg_comp.svg", height = 3, width = 2.1)
ggsave("length_profile_siRNA_loci_zygote_egg_comp.png", height = 3, width = 2.1)
```











##siRNA in siren - dot plot

```{r}
total_siRNA_length0 <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/length_profiel_fig2/lengths_summary.bam.siRNAs.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE) 

CLVS12_total <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_siRNAs/length_counts/lengths_summary.siRNAs.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```

```{r}
total_siRNA_length <- total_siRNA_length0 %>%
  rbind(CLVS12_total %>% 
          filter(str_detect(X1, "q20") == F))


colnames(total_siRNA_length) <- c("library", "20", "21", "22", "23", "24", "25")
```

```{r}
total_siRNA_length <- total_siRNA_length %>% 
  mutate(sample_ID = case_when(
    str_detect(library, "sperm.1") ~ "BS1", #1
    str_detect(library, "Sperm2") ~ "BS2",  #2
    str_detect(library, "Sperm4a") ~ "BS4a",  #3
    str_detect(library, "sperm4b") ~ "BS4b",  #4
    str_detect(library, "sperm6a") ~ "BS6a",  #5
    str_detect(library, "sperm6b") ~ "BS6b",  #6
    str_detect(library, "sperm.5b") ~ "SP5b", #7
    str_detect(library, "sperm.5c") ~ "SP5c", #8
    str_detect(library, "Bulk.15") ~ "B15",   #9
    str_detect(library, "Bulk.20") ~ "B20",   #10
    str_detect(library, "Bulk.25") ~ "B25",   #11
    str_detect(library, "seedling4") ~ "SD4", #12
    str_detect(library, "egg.9a") ~ "E9a",    #13
    str_detect(library, "egg.9b") ~ "E9b",    #14
    str_detect(library, "egg.9c") ~ "E9c",    #15
    str_detect(library, "eggcell4") ~ "E4",   #16
    str_detect(library, "eggcell3") ~ "E3",   #17
    str_detect(library, "eggcell2") ~ "E2",   #18
    str_detect(library, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(library, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(library, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(library, "SRR57") ~ "root",              #22
    str_detect(library, "ab_control") ~ "ddm1_con",     #23
    str_detect(library, "ddm1ab") ~ "ddm1",             #24
    str_detect(library, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(library, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(library, "SRR2544786") ~ "em_dry", #27  
    str_detect(library, "SRR5049778") ~ "an_pre",  #28
    str_detect(library, "SRR5049779") ~ "an_mei",  #29
    str_detect(library, "SRR5049780") ~ "an_mic",  #30
    str_detect(library, "SRR5049781") ~ "an_bi",   #31
    str_detect(library, "SRR5049786") ~ "ov_1st",  #32
    str_detect(library, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(library, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(library, "SRR5049789") ~ "ov_4th",  #35
    str_detect(library, "S3_1") ~ "S3_1",          #36
    str_detect(library, "S3_2") ~ "S3_2",          #37
    str_detect(library, "S3_3") ~ "S3_3",          #38
    str_detect(library, "S5_1") ~ "S5_1",          #39
    str_detect(library, "S5_2") ~ "S5_2",          #40
    str_detect(library, "S5_3") ~ "S5_3",          #41
    str_detect(library, "S7_1") ~ "S7_1",          #42
    str_detect(library, "S7_2") ~ "S7_2",          #43
    str_detect(library, "S7_3") ~ "S7_3",          #44
    str_detect(library, "SRR2544787") ~ "em_12h",  #45
    str_detect(library, "SRR2544788") ~ "em_24h",  #46
    str_detect(library, "SRR771501") ~ "em_7DAF",  #47
    str_detect(library, "SRR771500") ~ "en_7DAF",   #48
    str_detect(library, "SD7") ~ "SD7", #49
    str_detect(library, "zg1d") ~ "zg1d", #50
    str_detect(library, "zg1e") ~ "zg1e",  #51
    str_detect(library, "zg2d") ~ "zg2d",  #52
    str_detect(library, "zg2e") ~ "zg2e",  #53
    str_detect(library, "zg3a") ~ "zg3a", #54
    str_detect(library, "zg3b") ~ "zg3b"  #55
  )) %>% 
    inner_join(sample_des_size %>% 
                 select(`total read`, `general siRNA`, sample_ID, sample_type, genotype), by = "sample_ID")

head(total_siRNA_length)
```

```{r}
length_siren <- length_data %>% 
  filter(str_detect(id, "siren")) %>% 
  mutate(category = case_when(
    str_detect(id, "endosperm") ~ "endosperm siren",
    str_detect(id, "ovary") ~ "ovary siren"
  )) %>% 
  filter(is.na(category) == F) %>% 
  filter(sample_type == "egg"|
           sample_type == "seedling shoot" |
           sample_type == "sperm" |
           sample_type == "zygote"|
           sample_type == "ovary" |
           str_detect(sample_type, "DAF")) %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "endo") ~ "endosperm",
    str_detect(sample_type, "embryo") ~ "embryo",
    str_detect(sample_type, "seedling") ~ "seedling", 
    T ~ sample_type
  )) %>% 
  mutate(sample_type = factor(sample_type, levels = c(
      "ovary", "egg", "seedling", "sperm", "zygote", "embryo", "endosperm"  
  ))) %>%
  gather("length", "count", c(`20`, `21`, `22`, `23`, `24`, `25`)) %>% 
  mutate(proportion = count / `general siRNA`)

head(length_siren)
```
```{r}
siren24 <- length_siren %>% 
  filter(length == 24) %>% 
  select(category, sample_ID, sample_type, count) %>% 
  inner_join(total_siRNA_length %>% 
               select(sample_ID, `24`), by = c("sample_ID")) %>% 
  mutate(proportion = count/`24`) 

head(siren24)
```
```{r}
logit <- function(p){log(
  p / (1-p)
)}
invlogit <- function(x){
  1/(1 + exp(-x))
}
```

```{r}
model_siren <- lm(proportion %>% logit() ~ sample_type * category, siren24)
plot(model_siren)
anova(model_siren)
```
```{r}
est_siren <- emmeans(model_siren, pairwise ~ sample_type | category)
multcomp::cld(est_siren$emmeans, Letters = letters)
```


```{r}
siren24 %>% 
  ggplot(aes(x = sample_type, y = proportion * 100)) +
  facet_grid(category ~ .) +
  geom_point(aes(fill = sample_type), color = "white", shape = 21, position = position_jitter(0.1, seed = 666),
             alpha = 0.8, size = 3) + 
  stat_summary(geom = "errorbar", fun.data = "mean_cl_boot", width = 0.3, size = 1.25, alpha = 0.8) +
  scale_fill_manual(values = c("orangered3", "tomato1", 
                               "seagreen", "dodgerblue2", 
                               "violetred2", "magenta3", "lightgoldenrod4"),
                   limits = c("ovary", "egg", 
                              "seedling", "sperm", 
                              "zygote", "embryo", "endosperm")) +
  labs(x = NULL,
       y = "% 24nt siRNA",
       color = NULL,
       fill = NULL) +
  guides(fill = guide_legend(nrow = 3)) +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(panel.spacing.y = unit(1, "line")) +
  theme(axis.line = element_line(size = 1.2)) + 
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(color = "black")) +
  theme(axis.text.y=element_text(color = "black", hjust = 0.5)) +
  coord_flip()

ggsave("siren_by_sample_type.svg", width = 5, height = 5) 
ggsave("siren_by_sample_type.png", width = 5, height = 5)
```


```{r}
length_siren %>% 
  ggplot(aes(x = length, y = proportion * 100)) +
  facet_grid(. ~ category, switch = "y") +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type), 
               position = position_dodge(0.01), width = 0.5) + 
  scale_color_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2", "violetred2", "magenta3", "lightgoldenrod4"),
                   limits = c("ovary", "egg", "seedling", "sperm", "zygote", "embryo", "endosperm")) +
  labs(color = "siRNA transcriptome",
       x = "length (nt)",
       y = "% siRNA") +
  guides(color = guide_legend(nrow = 4)) + 
  theme_minimal() +
  theme(strip.placement = "outside") +
  theme(legend.position = "bottom") +
  theme(strip.text.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) 
```













Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
