---
title: "zygote_denovo_overlaps"
author: "Chenxin Li"
date: "7/31/2020"
output: html_notebook 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ggplot2)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(RColorBrewer)
library(stringr)
library(svglite)
library(cowplot)
library(emmeans)
library(ggridges)
```

18467 zygote-egg_complement_siRNA_loci.bed
97783 seedling_siRNA_loci.bed
79990 sperm_siRNA_loci.bed
15444 zygote-egg_complement-seedling_intersect_siRNA_loci.u.bed
3474 zygote-egg_complement-sperm_intersect_siRNA_loci.u.bed
2696 zygote-egg_complement-seedling-sperm_intersect_siRNA_loci.u.bed
22058 sperm-seedling_intersect_siRNA_loci.u.bed
19631 seedling-sperm_intersect_siRNA_loci.u.bed



#Upset
##upper left
```{r}
set_sizes <- data.frame(
  set = c("zygote de novo loci", "seedling siRNA loci", "sperm siRNA loci"),
  size = c(18467, 97783, 79990)
) %>% 
  mutate(set = factor(set, levels = c(
    "zygote de novo loci", "seedling siRNA loci", "sperm siRNA loci"
  )))
```

```{r}
upperleft <- set_sizes %>% 
  ggplot(aes(x = set, y= size)) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_bar(stat = "identity", aes(fill = set), alpha = 0.8) +
  geom_text(aes(label = as.character(size)), size = 5, angle = 90, hjust = 0, y = 1, fontface = "bold") +
  scale_fill_manual(values = c("violetred4", "seagreen", "dodgerblue2"), #this is my own custom palette  
                     limits = c("zygote de novo loci", "seedling siRNA loci", "sperm siRNA loci")) +  #feel free to use something else     
  scale_x_discrete(labels = NULL) +
  scale_y_continuous(labels = NULL) +
  labs(x = NULL,
       y = "set size") +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(panel.grid = element_blank()) 

upperleft
```

##upper right
```{r}
upperright <- get_legend(
  set_sizes %>% 
  ggplot(aes(x = set, y= size)) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_bar(stat = "identity", aes(fill = set), alpha = 0.8) +
  geom_text(aes(label = as.character(size)), size = 5, angle = 90, hjust = 0, y = 1, fontface = "bold") +
  scale_fill_manual(values = c("violetred4", "seagreen", "dodgerblue2"), #this is my own custom palette  
                     limits = c("zygote de novo loci", "seedling siRNA loci", "sperm siRNA loci")) +  #feel free to use something else     
  scale_x_discrete(labels = NULL) +
  scale_y_continuous(labels = NULL) +
  labs(x = NULL,
       y = "set size"
       ) +
  theme_minimal() +
  theme(legend.position = "right") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(panel.grid = element_blank()) 
  ) 
#same code as upper left, but w/ legend.position = right to get the legend

plot_grid(upperright)
```

##lowerleft
```{r}
set <- set_sizes$set
category <- c(
    "zygote specific" ,
    "zygote de novo loci by seedling siRNA loci",  #zygote-egg_complement-seedling_intersect_siRNA_loci.u.bed
    "zygote de novo loci by sperm siRNA loci",  #zygote-egg_complement-sperm_intersect_siRNA_loci.u.bed
    "zygote de novo loci by seedling siRNA loci by sperm siRNA loci"  #zygote-egg_complement-seedling-sperm_intersect_siRNA_loci.u.bed
)

overlap_matrix <- expand_grid(
  "set" = set,
  "category" = category
) %>% 
  mutate(intersect = case_when(
    str_detect(category, set %>% as.character()) ~ "Y",
    T ~ "N"
  )) %>% 
  mutate(intersect = case_when(
    set == "zygote de novo loci" &
      category == "zygote specific" ~ "Y",
    T ~ intersect
  ))
  

overlap_matrix
```

```{r}
lowerleft <- overlap_matrix %>% 
  mutate(category = factor(category, levels = c(
    "zygote de novo loci by seedling siRNA loci by sperm siRNA loci",
    "zygote de novo loci by sperm siRNA loci",
    "zygote de novo loci by seedling siRNA loci",
    "zygote specific" 
  ))) %>% #in the colored matrix the first y value apprears in the bottom, 
                                               #so the order need to be reversed 
  ggplot(aes(x = set, y = category))+
  geom_tile(aes(fill = set, alpha = intersect), color = "black", size = 1.5) +
  scale_fill_manual(values = c("violetred4", "seagreen", "dodgerblue2"), #this is my own custom palette  
                     limits = c("zygote de novo loci", "seedling siRNA loci", "sperm siRNA loci")) +  #feel free to use something else 
  scale_alpha_manual(values = c(0.8, 0),  #color the grid for Y, don't color for N. 
                     limits = c("Y", "N")) +
  scale_y_discrete(labels = NULL) +
  scale_x_discrete(labels = rep(" ", length(set))) + #I left white space here for better alignment w/ extended plots 
  labs(x = " ", #white space for better alignment w/ right side plots 
       y = "overlap") +
  theme_minimal() +
  theme(legend.position = "none") +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(panel.grid = element_blank())

lowerleft
```



##lower right 1 

15444 zygote-egg_complement-seedling_intersect_siRNA_loci.u.bed
3474 zygote-egg_complement-sperm_intersect_siRNA_loci.u.bed
2696 zygote-egg_complement-seedling-sperm_intersect_siRNA_loci.u.bed

```{r}
overlap_sizes <- data.frame(
  "category" = c("zygote specific", 
                 "zygote de novo by seedling siRNA loci",
                 "zygote de novo by sperm siRNA loci",
                 "zygote de novo by seedling siRNA loci by sperm siRNA loci"),
  "overlap_size" = c(2245, 15444, 3474, 2696)
) %>% 
  mutate(category = factor(category, levels = c(
    "zygote de novo by seedling siRNA loci by sperm siRNA loci",
    "zygote de novo by sperm siRNA loci",
    "zygote de novo by seedling siRNA loci",
     "zygote specific"
  ))) %>% 
  mutate(percent = overlap_size / 18467 * 100)

head(overlap_sizes)
```

```{r}
lowerright <- 
  overlap_sizes %>% 
  ggplot(aes(x = category, y = overlap_size)) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_vline(xintercept = -Inf, size = 1.5) +
  geom_bar(stat = "identity", fill = "grey80", color = NA, alpha = 0.8) +
  geom_text(aes(label = paste(
    overlap_size, "/18467 (", percent %>% signif(2), "%) ", sep = "" 
  )), size = 5, hjust = 0, vjust = 0.5, fontface = "bold", y = 0) +
  scale_y_continuous(breaks = c(0, max(overlap_sizes$overlap_size)) ,
                     labels = rep(" ", 2)) + #I left white space here for better alignment w/ extended plots
  scale_x_discrete(labels = NULL) +
  labs(y = "intersect. sizes",
       x = NULL) +
  theme_minimal() +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) +
  theme(panel.grid = element_blank()) + 
  coord_flip()

lowerright
```


##putting all the pieces together
```{r}
plot_grid(upperleft, upperright, lowerleft, lowerright, 
          nrow = 2, 
          ncol = 2,
          rel_heights = c(1, 1), #the more rows in the lower part, the longer it should be
          rel_widths = c(0.7, 1)) 

ggsave("zygote_de_novo_overlaps.svg", height = 4.5, width = 4)
ggsave("zygote_de_novo_overlaps.png", height = 4.5, width = 4)
```


#zygote specific characterization
##data
```{r}
#zygote de novo
zygote_denovo <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gamete_zygote_siRNA_loci/zygote-egg_complement_siRNA_loci.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 

#zygote not gametes
zygote_not_gamete <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gamete_zygote_siRNA_loci/zygote-gamete_complement_siRNA_loci.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 

#zygote/sperm
zygote_sperm_intersect <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/beds_1_2020/zygote-sperm_intersect_siRNA_loci.u.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 

#zygote/seedling  
zygote_seedling_intersect <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/beds_1_2020/zygote-seedling_intersect_siRNA_loci.u.bed", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

zygote_siRNA_loci <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_codes_and_output/zygote_siRNA_loci.csv")
```

##membership 
```{r}
head(zygote_not_gamete)
zygote_specific <- zygote_not_gamete %>% 
  filter(X4 %in% zygote_seedling_intersect$X4 == F)
```


```{r}
head(zygote_seedling_intersect)
head(zygote_denovo)
zygote_sperm_intersect_not_seedling <- zygote_denovo %>% 
  filter(X4 %in% zygote_seedling_intersect$X4 == F) %>% 
  filter(X4 %in% zygote_specific$X4 == F) 
```
```{r}
zygote_denovo_seedling_not_sperm <- zygote_denovo %>% 
   filter(X4 %in% zygote_seedling_intersect$X4) %>% 
   filter(X4 %in% zygote_sperm_intersect$X4 == F)
```

```{r}
zygote_denovo_three_way <- zygote_denovo %>% 
  filter(X4 %in% zygote_seedling_intersect$X4) %>% 
  filter(X4 %in% zygote_sperm_intersect$X4) 
```

 
```{r}
zygote_egg_seedling <- zygote_siRNA_loci %>% 
  filter(Name %in% zygote_denovo$X4 == F) %>% 
  filter(Name %in% zygote_seedling_intersect$X4) 

nrow(zygote_egg_seedling)
```

```{r}
zygote_egg_NOT_seedling <- zygote_siRNA_loci %>% 
  filter(Name %in% zygote_denovo$X4 == F) %>% 
  filter(Name %in% zygote_seedling_intersect$X4 == F) 

nrow(zygote_egg_NOT_seedling)
```

#Chisq test/Exact test
```{r}
nrow(zygote_siRNA_loci)
nrow(zygote_denovo_seedling_not_sperm) + nrow(zygote_denovo_three_way)
```

```{r}
fisher.test(rbind(
  c(nrow(zygote_denovo_seedling_not_sperm) + nrow(zygote_denovo_three_way), 
    nrow(zygote_denovo) - nrow(zygote_denovo_seedling_not_sperm) - nrow(zygote_denovo_three_way)),
  c(nrow(zygote_egg_seedling), nrow(zygote_egg_NOT_seedling) )
))$p.value

chisq.test(rbind(
  c(nrow(zygote_denovo_seedling_not_sperm) + nrow(zygote_denovo_three_way), 
    nrow(zygote_denovo) - nrow(zygote_denovo_seedling_not_sperm) - nrow(zygote_denovo_three_way)),
  c(nrow(zygote_egg_seedling), nrow(zygote_egg_NOT_seedling) )
))$p.value
```

##contingency table 
```{r}
contig_table <- data.frame(
  X1 = c("zygote de novo loci", "zygote egg intersect", "total"),
  "seedling intersect" = c(nrow(zygote_denovo_seedling_not_sperm) + nrow(zygote_denovo_three_way), 
                           nrow(zygote_egg_seedling), 
                           nrow(zygote_seedling_intersect)),
  "NOT seedling" = c(nrow(zygote_denovo) - nrow(zygote_denovo_seedling_not_sperm) - nrow(zygote_denovo_three_way),
                     nrow(zygote_egg_NOT_seedling),
                     nrow(zygote_siRNA_loci) - nrow(zygote_seedling_intersect))
) %>% 
  mutate(total = seedling.intersect + NOT.seedling)

contig_table
```
```{r}
contig_table %>% 
  gather("seedling_category", "loci", 2:4) %>%
  mutate(seedling_category = case_when(
    str_detect(seedling_category, "seedling.intersect") ~ "seedling\nintersect",
    str_detect(seedling_category, "NOT.seedling") ~ "NOT\nseedling",
    T ~ seedling_category
  )) %>% 
  mutate(seedling_category = factor(seedling_category, 
                                    levels = c(
                                      "seedling\nintersect",
                                      "NOT\nseedling",
                                      "total"
                                    ))) %>% 
  mutate(X1 = case_when(
    str_detect(X1 %>% as.character(), "egg") ~ "zygote\negg intersect",
    str_detect(X1 %>% as.character(), "de novo loci") ~ "zygote\nde novo loci",
    T ~ X1 %>% as.character()
  )) %>% 
  mutate(X1 = factor(X1, levels = c("total", "zygote\negg intersect", "zygote\nde novo loci"))) %>% 
  mutate(shade = case_when(
    X1 == "total" | 
      seedling_category == "total" ~ "T",
    T ~ "N"
  )) %>% 
  ggplot(aes(x = seedling_category, y = X1)) +
  geom_tile(color = "black", aes(alpha = shade), size = 1.1) +
  geom_text(aes(label = loci), size = 5, fontface = "bold") +
  scale_x_discrete(position = "top") +
  scale_alpha_manual(values = c(0.5, 0), limits = c("T", "N")) +
  labs(x = NULL,
       y = NULL) +
  theme_minimal() +
  theme(panel.grid = element_blank(),
        text = element_text(color = "black", size = 16, face = "bold"),
        axis.text = element_text(color = "black", face = "bold"),
        axis.text.y = element_text(hjust = 0.5),
        legend.position = "none"
        ) +
  coord_fixed()


ggsave("zygote_contingency_table.svg", width = 4.2, height = 4)
ggsave("zygote_contingency_table.png", width = 4.2, height = 4)
```


```{r}
zygote_special <- rbind(zygote_denovo_seedling_not_sperm %>% 
                          mutate(subset = "zygote de novo seedling intersect NOT sperm"), 
                        zygote_sperm_intersect_not_seedling %>% 
                          mutate(subset = "zygote de novo sperm intersect NOT seedling"),
                        zygote_denovo_three_way %>% 
                          mutate(subset = "zygote de novo seedling intersect sperm intersect"),
                        zygote_specific %>% 
                          mutate(subset = "zygote NOT gametes NOT seedling")) 

head(zygote_special)
```

##expression 
```{r}
zygote_special_s <- zygote_special %>% 
  group_by(subset) %>% 
  summarise(
    sum.RPM = sum(X7),
            NN = n(),
            median = median(X7),
            IQR = IQR(X7),
            Q3 = quantile(X7, 0.75),
            Q1 = quantile(X7, 0.25),
            UL = quantile(X7, 0.975),
            LL = quantile(X7, 0.025)) %>% 
  ungroup() %>% 
  mutate(percent = NN/18467 * 100)

zygote_special_s
```

```{r}
pairwise.wilcox.test(x = zygote_special$X7, g = zygote_special$subset)$p.value %>% 
  as.data.frame()
```

```{r}
model_special <- lm(log(X7) ~ subset, data = zygote_special)
est_special <- emmeans(model_special, pairwise ~ subset)
multcomp::cld(est_special$emmeans, Letters = letters)
```

```{r}
expression_plot <- zygote_special_s %>%
  mutate(subset = case_when(
    str_detect(subset, "NOT gametes NOT seedling") ~ "zygote\nNOT gametes\nNOT seedling",
    str_detect(subset, "sperm intersect NOT") ~ "zygote de novo\nsperm intersect\nNOT seedling",
    str_detect(subset, "seedling intersect NOT") ~ "zygote de novo\nseedling intersect\nNOT sperm",
    str_detect(subset, "seedling intersect sperm") ~ "zygote de novo\nseedling intersect\nsperm intersect"
  )) %>% 
  ggplot(aes(x = subset)) +
  geom_crossbar(aes(ymin = Q1, ymax = Q3, y = median, fill = subset), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = subset, xend = subset,  y = Q3, yend = UL)) +
  geom_segment(aes(x = subset, xend = subset,  y = Q1, yend = LL)) +
  geom_errorbar(aes(ymin = UL, ymax = UL), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL, ymax = LL), width = 0.3, size = 1) + 
  geom_text(aes(label = paste(NN, " loci", " (", round(percent, 2), "%)", sep = ""), y = UL),
           size = 5, fontface = "bold", hjust = -0.05) +
  #annotate(geom = "text", label = c("b", "c", "ab", "a"), x = c(1, 2, 3, 4), y = 75, size = 5, fontface = "bold") +
  scale_fill_manual(values = c("violetred2",  "dodgerblue2", "seagreen", "grey75"),
                    limits = c(
                      "zygote\nNOT gametes\nNOT seedling",
                      "zygote de novo\nsperm intersect\nNOT seedling",
                      "zygote de novo\nseedling intersect\nNOT sperm",
                      "zygote de novo\nseedling intersect\nsperm intersect" 
                    )) +
  #scale_x_discrete(labels = NULL) + 
  scale_y_log10(breaks = c(10, 100), limits = c(2, 100)) +
  labs(x = NULL,
       y = "RPM",
       fill = NULL,
       color = NULL) +  
  theme_minimal() +
  theme(legend.position = "none", legend.direction = "vertical") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black", hjust = 0.5)) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  theme(panel.grid = element_blank()) +
  coord_flip()

expression_plot

ggsave("zygote_loci_special.svg", width = 7, height = 4.5)
ggsave("zygote_loci_special.png", width = 7, height = 4.5)
```

##gene distance
```{r}
zygote_denovo_gene_distance <-  read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/gene_distances/zygote-egg_complement_siRNA_loci.bed.closest_gene.txt", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE) 

head(zygote_denovo_gene_distance)
```

```{r}
zygote_special_dist <- zygote_special %>% 
  inner_join(zygote_denovo_gene_distance %>% 
               select(X4, X34, X35), by = "X4")

head(zygote_special_dist)
```

```{r}
zygote_special_dist_s <- zygote_special_dist %>% 
  group_by(subset) %>% 
  summarise(
    median = median(X35),
            Q3 = quantile(X35, 0.75),
            Q1 = quantile(X35, 0.25),
            UL = quantile(X35, 0.975),
            LL = quantile(X35, 0.025)
  ) %>% 
  ungroup()

zygote_special_dist_s
```

```{r}
pairwise.wilcox.test(x = zygote_special_dist$X35, g = zygote_special_dist$subset)$p.value %>% 
  as.data.frame()
```

```{r}
distance_plot <- zygote_special_dist_s %>%
   mutate(text.pos = case_when(
    str_detect(subset, "NOT gametes NOT seedling") ~ 11,  
    str_detect(subset, "sperm intersect NOT") ~ 16,
    str_detect(subset, "seedling intersect NOT") ~ 13,
    str_detect(subset, "seedling intersect sperm intersect") ~ 14
  )) %>% 
  mutate(subset = case_when(
    str_detect(subset, "NOT gametes NOT seedling") ~ "zygote\nNOT gametes\nNOT seedling",
    str_detect(subset, "sperm intersect NOT") ~ "zygote de novo\nsperm intersect\nNOT seedling",
    str_detect(subset, "seedling intersect NOT") ~ "zygote de novo\nseedling intersect\nNOT sperm",
    str_detect(subset, "seedling intersect sperm") ~ "zygote de novo\nseedling intersect\nsperm intersect"
  )) %>%  
  ggplot(aes(x = subset)) +
  geom_crossbar(aes(ymin = Q1/1000, ymax = Q3/1000, y = median/1000, fill = subset), width = 0.7, alpha = 0.8) +
  geom_segment(aes(x = subset, xend = subset,  y = Q3/1000, yend = UL/1000)) +
  geom_segment(aes(x = subset, xend = subset,  y = Q1/1000, yend = LL/1000)) +
  geom_errorbar(aes(ymin = UL/1000, ymax = UL/1000), width = 0.3, size = 1) +
  geom_errorbar(aes(ymin = LL/1000, ymax = LL/1000), width = 0.3, size = 1) + 
  geom_text(aes(label = paste("median = ", signif(median/1000, 3), " kb", sep = ""), 
                x = subset, y = text.pos),
                vjust = -0.2,  hjust = 0, size = 5, fontface = "bold") +
  scale_fill_manual(values = c("violetred2",  "dodgerblue2", "seagreen", "grey75"),
                    limits = c(
                      "zygote\nNOT gametes\nNOT seedling",
                      "zygote de novo\nsperm intersect\nNOT seedling",
                      "zygote de novo\nseedling intersect\nNOT sperm",
                      "zygote de novo\nseedling intersect\nsperm intersect" 
                    )) +
  labs(x = NULL,
       y = "kilobases to nearest gene",
       fill = NULL,
       color = NULL) +  
  theme_minimal() +
  theme(legend.position = "none", legend.direction = "vertical") + 
  theme(strip.text = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black", hjust = 0.5)) + 
  theme(axis.text.y = element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

distance_plot

ggsave("zygote_special_gene_dist.svg", width = 7, height = 4.5)
ggsave("zygote_special_gene_dist.png", width = 7, height = 4.5)
```



##expression in sperm 
find corresponding loci in sperm 

```{r}
sperm_siRNA_loci <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/shortstack_codes_and_output/sperm_siRNA_loci.csv") 

head(sperm_siRNA_loci)
```

```{r}
zygote_special_simplify <- zygote_special %>% 
  mutate(X1 = as.numeric(X1)) %>% 
    mutate(X2 = as.numeric(X2)) %>% 
    mutate(X3 = as.numeric(X3)) %>% 
    filter(is.na(X1) == F) %>% 
  filter(str_detect(subset, "sperm intersect NOT")) %>% 
  select(X1, X2, X3) %>% 
  as.matrix()

colnames(zygote_special_simplify) <- NULL
head(zygote_special_simplify)
```

```{r}
match_location <- function(df, r){
  df %>% 
  filter(Chr == r[1]) %>% 
  mutate(overlap = case_when(
    end >= r[2] & end <= r[3] ~ "left",
    start <= r[3] & start >= r[2] ~ "right", 
    start <= r[2] & end >= r[3] ~ "both",
    T ~ "no"
  )) %>%   
    filter(overlap != "no") %>% 
  #mutate(anchor = (start + end)/2) %>% 
  #filter(anchor > r[2] &
  #         anchor < r[3]) %>% 
  summarise(
    RPM = sum(RPM),
    Length = sum(Length),
    Chr = min(Chr),
    start = min(start),
    end = max(end) 
  )  
}
```

```{r}
head(zygote_sperm_intersect_not_seedling %>% 
       mutate(X1 = as.numeric(X1)) %>% 
    mutate(X2 = as.numeric(X2)) %>% 
    mutate(X3 = as.numeric(X3)) %>% 
    filter(is.na(X1) == F) )
```


```{r}
sperm_exp <- apply(FUN = match_location, df = sperm_siRNA_loci, zygote_special_simplify, MARGIN = 1) %>% 
  bind_rows() %>% 
  cbind(zygote_sperm_intersect_not_seedling %>% 
          mutate(X1 = as.numeric(X1)) %>% 
    mutate(X2 = as.numeric(X2)) %>% 
    mutate(X3 = as.numeric(X3)) %>% 
    filter(is.na(X1) == F))

head(sperm_exp)
```

```{r}
summary(sperm_exp$RPM)
summary(sperm_exp$X7)
```

```{r}
test <- cor.test(sperm_exp$RPM, sperm_exp$X7, method = "s")
test$p.value
test$estimate
```

```{r}
sperm_exp %>% 
  ggplot(aes(x = RPM, y = X7 + 1)) +
  geom_jitter(width = 0.1, size = 2, alpha = 0.8, shape = 21, color = "white", fill = "black") +
  scale_x_log10(breaks = c(10, 100, 1000), limits = c(1, 10000)) +
  scale_y_log10(breaks = c(10, 100, 1000), limits = c(1, 10000)) +
  labs(x = "RPM in sperm",
       y = "RPM in zygote") +
  theme_minimal() +
  theme(axis.line = element_line(size = 1.2)) +
  theme(panel.grid = element_blank()) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black")) +
  annotation_logticks() +
  coord_fixed()


```
```{r}
match_location2 <- function(df, r){
  df %>% 
  filter(Chr == r[1]) %>% 
  mutate(overlap = case_when(
    end >= r[2] & end <= r[3] ~ "left",
    start <= r[3] & start >= r[2] ~ "right", 
    start <= r[2] & end >= r[3] ~ "both",
    T ~ "no"
  )) %>%   
    filter(overlap != "no") 
}
```

```{r}
sperm_exp_2 <- apply(FUN = match_location2, df = sperm_siRNA_loci, zygote_special_simplify, MARGIN = 1) %>% 
  bind_rows() 

head(sperm_exp_2)
```

 

```{r}
801/79990
sperm_loci_top <- sperm_siRNA_loci %>% 
  mutate(rank = rank(RPM, ties.method = "first")) %>% 
  mutate(percent.rank = rank/max(rank) * 100) %>% 
  filter(percent.rank > 99) 


  #mutate(member = case_when(
  #  Name %in% sperm_exp_2$Name ~ "Y",
  #  T ~ "N"
  #))
  
head(sperm_loci_top)
```

```{r}
sperm_exp_2 %>% 
  filter(Name %in% sperm_loci_top$Name) %>% 
  summarise(cutoff = min(RPM))
```

```{r}
sperm_exp %>% 
  mutate(top1 = case_when(
    RPM >= 31.4 ~ "Y",
    T ~ "N"
  )) %>% 
  ggplot(aes(x = RPM, y = X7 + 1)) +
  geom_jitter(width = 0.1, size = 2, alpha = 0.8, shape = 21, color = "white", 
              aes(fill = top1)) +
  annotate(geom = "text", 
           label = paste("rho = ", signif(test$estimate, 2), "\nSpearman", sep = ""),
           x = Inf, y = Inf, hjust = 1.1, vjust = 1.1, size = 5, fontface = "bold") +
  scale_fill_manual(values = c("black", "dodgerblue2"),
                    limits = c("N", "Y")) +
  scale_x_log10(breaks = c(10, 100, 1000), limits = c(1, 10000)) +
  scale_y_log10(breaks = c(10, 100, 1000), limits = c(1, 10000)) +
  labs(x = "RPM in sperm",
       y = "RPM in zygote",
       fill = "Top 1% RPM\nin sperm") +
  guides(fill = guide_legend(nrow = 1)) +
  theme_minimal() +
  theme(axis.line = element_line(size = 1.2)) +
  theme(panel.grid = element_blank()) +
  theme(text = element_text(size = 18, face="bold", colour = "black")) +
  theme(axis.text.x = element_text(colour = "black")) + 
  theme(axis.text.y = element_text(colour = "black")) +
  theme(legend.title = element_text(size = 14, hjust = 0.5), 
        legend.position = "bottom") +
  annotation_logticks() +
  coord_fixed()

ggsave("sperm_vs_z_not_seedling.svg", height = 5, width = 4)
ggsave("sperm_vs_z_not_seedling.png", height = 5, width = 4) 
```



#write beds
```{r}
write_delim(zygote_specific, delim = "\t", col_names = F, "zygote_specific.bed")
write_delim(zygote_denovo_seedling_not_sperm, delim = "\t", col_names = F, "zygote_de_novo_seedling_intersect_NOT_sperm.bed")
write_delim(zygote_sperm_intersect_not_seedling, delim = "\t", col_names = F, "zygote_de_novo_sperm_intersect_NOT_seedling.bed")
write_delim(zygote_denovo_three_way, delim = "\t", col_names = F, "zygote_de_novo_seedling_intersect_sperm_intersect.bed")
```


```{r}
write_excel_csv(zygote_special_dist, "zygote_special_dist.csv") 
```


#TE overlaps
##data
```{r}
TE_cov_list <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_special_TE_cov", pattern = "*.cov", full.names = T)

TE_overlaps <- sapply(TE_cov_list, read_table2, simplify = FALSE, col_names = F) %>% 
  bind_rows(.id = "id")

head(TE_overlaps)
```

```{r}
TE_overlap_counts <- TE_overlaps %>% 
  mutate(overlap = case_when(
    X26 == 0 ~ "N",
    X26 >0 ~ "Y"
  )) %>% 
  filter(overlap == "Y") %>% 
  group_by(id) %>% 
  count() %>% 
   mutate(TE = case_when(
   str_detect(id, "TIR") ~ "TIR",      
   str_detect(id, "CentO") ~ "CentO",
   str_detect(id, "Gypsy") ~ "Gypsy",
   str_detect(id, "Copia") ~ "Copia",
   str_detect(id, "LINE") ~ "LINE",
   str_detect(id, "SINE") ~ "SINE",
   str_detect(id, "CACTA") ~ "CACTA",
   str_detect(id, "Helitron") ~ "Helitron",
   str_detect(id, "gene") ~ "genes"
  )) %>% 
   mutate(ID = case_when(
     str_detect(id, "zygote_specific") ~ "zygote specific",
     str_detect(id, "sperm_intersect_NOT") ~ "zygote de novo\nsperm intersect\nNOT seedling",
     str_detect(id, "seedling_intersect_NOT") ~ "zygote de novo\nseedling intersect\nNOT sperm",
     str_detect(id, "seedling_intersect_sperm") ~ "zygote de novo\nseedling intersect\nsperm intersect"
   ))

head(TE_overlap_counts)
```


```{r}
total <- TE_overlaps %>% 
  filter(str_detect(id, "CACTA")) %>% 
  group_by(id) %>% 
  count() %>% 
  mutate(ID = case_when(
     str_detect(id, "zygote_specific") ~ "zygote specific",
     str_detect(id, "sperm_intersect_NOT") ~ "zygote de novo\nsperm intersect\nNOT seedling",
     str_detect(id, "seedling_intersect_NOT") ~ "zygote de novo\nseedling intersect\nNOT sperm",
     str_detect(id, "seedling_intersect_sperm") ~ "zygote de novo\nseedling intersect\nsperm intersect"
   )) %>% 
  mutate(total = n) %>% 
  ungroup() %>% 
  select(-id, -n)
```

```{r}
TE_overlap_proportion <- TE_overlap_counts %>% 
  full_join(total, by = "ID") %>% 
  mutate(percent = n/total * 100) 

head(TE_overlap_proportion)
```

##plot
```{r}
TE_overlap_proportion %>% 
  mutate(TE = factor(TE, levels = c(
    "Helitron", "SINE", "LINE", "TIR", "CentO", "Copia", "CACTA", "Gypsy" 
  ))) %>% 
  filter(TE != "genes") %>% 
  ggplot(aes(x = ID, y = percent)) +
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  scale_fill_manual(values = c("indianred1", "indianred3", "indianred4", 
                               "orange2", "seagreen", "skyblue2", "skyblue4", "grey70"),
                   limits = c("Helitron", "SINE", "LINE", "TIR", "CentO", "Copia", "CACTA", "Gypsy")) +
  labs(x = NULL,
       y = "overlapping loci (%)",
       fill = NULL) +
  guides(fill = guide_legend(nrow = 2)) +
  theme_minimal() +
  theme(legend.position = "bottom") +
  theme(text = element_text(size = 18, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black", hjust = 0.5)) +
  coord_flip()

ggsave("zygote_special_TE_stacked.svg", height = 5, width = 7)
ggsave("zygote_special_TE_stacked.png", height = 5, width = 7)
```

```{r}
TE_overlap_proportion %>% 
  filter(TE == "TIR"|
           TE == "Gypsy") %>% 
  mutate(ID = factor(ID, levels = c(
    "zygote specific",
    "zygote de novo\nsperm intersect\nNOT seedling",
     "zygote de novo\nseedling intersect\nsperm intersect",
     "zygote de novo\nseedling intersect\nNOT sperm"
  ))) %>% 
  ggplot(aes(x = TE, y = percent)) +
  facet_grid(ID ~ ., switch = "y") + 
  geom_bar(stat = "identity", aes(fill = TE), alpha = 0.8) +
  geom_text(aes(label = paste(
    signif(percent, 2), "%", sep = ""
  )), 
            size = 5, fontface = "bold", hjust = 1) +
  scale_fill_manual(values = c("grey70", "orange2")) +
  scale_x_discrete(labels = NULL) +
  labs(x = NULL,
       y = "overlapping loci (%)",
       fill = NULL) +
  theme_minimal() +
  theme(legend.position = "bottom") +
  theme(text = element_text(size = 18, face="bold")) +
  theme(panel.grid.minor.y = element_blank(), 
        panel.grid.major.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(axis.text.x=element_text(colour = "black")) +
  theme(axis.text.y=element_text(colour = "black", hjust = 0.5)) +
  theme(strip.placement = "outside", legend.position = c(0.85, 0.85)) +
  theme(strip.text.y = element_text(angle = 180, face = "bold")) +
  coord_flip() 

ggsave("zygote_special_TE_overlap_2.svg", width = 7, height = 4.5)
ggsave("zygote_special_TE_overlap_2.png", width = 7, height = 4.5)
```

#length dist
```{r}
zygote_special_length <- zygote_special %>% 
  mutate(length = (X3- X2)/1000) 
```

```{r}
pairwise.wilcox.test(x = zygote_special_length$length, g = zygote_special_length$subset)$p.value %>% 
  as.data.frame()
```

```{r}
zygote_special_length_s <- zygote_special_length %>% 
  group_by(subset) %>% 
  summarise(
    median = median(length),
            IQR = IQR(length),
            Q3 = quantile(length, 0.75),
            Q1 = quantile(length, 0.25),
            UL = quantile(length, 0.975),
            LL = quantile(length, 0.025)
  ) %>% 
  ungroup() 

zygote_special_length_s
```

#length profile, relative abundances
##data
```{r}
length_profile_list <- list.files(path = "C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/zygote_special_length_summary", pattern = "*.txt", full.names = T)

length_data  <- sapply(length_profile_list, read_delim, simplify = FALSE, delim = "\t") %>% 
  bind_rows(.id = "id")

head(length_data)
```

```{r}
sample_des <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/sample_des_clvs12.xlsx")

smRNA_comp <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS12/codes_and_output/smRNA_comp.csv")
```

```{r}
sample_des_size <- smRNA_comp %>% 
  inner_join(sample_des, by = c("sample_ID")) %>% 
  select(-2, -3, -4, -5, -6)
sample_des_size
```
```{r}
length_data <- length_data %>%
  mutate(library = X1) %>% 
    mutate(sample_ID = case_when(
    str_detect(library, "sperm.1") ~ "BS1", #1
    str_detect(library, "Sperm2") ~ "BS2",  #2
    str_detect(library, "Sperm4a") ~ "BS4a",  #3
    str_detect(library, "sperm4b") ~ "BS4b",  #4
    str_detect(library, "sperm6a") ~ "BS6a",  #5
    str_detect(library, "sperm6b") ~ "BS6b",  #6
    str_detect(library, "sperm.5b") ~ "SP5b", #7
    str_detect(library, "sperm.5c") ~ "SP5c", #8
    str_detect(library, "Bulk.15") ~ "B15",   #9
    str_detect(library, "Bulk.20") ~ "B20",   #10
    str_detect(library, "Bulk.25") ~ "B25",   #11
    str_detect(library, "seedling4") ~ "SD4", #12
    str_detect(library, "egg.9a") ~ "E9a",    #13
    str_detect(library, "egg.9b") ~ "E9b",    #14
    str_detect(library, "egg.9c") ~ "E9c",    #15
    str_detect(library, "eggcell4") ~ "E4",   #16
    str_detect(library, "eggcell3") ~ "E3",   #17
    str_detect(library, "eggcell2") ~ "E2",   #18
    str_detect(library, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(library, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(library, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(library, "SRR57") ~ "root",              #22
    str_detect(library, "ab_control") ~ "ddm1_con",     #23
    str_detect(library, "ddm1ab") ~ "ddm1",             #24
    str_detect(library, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(library, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(library, "SRR2544786") ~ "em_dry", #27  
    str_detect(library, "SRR5049778") ~ "an_pre",  #28
    str_detect(library, "SRR5049779") ~ "an_mei",  #29
    str_detect(library, "SRR5049780") ~ "an_mic",  #30
    str_detect(library, "SRR5049781") ~ "an_bi",   #31
    str_detect(library, "SRR5049786") ~ "ov_1st",  #32
    str_detect(library, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(library, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(library, "SRR5049789") ~ "ov_4th",  #35
    str_detect(library, "S3_1") ~ "S3_1",          #36
    str_detect(library, "S3_2") ~ "S3_2",          #37
    str_detect(library, "S3_3") ~ "S3_3",          #38
    str_detect(library, "S5_1") ~ "S5_1",          #39
    str_detect(library, "S5_2") ~ "S5_2",          #40
    str_detect(library, "S5_3") ~ "S5_3",          #41
    str_detect(library, "S7_1") ~ "S7_1",          #42
    str_detect(library, "S7_2") ~ "S7_2",          #43
    str_detect(library, "S7_3") ~ "S7_3",          #44
    str_detect(library, "SRR2544787") ~ "em_12h",  #45
    str_detect(library, "SRR2544788") ~ "em_24h",  #46
    str_detect(library, "SRR771501") ~ "em_7DAF",  #47
    str_detect(library, "SRR771500") ~ "en_7DAF",   #48
    str_detect(library, "SD7") ~ "SD7", #49
    str_detect(library, "zg1d") ~ "zg1d", #50
    str_detect(library, "zg1e") ~ "zg1e",  #51
    str_detect(library, "zg2d") ~ "zg2d",  #52
    str_detect(library, "zg2e") ~ "zg2e",  #53
    str_detect(library, "zg3a") ~ "zg3a", #54
    str_detect(library, "zg3b") ~ "zg3b"  #55
  )) %>% 
    inner_join(sample_des_size %>% 
                 select(`total read`, `general siRNA`, sample_ID, sample_type, genotype), by = "sample_ID")

head(length_data)
```

```{r}
length_data1 <- length_data %>% 
  mutate(category = case_when(
     str_detect(id, "zygote_specific") ~ "zygote specific",
     str_detect(id, "sperm_intersect_NOT") ~ "zygote de novo\nsperm intersect\nNOT seedling",
     str_detect(id, "seedling_intersect_NOT") ~ "zygote de novo\nseedling intersect\nNOT sperm",
     str_detect(id, "seedling_intersect_sperm") ~ "zygote de novo\nseedling intersect\nsperm intersect"
   )) %>%
  mutate(category = factor(category, levels = c(
      "zygote specific", 
      "zygote de novo\nsperm intersect\nNOT seedling",
      "zygote de novo\nseedling intersect\nNOT sperm",
      "zygote de novo\nseedling intersect\nsperm intersect" 
  ))) %>% 
  filter(sample_type == "egg"|
           sample_type == "seedling shoot" |
           sample_type == "sperm" |
           sample_type == "zygote"|
           str_detect(sample_type, "DAF")) %>%
  filter(str_detect(sample_type, "endo") == F) %>% 
  gather("length", "count", c(`20`, `21`, `22`, `23`, `24`, `25`)) %>% 
  select(category, sample_type, sample_ID, length, count, `general siRNA`) %>% 
  mutate(proportion = count / `general siRNA`)

head(length_data1)
```
```{r}
length_data1 %>% 
  filter(sample_type == "zygote") %>% 
  filter(category == "zygote specific") %>% 
  group_by(sample_ID) %>% 
  summarise(percent = sum(proportion * 100))
```

##plot 
```{r}
length_data1 %>% 
  mutate(sample_type = case_when(
    str_detect(sample_type, "embryo") ~ "embryo",
    str_detect(sample_type, "seedling") ~ "seedling",
    T ~ sample_type
  )) %>% 
  ggplot(aes(x = length, y = proportion * 100)) +
  facet_grid(. ~ category, switch = "y") +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type), 
               position = position_dodge(0.01), width = 0.5) + 
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2", "grey20"),
                    limits = c("egg", "seedling", "sperm", "zygote", "embryo")) +
  labs(color = "siRNA transcriptome",
       x = "length (nt)",
       y = "% siRNA") +
  guides(color = guide_legend(nrow = 2)) + 
  theme_minimal() +
  theme(strip.placement = "outside") +
  theme(legend.position = "bottom") +
  theme(strip.text.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(colour = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black")) 

ggsave("zygote_specific_length.svg", height = 4, width = 8.5)
ggsave("zygote_specific_length.png", height = 4, width = 8.5)
```

```{r}
length_data1 %>% 
  filter(str_detect(category, "specific")) %>% 
  ggplot(aes(x = length, y = proportion * 100)) +
  facet_grid(. ~ category, switch = "y") +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.5, alpha = 0.7) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(color = sample_type), 
               position = position_dodge(0.01), width = 0.5) + 
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "violetred2", "grey20"),
                    limits = c("egg", "seedling shoot", "sperm", "zygote", "embryo 7-8 DAF")) +
  labs(color = "siRNA transcriptome",
       x = NULL,
       y = NULL) +
  guides(color = guide_legend(nrow = 2)) + 
  theme_minimal() +
  theme(strip.placement = "outside") +
  theme(legend.position = "none") +
  theme(strip.text.y = element_blank()) +
  theme(axis.line = element_line(size = 1.2)) +
  theme(text = element_text(size= 18, face="bold")) +
  theme(axis.text.x=element_text(color = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y=element_text(colour = "black"))  

```


#graphical summary
```{r}
non_quant <- data.frame(
  tissue = c("egg cell", "zygote", "embryo", "seedling"),
  ov = c(75, 75, 5, 1),
  z_sp = c(0, 20, 0, 0),
  can = c(0, 0, 70, 100)
) %>% 
  gather("siRNA_type", "abundance", 2:4) 

non_quant
```
```{r}
set.seed(666)
non_quant2 <- data.frame(
  x = seq(0, 40, by = 0.01)
) %>% 
  mutate(zsp = 1/(1*sqrt(2*pi)) * exp(-(x - 15)^2/2^2)) %>% 
  mutate(ov = 1.8/(1 + exp(-(-1.5) *(x - 18)))) %>% 
  mutate(sd = 2/(1 + exp(-(0.5) *(x - 19)))) %>% 
  gather("type", "height", 2:4)

head(non_quant2)
```
```{r}
non_quant2 %>% 
  mutate(y = rep(1, nrow(non_quant2))) %>% 
  mutate(type = case_when(
    type == "zsp" ~ "zygote specific siRNA",
    type == "ov" ~ "ovary siren siRNA",
    type == "sd" ~ "canonical vegetative siRNA"
  )) %>% 
  mutate(type = factor(type, levels = c(
    c("ovary siren siRNA",
                               "zygote specific siRNA",
                               "canonical vegetative siRNA")
  ))) %>% 
  filter(str_detect(type, "specific") == F) %>% 
  ggplot(aes(x = x, y = y, height = height)) +
  facet_grid(type ~., space = "free", scales = "free") +
  #geom_rect(fill = "grey80", xmin = 0, xmax = 40, ymin = 0, ymax = Inf) +
  geom_ridgeline(aes(group = type, fill = type), alpha = 0.8,
                 size = 1.1) +
  scale_fill_manual(values = c("tomato1", "seagreen"),
                    limits = c("ovary siren siRNA",
                               #"zygote specific siRNA",
                               "canonical vegetative siRNA")) +
  scale_x_continuous(breaks = c(5, 14, 25, 35),
                     labels = c("egg", "zygote", "embryo", "seedling")) + 
  geom_vline(xintercept = 10, size = 0.75) +
  geom_vline(xintercept = 18, size = 0.75) +
  labs(x = NULL,
       y = NULL,
       fill = NULL) +
  guides(fill = guide_legend(nrow = 3)) +
  theme_minimal() +
  theme(
    panel.grid = element_blank(),
    legend.position = "bottom",
    text = element_text(size = 18, color = "black", face = "bold"), 
    axis.text.x = element_text(color = "black", hjust = 0.5),
    axis.text.y = element_blank(),
    strip.text = element_blank()
    )

ggsave("non_quantitative_summary.svg", height = 5, width = 5)
ggsave("non_quantitative_summary.png", height = 5, width = 5)
```
 

 








